--- a/src/TabBar.hpp
+++ b/src/TabBar.hpp
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <QApplication>
+#include <atomic>
 #include <QDrag>
 #include <QJsonDocument>
 #include <QJsonObject>
@@ -18,6 +19,7 @@ class TabBar : public QTabBar
     Q_OBJECT
 
 public:
+    static inline std::atomic<bool> s_drop_accepted{false};
     static constexpr const char *MIME_TYPE = "application/lektra-tab";
     explicit TabBar(QWidget *parent = nullptr);
     struct TabData
