--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,9 @@
 find_package(Qt6 REQUIRED COMPONENTS Gui Widgets Core Concurrent) #OpenGLWidgets )
 find_package(PkgConfig REQUIRED)
 
+# Use system mupdf
+pkg_check_modules(MUPDF REQUIRED mupdf)
+
 # SyncTeX
 find_path(SYNCTEX_INCLUDE_DIR
     NAMES synctex_parser.h synctex/synctex_parser.h
@@ -127,9 +130,8 @@
 qt_add_resources(RESOURCES resources.qrc)
 
 # Add MuPDF headers
-target_include_directories(${PROJECT_NAME} PRIVATE
-    ${CMAKE_SOURCE_DIR}/external/mupdf/include
-)
+target_include_directories(${PROJECT_NAME} PRIVATE ${MUPDF_INCLUDE_DIRS})
+target_link_directories(${PROJECT_NAME} PRIVATE ${MUPDF_LIBRARY_DIRS})
 
 target_sources(${PROJECT_NAME} PRIVATE ${RESOURCES})
 
@@ -147,8 +149,7 @@
 
 target_link_libraries(${PROJECT_NAME} PRIVATE
     Qt6::Gui Qt6::Widgets Qt6::Core Qt6::Concurrent
-    ${CMAKE_SOURCE_DIR}/external/mupdf/build/release/libmupdf.a
-    ${CMAKE_SOURCE_DIR}/external/mupdf/build/release/libmupdf-third.a
+    ${MUPDF_LIBRARIES}
 )
 
 