*** src/modules/mod_png/mptr_png.cpp.old	2010-06-24 17:12:13.108466453 +0200
--- src/modules/mod_png/mptr_png.cpp	2010-06-24 17:13:14.333341937 +0200
***************
*** 309,315 ****
  				float b=gamma().b_U8_to_F32((unsigned char)png_ptr->palette[row_pointers[y][x]].blue);
  				float a=1.0;
  				if(info_ptr->valid & PNG_INFO_tRNS)
! 				    a = (float)(unsigned char)png_ptr->trans[row_pointers[y][x]]*(1.0/255.0);
  				surface_buffer[y][x]=Color(
  					r,
  					g,
--- 309,315 ----
  				float b=gamma().b_U8_to_F32((unsigned char)png_ptr->palette[row_pointers[y][x]].blue);
  				float a=1.0;
  				if(info_ptr->valid & PNG_INFO_tRNS)
! 				    a = (float)(unsigned char)png_ptr->trans_alpha[row_pointers[y][x]]*(1.0/255.0);
  				surface_buffer[y][x]=Color(
  					r,
  					g,

