--- wrapper/ibus/SConstruct	2010-07-08 11:11:33.541686978 +0800
+++ wrapper/ibus/SConstruct	2010-07-08 11:23:18.291618045 +0800
@@ -81,10 +81,8 @@
     if not conf.CheckPKG('ibus-1.0'):
         Exit(1)
 
-    if not conf.CheckPKG('sunpinyin-2.0'):
-        Exit(1)
     env = conf.Finish()
-    env.ParseConfig('pkg-config ibus-1.0 sunpinyin-2.0 --libs --cflags')
+    env.ParseConfig('pkg-config ibus-1.0 --libs --cflags')
 
 DoConfigure()
 
--- wrapper/xim/SConstruct	2010-07-08 11:11:33.578353715 +0800
+++ wrapper/xim/SConstruct	2010-07-08 11:25:12.771607009 +0800
@@ -94,11 +94,8 @@
     if not conf.CheckPKG('x11'):
         Exit(1)
 
-    if not conf.CheckPKG('sunpinyin-2.0'):
-        Exit(1)
-
     env = conf.Finish()
-    env.ParseConfig('pkg-config gtk+-2.0 x11 sunpinyin-2.0 --libs --cflags')
+    env.ParseConfig('pkg-config gtk+-2.0 x11 --libs --cflags')
     
 DoConfigure()
 
