diff -Naur sunpinyin-2.0.orig/wrapper/xim/settings.c sunpinyin-2.0/wrapper/xim/settings.c
--- sunpinyin-2.0.orig/wrapper/xim/settings.c	2010-04-10 20:37:24.000000000 +0900
+++ sunpinyin-2.0/wrapper/xim/settings.c	2010-05-04 10:16:45.000000000 +0900
@@ -294,7 +294,7 @@
     if (fp == NULL) {
         char config_dir[256];
         snprintf(config_dir, 256, "%s/.sunpinyin", getenv("HOME"));
-        mkdir(config_dir, 0600);
+        mkdir(config_dir, 0700);
         char cmd[256];
         snprintf(cmd, 256, "cp %s %s", DEFAULT_SETTING_FILE, path);
         system(cmd);
