diff -Nur qstardict-0.13.1.orig/plugins/stardict/dictziplib.cpp qstardict-0.13.1/plugins/stardict/dictziplib.cpp
--- qstardict-0.13.1.orig/plugins/stardict/dictziplib.cpp	2009-02-10 20:33:29.000000000 +0800
+++ qstardict-0.13.1/plugins/stardict/dictziplib.cpp	2010-07-02 17:41:16.488129188 +0800
@@ -305,7 +305,7 @@
 
 #ifdef Q_OS_UNIX
     if (stat(fname.c_str(), &sb) || !S_ISREG(sb.st_mode))
-#elif def Q_OS_WIN32
+#elif defined Q_OS_WIN32
     if (_stat(fname.c_str(), &sb) || !(sb.stMode & _S_IFREG))
 #endif
     {
