From d8cb6a4ef1b51656566529f2315aa0828d693846 Mon Sep 17 00:00:00 2001
From: Nikoli <nikoli@lavabit.com>
Date: Tue, 17 Apr 2012 11:31:27 +0200
Subject: [PATCH] lsinitrd: support symlinks

---
 lsinitrd.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lsinitrd.sh b/lsinitrd.sh
index 1eaa37d..e6767dc 100755
--- a/lsinitrd.sh
+++ b/lsinitrd.sh
@@ -34,7 +34,7 @@ image="${1:-/boot/initramfs-$(uname -r).img}"
 [[ -f "$image" ]]    || { echo "$image does not exist" ; exit 1 ; }
 
 CAT=zcat
-FILE_T=$(file "$image")
+FILE_T=$(file --dereference "$image")
 
 if echo "test"|xz|xz -dc --single-stream >/dev/null 2>&1; then
     XZ_SINGLE_STREAM="--single-stream"
-- 
1.7.8.6

