From 7f3e0c13f1c10e93c3823b6367f9ac83865f3d90 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20=C5=BBo=C5=82nowski?= <aidecoe@aidecoe.name>
Date: Fri, 25 Feb 2011 01:44:25 +0100
Subject: [PATCH] 90multipath: added kpartx.rules; multipath.rules - different prefix

---
 modules.d/90multipath/module-setup.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/modules.d/90multipath/module-setup.sh b/modules.d/90multipath/module-setup.sh
index f78517c..3f4f612 100755
--- a/modules.d/90multipath/module-setup.sh
+++ b/modules.d/90multipath/module-setup.sh
@@ -58,6 +58,6 @@ install() {
 
     inst_hook pre-trigger 02 "$moddir/multipathd.sh"
     inst_hook pre-pivot   02 "$moddir/multipathd-stop.sh"
-    inst_rules 40-multipath.rules
+    inst_rules 40-multipath.rules 65-multipath.rules 66-kpartx.rules
 }
 
-- 
1.7.4.1

