From 86b2798e43f41412278397b8089ef3367529eaa7 Mon Sep 17 00:00:00 2001
From: Xiangzhe <xiangzhedev@gmail.com>
Date: Sun, 10 May 2026 12:04:11 +0800
Subject: [PATCH] Skip libvncserver when building with neatvnc

Signed-off-by: Xiangzhe <xiangzhedev@gmail.com>
---
 reframe-server/meson.build | 42 ++++++++++++++++++++------------------
 1 file changed, 22 insertions(+), 20 deletions(-)

diff --git a/reframe-server/meson.build b/reframe-server/meson.build
index 8a27483..321ec82 100644
--- a/reframe-server/meson.build
+++ b/reframe-server/meson.build
@@ -49,29 +49,31 @@ executable(
   install: true
 )
 
-lvnc_sources = files('rf-lvnc-server.c')
+if not get_option('neatvnc') or not neatvnc.found()
+  lvnc_sources = files('rf-lvnc-server.c')
 
-lvnc_headers = files('rf-lvnc-server.h')
+  lvnc_headers = files('rf-lvnc-server.h')
 
-lvnc_dependencies = []
-libvncserver = dependency('libvncserver', required: true)
-lvnc_dependencies += [
-  glib,
-  gio,
-  gmodule,
-  gobject,
-  libvncserver,
-  reframe_common_dep
-]
+  lvnc_dependencies = []
+  libvncserver = dependency('libvncserver', required: true)
+  lvnc_dependencies += [
+    glib,
+    gio,
+    gmodule,
+    gobject,
+    libvncserver,
+    reframe_common_dep
+  ]
 
-shared_module(
-  meson.project_name() + '-libvncserver',
-  sources: lvnc_sources,
-  dependencies: lvnc_dependencies,
-  include_directories: include_directories,
-  install: true,
-  install_dir: moduledir / 'vnc'
-)
+  shared_module(
+    meson.project_name() + '-libvncserver',
+    sources: lvnc_sources,
+    dependencies: lvnc_dependencies,
+    include_directories: include_directories,
+    install: true,
+    install_dir: moduledir / 'vnc'
+  )
+endif
 
 if get_option('neatvnc') and neatvnc.found()
   nvnc_sources = files('rf-nvnc-server.c')
