--- common/lib/modules/fglrx/build_mod/firegl_public.c	2012-10-18 00:29:21.778369464 +0200
+++ common/lib/modules/fglrx/build_mod/firegl_public.c.new	2012-10-18 00:30:39.647416026 +0200
@@ -222,6 +222,10 @@
 #define preempt_enable()
 #endif
 
+#ifndef VM_RESERVED
+#define  VM_RESERVED   (VM_DONTEXPAND | VM_DONTDUMP)
+#endif
+
 // ============================================================
 /* globals */
