--- src/gui/kernel/qwidget.cpp
+++ src/gui/kernel/qwidget.cpp
@@ -9095,8 +9095,6 @@
         break;
 #endif
     case Qt::WA_NativeWindow:
-        if (!qApp->testAttribute(Qt::AA_DontCreateNativeWidgetSiblings) && parentWidget())
-            parentWidget()->d_func()->enforceNativeChildren();
         if (on && !internalWinId() && testAttribute(Qt::WA_WState_Created))
             d->createWinId();
         break;
