--- tools/qdoc3/tokenizer.cpp
+++ tools/qdoc3/tokenizer.cpp
@@ -475,7 +475,7 @@ void Tokenizer::initialize(const Config
     defines = new QRegExp(d.join("|"));
     falsehoods = new QRegExp(config.getStringList(CONFIG_FALSEHOODS).join("|"));
 
-    memset(kwordHashTable, sizeof(kwordHashTable), 0);
+    memset(kwordHashTable, 0, sizeof(kwordHashTable));
     for (int i = 0; i < Tok_LastKeyword - Tok_FirstKeyword + 1; i++)
 	insertKwordIntoHash(kwords[i], i + 1);
 
