--- configure
+++ configure
@@ -25,7 +25,6 @@
 
 # initialize global variables
 QMAKE_SWITCHES=
-QMAKE_VARS=
 QMAKE_CONFIG=
 QT_CONFIG=
 
@@ -927,7 +926,7 @@
 	    QMakeVar add QMAKE_CFLAGS -pg
 	    QMakeVar add QMAKE_CXXFLAGS -pg
 	    QMakeVar add QMAKE_LFLAGS -pg
-            QMAKE_VARS="$QMAKE_VARS CONFIG+=nostrip"
+	    QMakeVar add CONFIG nostrip
         else
             UNKNOWN_OPT=yes
         fi
@@ -2045,6 +2044,8 @@
         echo "System architecture: '$ARCH'"
 fi
 
+QMakeVar add CONFIG nostrip
+
 #-------------------------------------------------------------------------------
 # tests that don't need qmake (must be run before displaying help)
 #-------------------------------------------------------------------------------
