diff --git a/kernel/sched/bmq_imp.h b/kernel/sched/bmq_imp.h
index 83c2d019c446..3faba5f9bb69 100644
--- a/kernel/sched/bmq_imp.h
+++ b/kernel/sched/bmq_imp.h
@@ -62,7 +62,7 @@ static inline void time_slice_expired(struct task_struct *p, struct rq *rq)
 	}
 }
 
-static inline int task_running_nice(struct task_struct *p)
+inline int task_running_nice(struct task_struct *p)
 {
 	return (p->prio + p->boost_prio > DEFAULT_PRIO + MAX_PRIORITY_ADJ);
 }
-- 
