--- junklib.c.old	2011-08-18 15:51:12.871996620 +0800
+++ junklib.c	2011-08-18 15:49:47.519063678 +0800
@@ -682,9 +682,9 @@
         enc = UTF8_STR;
     }
     else if (encoding == 0) {
-        // hack to add limited cp1251 recoding support
+        // hack to add limited gbk recoding support
         if (can_be_russian (str)) {
-            enc = "cp1251";
+            enc = "gbk";
         }
         else {
             enc = "iso8859-1";
@@ -766,7 +766,7 @@
     }
     const char *enc = "iso8859-1";
     if (can_be_russian (str)) {
-        enc = "cp1251";
+        enc = "gbk";
     }
 
     int len = junk_iconv (str, sz, out, sizeof (out), enc, UTF8_STR);
@@ -3385,9 +3385,9 @@
     }
     // check if that could be non-latin1 (too many nonascii chars)
     if (can_be_russian (s)) {
-        return "cp1251";
+        return "gbk";
     }
-    return "cp1252";
+    return "iso8859-1";
 }
 
 int
