diff -r 6a02537526db src/Scanner.l
--- a/src/Scanner.l	Sun Dec 04 22:02:03 2011 +0800
+++ b/src/Scanner.l	Wed Oct 31 22:36:37 2012 +0800
@@ -25,7 +25,7 @@
 #define	YY_INPUT			ReadLexBuff
 #define	YY_FATAL_ERROR		FatalLexError
 
-static void ReadLexBuff(char* pcBuff, int& riResult, size_t uMaxSize);
+static void ReadLexBuff(char* pcBuff, size_t& riResult, size_t uMaxSize);
 static void FatalLexError(yyconst char msg[]);
 
 static char* _pszLexBuff;
@@ -159,7 +159,7 @@
 
 %%
 
-static void ReadLexBuff(char* pcBuff, int& riResult, size_t uMaxSize)
+static void ReadLexBuff(char* pcBuff, size_t& riResult, size_t uMaxSize)
 {
 	wxASSERT( _pszLexBuff != NULL );
 	
@@ -168,11 +168,9 @@
 	}
 
 	wxASSERT( sizeof(YY_CHAR) == sizeof(char) );
-	size_t uCharsInBuff = strlen(_pszLexBuff);
-	size_t uCharsRead = min(uMaxSize, uCharsInBuff);
-	riResult = uCharsRead;
-	memcpy(pcBuff, _pszLexBuff, uCharsRead);
-	_pszLexBuff += uCharsRead;
+	riResult = min(uMaxSize, strlen(_pszLexBuff));
+	memcpy(pcBuff, _pszLexBuff, riResult);
+	_pszLexBuff += riResult;
 }
 
 static void FatalLexError(yyconst char msg[])
