From c406218562245056f5fde2c782c3907d2e8497cb Mon Sep 17 00:00:00 2001
From: microcai <microcai@fedoraproject.org>
Date: Wed, 7 Sep 2011 20:47:49 +0800
Subject: [PATCH] b

---
 engine.py |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/engine.py b/engine.py
index afd7f7a..29c93a5 100644
--- a/engine.py
+++ b/engine.py
@@ -63,7 +63,7 @@ class Engine(ibus.EngineBase):
         im_open_decoder()
         self.__is_invalidate = False
         self.__prepinyin_string = u""
-        self.__lookup_table = ibus.LookupTable()
+        self.__lookup_table = ibus.LookupTable(page_size=9)
 
         # 0 = english input mode
         # 1 = chinese input mode
-- 
1.7.6.1

