diff --git a/layout/KeyboardLayoutWidget.cpp b/layout/KeyboardLayoutWidget.cpp
index c104516..2823904 100644
--- a/layout/KeyboardLayoutWidget.cpp
+++ b/layout/KeyboardLayoutWidget.cpp
@@ -513,6 +513,11 @@ void KeyboardLayoutWidget::initColors()
 
 void KeyboardLayoutWidget::focusOutEvent(QFocusEvent* event)
 {
+    if (!xkb) {
+        QWidget::focusOutEvent(event);
+        return;
+    }
+
     bool update = false;
     for (int i = xkb->min_key_code; i <= xkb->max_key_code; i ++) {
         if (keys[i].pressed) {
@@ -686,6 +691,9 @@ void KeyboardLayoutWidget::initInicatorDoodad(XkbDoodadRec * xkbdoodad, Doodad&
 
 void KeyboardLayoutWidget::generatePixmap(bool force)
 {
+    if (!xkb)
+        return;
+
     double ratioX = (double) width() / xkb->geom->width_mm;
     double ratioY = (double) height() / xkb->geom->height_mm;
 
@@ -1418,6 +1426,8 @@ void KeyboardLayoutWidget::resizeEvent(QResizeEvent* event)
 bool KeyboardLayoutWidget::x11Event(XEvent* event)
 {
     do {
+        if (!xkb)
+            break;
         if (event->type != KeyPress && event->type != KeyRelease)
             break;
         DrawingKey* key = &keys[event->xkey.keycode];
diff --git a/layout/KeyboardLayoutWidget.h b/layout/KeyboardLayoutWidget.h
index 29ff256..fae83b4 100644
--- a/layout/KeyboardLayoutWidget.h
+++ b/layout/KeyboardLayoutWidget.h
@@ -4,15 +4,12 @@
 #define KEYBOARDLAYOUTWIDGET_H
 
 #include <QWidget>
-
-#include <X11/Xlib.h>
-#include <X11/extensions/XKBstr.h>
-#include <X11/extensions/XKBgeom.h>
 #include <QMap>
 
 class QPainter;
 struct Doodad;
 struct _XkbDesc;
+struct _XkbShapeDoodad;
 union _XkbDoodad;
 
 typedef enum {
@@ -104,14 +101,14 @@ protected:
     void drawKeyLabel(QPainter* painter, uint keycode, int angle, int arg4, int originY, int x2, int y2, bool pressed);
     void drawKeyLabelHelper(QPainter* painter, int keysym, int angle, int glp, int x, int y, int width, int height, int padding, bool is_pressed);
 
-    void drawShapeDoodad(QPainter* painter, Doodad* doodad, XkbShapeDoodadPtr shapeDoodad);
-    void drawTextDoodad(QPainter* painter, Doodad* doodad, XkbTextDoodadPtr textDoodad);
-    void drawIndicatorDoodad(QPainter* painter, Doodad* doodad, XkbIndicatorDoodadPtr indicatorDoodad);
+    void drawShapeDoodad(QPainter* painter, Doodad* doodad, struct _XkbShapeDoodad* shapeDoodad);
+    void drawTextDoodad(QPainter* painter, Doodad* doodad, struct _XkbTextDoodad* textDoodad);
+    void drawIndicatorDoodad(QPainter* painter, Doodad* doodad, struct _XkbIndicatorDoodad* indicatorDoodad);
 
-    int calcShapeOriginOffsetX(XkbOutlinePtr outline);
-    void drawOutline(QPainter* painter, XkbOutlinePtr outline, QColor color, int angle, int originX, int originY);
+    int calcShapeOriginOffsetX(struct _XkbOutline* outline);
+    void drawOutline(QPainter* painter, struct _XkbOutline* outline, QColor color, int angle, int originX, int originY);
     void drawRectangle(QPainter* painter, QColor color, int angle, int xkb_x, int xkb_y, int xkb_width, int xkb_height, unsigned int radius);
-    void drawPolygon(QPainter* painter, QColor color, int originX, int originY, XkbPointPtr points, unsigned int num_points, unsigned int radius);
+    void drawPolygon(QPainter* painter, QColor color, int originX, int originY, struct _XkbPoint* points, unsigned int num_points, unsigned int radius);
     void rotateCoordinate(int originX, int originY, int x, int y, int angle, int* rotated_x, int* rotated_y);
     int xkbToPixmapCoord (int n);
     double xkbToPixmapDouble (double d);
@@ -119,7 +116,7 @@ protected:
     void drawCurveRectangle(QPainter* painter, bool filled, QColor color, int x, int y, int width, int height, double radius);
     void roundedCorner (QPainterPath& path, QPointF b, QPointF c, double radius);
     void resizeEvent(QResizeEvent* event);
-    void setKeyboard(XkbComponentNamesPtr xkbDesc);
+    void setKeyboard(struct _XkbComponentNames* xkbDesc);
 
 private:
     QList<DrawingItem*> keyboardItems;
@@ -138,4 +135,4 @@ private:
     QMap<uint, uint> deadMap;
 };
 
-#endif
\ No newline at end of file
+#endif
