--- jfbterm-0.4.7.orig/term.c	2003-09-15 23:45:31.000000000 +0800
+++ jfbterm-0.4.7/term.c	2007-02-12 15:29:21.000000000 +0800
@@ -55,6 +55,8 @@
 
 #include "config.h"
 
+#include <ucimf.h>
+
 int gChildProcessId = 0;
 
 TTerm gTerm;
@@ -175,6 +177,9 @@
 	tterm_set_utmp(p);
 	signal(SIGCHLD, sigchld);
 	atexit(application_final);
+			  
+	/* Ucimf initialization */
+        ucimf_init();
 
 	/* not available
 	 * VtInit();
@@ -207,6 +212,11 @@
 		}
 		if (FD_ISSET(0, &fds)) {
 			ret = read(0, buf, BUF_SIZE);
+			
+			// ucimf input method framework switch on/off 
+			ucimf_switch( buf, &ret );
+			ucimf_process_stdin( buf, &ret );
+			
 #ifdef JFB_ENABLE_DIMMER
 			idle_time = 0;
 			if (blank) {
@@ -223,7 +223,10 @@
			if (ret > 0) {
 				/* write(1, buf, ret); */
 				tvterm_emulate(&(p->vterm), buf, ret);
+			  	ucimf_refresh_begin();
 				tvterm_refresh(&(p->vterm));
+				ucimf_cursor_position( p->vterm.pen.x * gFontsWidth, p->vterm.pen.y*gFontsHeight);
+				ucimf_refresh_end();
 			}
 		}
 	}
+	ucimf_exit();
 }
 
 void tterm_wakeup_shell(TTerm* p, const char* tn)
--- jfbterm-0.4.7.orig/Makefile.am	2003-09-14 04:58:53.000000000 +0800
+++ jfbterm-0.4.7/Makefile.am	2007-02-12 15:29:21.000000000 +0800
@@ -14,7 +14,9 @@
 		util.c util.h \
 		csv.c csv.h \
 		vterm.c vtermlow.c vterm.h
-jfbterm_LDADD = -lutil
+
+jfbterm_LDADD = -L$(libdir) -lutil -lucimf
+AM_CFLAGS = -I$(includedir)
 
 JFBTERM_CONFIG_FILE = jfbterm.conf.sample
 
