diff -ruN DBDesigner4.orig/startdbd DBDesigner4/startdbd
--- DBDesigner4.orig/startdbd	2003-10-31 01:48:13.000000000 +0100
+++ DBDesigner4/startdbd	2006-08-30 14:27:50.000000000 +0200
@@ -20,10 +20,10 @@
 # First check to see if we have an LD_LIBRARY_PATH environment variable
 if [ -n "$LD_LIBRARY_PATH" ]; then
 # we do, so prepend our path first
-  export LD_LIBRARY_PATH="$app_ld_path:$LD_LIBRARY_PATH"
+  export LD_LIBRARY_PATH="$app_ld_path:/opt/kylix3:$LD_LIBRARY_PATH"
 else
   # we do not, so we will create the env var.
-  export LD_LIBRARY_PATH="$app_ld_path"
+  export LD_LIBRARY_PATH="$app_ld_path:/opt/kylix3"
 fi
 
 # make sure we have something specified for the LANG environment variable
@@ -59,7 +59,7 @@
 fi
 
 #use the original qt files
-export CLX_USE_LIBQT=yes
+# export CLX_USE_LIBQT=yes
 
 #QT smooth fonts
 export QT_XFT=true
