--- /usr/portage/dev-db/dbdesigner/dbdesigner-4.0.5.4.ebuild	2007-01-31 09:35:52.000000000 -0400
+++ dbdesigner-4.0.5.4.ebuild	2007-08-17 14:28:13.000000000 -0400
@@ -34,7 +34,7 @@
 	ln -s dbxres.en.1.0 dbxres.en.1
 	ln -s libmidas.so.1.0 libmidas.so.1
 	ln -s libmysqlclient.so.10.0.0 libmysqlclient.so
-	ln -s libqt.so.2.3.2 libqt.so.2
+	ln -s libborqt-6.9.0-qt2.3.so libqt.so.2
 	ln -s libqtintf-6.9.0-qt2.3.so libqtintf-6.9-qt2.3.so
 	ln -s libsqlmy23.so.1.0 libsqlmy23.so
 	ln -s libsqlmy23.so libsqlmy.so
