diff -Naur scim-bridge-0.4.14.orig/client-qt/qt4/Makefile.am scim-bridge-0.4.14/client-qt/qt4/Makefile.am
--- scim-bridge-0.4.14.orig/client-qt/qt4/Makefile.am	2007-11-07 18:57:46.000000000 +0900
+++ scim-bridge-0.4.14/client-qt/qt4/Makefile.am	2008-01-28 01:24:30.000000000 +0900
@@ -24,7 +24,7 @@
 
 noinst_HEADERS = scim-bridge-client-qt.h scim-bridge-client-imcontext-qt.h  scim-bridge-client-key-event-utility-qt.h scim-bridge-client-common-qt.h
 
-moduledir = @QT4_PREFIX@/plugins/inputmethods
+moduledir = @QT4_LIBDIR@/plugins/inputmethods
 module_LTLIBRARIES = im-scim-bridge.la
 
 im_scim_bridge_la_SOURCES = im-scim-bridge-qt.cpp \
diff -Naur scim-bridge-0.4.14.orig/configure.ac scim-bridge-0.4.14/configure.ac
--- scim-bridge-0.4.14.orig/configure.ac	2007-12-06 00:50:13.000000000 +0900
+++ scim-bridge-0.4.14/configure.ac	2008-01-28 01:24:30.000000000 +0900
@@ -115,6 +115,10 @@
     QT4_PREFIX=`$PKG_CONFIG --variable=prefix QtCore`
   fi
   AC_SUBST(QT4_PREFIX)
+  if test -z "$QT4_LIBDIR"; then
+    QT4_LIBDIR=`$PKG_CONFIG --variable=libdir QtCore`
+  fi
+  AC_SUBST(QT4_LIBDIR)
 fi
 
 ###########################################################
