--- data/Makefile.am.orig	2009-01-15 03:48:07.825015023 +0800
+++ data/Makefile.am	2009-01-15 03:49:10.156017173 +0800
@@ -1,12 +1,12 @@
 tabledir = $(pkgdatadir)/data
 table_DATA = cj.mb erbi.mb punc.mb pybase.mb pyphrase.mb pySym.mb sp.dat \
-	     wbx.mb wanfeng.mb wbpy.mb qxm.mb zrm.mb db.mb \
+	     wbx.mb wanfeng.mb wbpy.mb qxm.mb zrm.mb zhengma.mb db.mb \
 	     tables.conf vk.conf gbks2t.tab config AutoEng.dat \
              pyPhrase.org qxm.txt zrm.txt erbi.txt wbpy.txt gbkpy.org db.txt \
-             wanfeng.txt cj.txt wbx.txt
+             wanfeng.txt cj.txt wbx.txt zhengma.txt
 EXTRA_DIST = $(table_DATA)
 
-mb_data = cj.mb erbi.mb wanfeng.mb wbpy.mb qxm.mb db.mb wbx.mb zrm.mb
+mb_data = cj.mb erbi.mb wanfeng.mb wbpy.mb qxm.mb db.mb wbx.mb zrm.mb zhengma.mb
 
 %.mb: %.txt
 	../tools/txt2mb $< $@

--- data/tables.conf.orig	2009-01-15 04:01:53.296256248 +0800
+++ data/tables.conf	2009-01-15 04:02:33.065002134 +0800
@@ -31,6 +31,19 @@
 提示编码=0
 
 [码表]
+名称=郑码
+码表=zhengma.mb
+调频=2
+拼音=1
+拼音键=`
+自动上屏=-1
+空码自动上屏=-1
+自动词组=1
+保存自动词组=3
+精确匹配=0
+提示编码=0
+
+[码表]
 名称=二笔
 码表=erbi.mb
 调频=0
