diff -r f6cd2db5d10f -r faf8d566e65f CMakeLists.txt
--- a/CMakeLists.txt	Sat Sep 03 13:01:35 2011 +0800
+++ b/CMakeLists.txt	Sat Sep 03 17:20:16 2011 +0800
@@ -126,10 +126,10 @@
 set(FCITX4_ADDON_CONFIG_INSTALL_DIR ${addonconfigdir})
 set(FCITX4_PACKAGE_NAME ${package})
 
-set(CMAKE_C_FLAGS "-Wall -Wextra -Wno-sign-compare -Wno-unused-parameter -fvisibility=hidden")
-set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wno-sign-compare -Wno-unused-parameter -fvisibility=hidden")
-set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-undefined,--as-needed")
-set(CMAKE_MODULE_LINKER_FLAGS "-Wl,--no-undefined,--as-needed")
+set(CMAKE_C_FLAGS "-Wall -Wextra -Wno-sign-compare -Wno-unused-parameter -fvisibility=hidden ${CMAKE_C_FLAGS}")
+set(CMAKE_CXX_FLAGS "-Wall -Wextra -Wno-sign-compare -Wno-unused-parameter -fvisibility=hidden ${CMAKE_CXX_FLAGS}")
+set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--no-undefined,--as-needed ${CMAKE_SHARED_LINKER_FLAGS}")
+set(CMAKE_MODULE_LINKER_FLAGS "-Wl,--no-undefined,--as-needed ${CMAKE_MODULE_LINKER_FLAGS}")
 add_definitions("-D_GNU_SOURCE -DFCITX_HAVE_CONFIG_H")
 
 configure_file(${PROJECT_SOURCE_DIR}/config.h.in ${PROJECT_BINARY_DIR}/config.h)
