diff -ur data/png/CMakeLists.txt data.new/png/CMakeLists.txt
--- data/png/CMakeLists.txt	2011-10-05 21:01:47.629326790 +0800
+++ data.new/png/CMakeLists.txt	2011-10-05 21:23:43.905871069 +0800
@@ -14,6 +14,7 @@
 fcitx-chttrans-inactive.png
 fcitx-eng.png
 fcitx-erbi.png
+fcitx-zhengma.png
 fcitx-remind-active.png
 fcitx-remind-inactive.png
 fcitx-pinyin.png
diff -ur data/table/CMakeLists.txt data.new/table/CMakeLists.txt
--- data/table/CMakeLists.txt	2011-10-05 21:01:47.649326542 +0800
+++ data.new/table/CMakeLists.txt	2011-10-05 21:28:24.128367807 +0800
@@ -7,12 +7,16 @@
  ${CMAKE_CURRENT_BINARY_DIR}/wbx.mb
  ${CMAKE_CURRENT_BINARY_DIR}/zrm.mb
  ${CMAKE_CURRENT_BINARY_DIR}/cj.mb
+ ${CMAKE_CURRENT_BINARY_DIR}/zm.mb
 )
+set(ZMORG_DATA zm.txt)
 set(TABLEORG_DATA db.txt erbi.txt qxm.txt wanfeng.txt wbpy.txt wbx.txt zrm.txt cj.txt)
-set(TABLE_NAME db erbi qxm wanfeng wbpy wbx zrm cj)
-set(TABLE_CONF db.conf erbi.conf qxm.conf wanfeng.conf wbpy.conf wbx.conf zrm.conf cangjie.conf)
+set(TABLE_NAME db erbi qxm wanfeng wbpy wbx zrm cj zm)
+set(TABLE_CONF db.conf erbi.conf qxm.conf wanfeng.conf wbpy.conf wbx.conf zrm.conf cangjie.conf zhengma.conf)
 set(TABLE_URL http://fcitx.googlecode.com/files/table.tar.gz)
 set(TABLE_TAR ${CMAKE_CURRENT_SOURCE_DIR}/table.tar.gz)
+set(ZM_URL https://github.com/transtone/transconfig/raw/master/zm.tar.gz)
+set(ZM_TAR ${CMAKE_CURRENT_SOURCE_DIR}/zm.tar.gz)
 
 set(INSTALL_TABLE_CONF "")
 foreach(tableconffile ${TABLE_CONF})
@@ -26,6 +30,12 @@
 add_custom_command(OUTPUT ${TABLEORG_DATA}
                    COMMAND ${PROJECT_SOURCE_DIR}/data/download.sh "${WGET}" "${TABLE_URL}" "${TABLE_TAR}"
                    COMMAND ${TAR} xzmvf ${TABLE_TAR})
+add_custom_command(OUTPUT ${ZM_TAR}
+                   COMMAND ${WGET} -O ${ZM_TAR} ${ZM_URL})
+add_custom_command(OUTPUT ${ZMORG_DATA}
+                   DEPENDS ${ZM_TAR}
+                   COMMAND ${TAR} xzmvf ${ZM_TAR})
+   
 
 foreach(tblname ${TABLE_NAME})
 add_custom_command(OUTPUT ${tblname}.mb
@@ -33,4 +43,4 @@
                    COMMAND txt2mb ${tblname}.txt ${tblname}.mb)
 endforeach()
 
-install(FILES ${TABLE_DATA} DESTINATION ${pkgdatadir}/table )
\ 文件尾没有 newline 字符
+install(FILES ${TABLE_DATA} DESTINATION ${pkgdatadir}/table )
diff -ur po/zh_CN.po po.new/zh_CN.po
--- po/zh_CN.po	2011-10-05 21:01:47.674326229 +0800
+++ po.new/zh_CN.po	2011-10-05 21:22:12.790010174 +0800
@@ -445,6 +445,10 @@
 msgid "Erbi"
 msgstr "二笔"
 
+#: /home/saber/Develop/fcitx/build/po/tmp/zhengma.conf.in.h:1
+msgid "Zhengma"
+msgstr "郑码"
+
 #: /home/saber/Develop/fcitx/build/po/../..//src/ui/classic/fcitx-skin-installer.sh:69
 #, sh-format
 msgid ""
diff -ur skin/classic/CMakeLists.txt skin.new/classic/CMakeLists.txt
--- skin/classic/CMakeLists.txt	2011-10-05 21:01:47.675326216 +0800
+++ skin.new/classic/CMakeLists.txt	2011-10-05 21:19:23.483126802 +0800
@@ -4,8 +4,8 @@
 punc_active.png punc_inactive.png
  active.png  bingchan.png erbi.png  fcitx_skin.conf
  input.png  remind_active.png remind_inactive.png  quwei.png  vk_active.png vk_inactive.png wubi.png wbpy.png ziranma.png
- bar.png cangjie.png  cn.png en.png inactive.png logo.png
+ bar.png cangjie.png  cn.png en.png inactive.png logo.png zhengma.png
  dianbaoma.png shuangpin.png wanfeng.png pinyin.png prev.png next.png menu.png
 )
 
-install(FILES ${FCITX_CLASSIC_SKIN_FILES} DESTINATION ${pkgdatadir}/skin/classic)
\ 文件尾没有 newline 字符
+install(FILES ${FCITX_CLASSIC_SKIN_FILES} DESTINATION ${pkgdatadir}/skin/classic)
diff -ur skin/dark/CMakeLists.txt skin.new/dark/CMakeLists.txt
--- skin/dark/CMakeLists.txt	2011-10-05 21:01:47.693325990 +0800
+++ skin.new/dark/CMakeLists.txt	2011-10-05 21:20:02.016645067 +0800
@@ -1,11 +1,11 @@
 set(FCITX_DARK_SKIN_FILES
 active.png  bingchan.png erbi.png  fcitx_skin.conf 
 input.png quwei.png wubi.png wbpy.png ziranma.png
-bar.png cangjie.png  cn.png en.png  inactive.png logo.png
+bar.png cangjie.png  cn.png en.png  inactive.png logo.png zhengma.png
 dianbaoma.png shuangpin.png wanfeng.png pinyin.png prev.png next.png menu.png keyboard.png
 remind_active.png remind_inactive.png vk_active.png vk_inactive.png
 punc_active.png punc_inactive.png fullwidth_active.png fullwidth_inactive.png
 chttrans_active.png chttrans_inactive.png
 )
 
-install(FILES ${FCITX_DARK_SKIN_FILES} DESTINATION ${pkgdatadir}/skin/dark)
\ 文件尾没有 newline 字符
+install(FILES ${FCITX_DARK_SKIN_FILES} DESTINATION ${pkgdatadir}/skin/dark)
diff -ur skin/default/CMakeLists.txt skin.new/default/CMakeLists.txt
--- skin/default/CMakeLists.txt	2011-10-05 21:01:47.711325766 +0800
+++ skin.new/default/CMakeLists.txt	2011-10-05 21:20:43.349128340 +0800
@@ -5,8 +5,8 @@
 active.png  bingchan.png erbi.png  fcitx_skin.conf 
 input.png  remind_active.png remind_inactive.png  quwei.png vk_active.png vk_inactive.png
 wubi.png wbpy.png ziranma.png
-bar.png cangjie.png  cn.png en.png  inactive.png logo.png
+bar.png cangjie.png  cn.png en.png  inactive.png logo.png zhengma.png
 dianbaoma.png shuangpin.png wanfeng.png pinyin.png prev.png next.png keyboard.png menu.png
 )
 
-install(FILES ${FCITX_DEFAULT_SKIN_FILES} DESTINATION ${pkgdatadir}/skin/default)
\ 文件尾没有 newline 字符
+install(FILES ${FCITX_DEFAULT_SKIN_FILES} DESTINATION ${pkgdatadir}/skin/default)
