--- src/config.h.orig	2006-09-12 14:24:54 +0000
+++ src/config.h	2006-09-12 14:25:23 +0000
@@ -39,11 +39,7 @@
 #else
 #   define LEASEFILE "/var/lib/misc/dnsmasq.leases"
 #endif
-#if defined(__FreeBSD__)
-#   define CONFFILE "/usr/local/etc/dnsmasq.conf"
-#else
-#   define CONFFILE "/etc/dnsmasq.conf"
-#endif
+#define CONFFILE "/etc/dnsmasq.conf"
 #define DEFLEASE 3600 /* default lease time, 1 hour */
 #define CHUSER "nobody"
 #define CHGRP "dip"
