diff --git a/build/config.rb b/build/config.rb
index 8905bd8..64199fb 100644
--- a/build/config.rb
+++ b/build/config.rb
@@ -24,11 +24,7 @@
 OPTIMIZE = boolean_option("OPTIMIZE")
 CC       = string_option("CC", "gcc")
 CXX      = string_option("CXX", "g++")
-if OPTIMIZE
-	OPTIMIZATION_FLAGS = "#{PlatformInfo.debugging_cflags} -O2 -DBOOST_DISABLE_ASSERTS".strip
-else
-	OPTIMIZATION_FLAGS = "#{PlatformInfo.debugging_cflags} -DPASSENGER_DEBUG -DBOOST_DISABLE_ASSERTS".strip
-end
+OPTIMIZATION_FLAGS = "#{ENV['CXXFLAGS']} -DNDEBUG -DBOOST_DISABLE_ASSERTS".strip
 
 # Extra compiler flags that should always be passed to the C/C++ compiler.
 # Should be included last in the command string, even after PlatformInfo.portability_cflags.
@@ -39,4 +35,4 @@ EXTRA_CXXFLAGS = "-Wall #{OPTIMIZATION_FLAGS}".strip
 EXTRA_LDFLAGS  = ""
 
 # Whether to use the vendored libev or the system one.
-USE_VENDORED_LIBEV = boolean_option("USE_VENDORED_LIBEV", true)
\ No newline at end of file
+USE_VENDORED_LIBEV = false
