from http://code.google.com/p/phusion-passenger/issues/detail?id=595

--- ../passenger-3.0.1/ext/nginx/config.orig	2010-12-08 13:48:23.964424406 +0000
+++ ../passenger-3.0.1/ext/nginx/config	2010-12-08 13:49:12.078865225 +0000
@@ -2,10 +2,7 @@
    ! test -f "$ngx_addon_dir/../common/libboost_oxt.a"; then
     echo "*** The Phusion Passenger support files are not yet compiled. Compiling them for you... ***"
     echo "*** Running 'rake nginx RELEASE=yes' in $ngx_addon_dir... ***"
-    if ! cd $ngx_addon_dir; then
-        exit 1
-    fi
-    if ! rake nginx RELEASE=yes; then
+    if ! (cd $ngx_addon_dir && rake nginx RELEASE=yes) then
         exit 1
     fi
     echo "*** Phusion Passenger support files have been successfully compiled. ***"
