--- build/config.rb	2010-10-01 12:22:34.000000000 +0200
+++ build/config.rb	2010-11-02 10:56:25.309314070 +0100
@@ -24,12 +24,7 @@
 OPTIMIZE = boolean_option("OPTIMIZE")
 CC       = string_option("CC", "gcc")
 CXX      = string_option("CXX", "g++")
-# TODO: consider -fcommon
-if OPTIMIZE
-	OPTIMIZATION_FLAGS = "#{PlatformInfo.debugging_cflags} -O2 -DBOOST_DISABLE_ASSERTS".strip
-else
-	OPTIMIZATION_FLAGS = "#{PlatformInfo.debugging_cflags} -DPASSENGER_DEBUG -DBOOST_DISABLE_ASSERTS".strip
-end
+OPTIMIZATION_FLAGS = "#{ENV['CXXFLAGS']}"
 
 # Extra compiler flags that should always be passed to the C/C++ compiler.
 # Should be included last in the command string, even after PlatformInfo.portability_cflags.
--- build/packaging.rb	2010-12-08 04:10:38.000000000 +0600
+++ build/packaging.rb	2011-02-20 05:05:16.000000000 +0600
@@ -94,29 +94,23 @@
 task :clobber => :'package:clean'
 
 desc "Create a fakeroot, useful for building native packages"
-task :fakeroot => [:apache2, :nginx] + Packaging::ASCII_DOCS do
+task :fakeroot => Packaging::ASCII_DOCS do
 	require 'rbconfig'
 	require 'fileutils'
 	include Config
-	fakeroot = "pkg/fakeroot"
+	fakeroot = ENV['D']
 	
 	# We don't use CONFIG['archdir'] and the like because we want
 	# the files to be installed to /usr, and the Ruby interpreter
 	# on the packaging machine might be in /usr/local.
-	fake_libdir = "#{fakeroot}/usr/lib/ruby/#{CONFIG['ruby_version']}"
-	fake_native_support_dir = "#{fakeroot}/usr/lib/ruby/#{CONFIG['ruby_version']}/#{CONFIG['arch']}"
+	fake_libdir = "#{fakeroot}/#{CONFIG['sitedir']}/#{CONFIG['ruby_version']}"
+	fake_native_support_dir = "#{fakeroot}/#{CONFIG['sitedir']}/#{CONFIG['ruby_version']}/#{CONFIG['arch']}"
 	fake_agents_dir = "#{fakeroot}#{NATIVELY_PACKAGED_AGENTS_DIR}"
 	fake_helper_scripts_dir = "#{fakeroot}#{NATIVELY_PACKAGED_HELPER_SCRIPTS_DIR}"
 	fake_resources_dir = "#{fakeroot}/usr/share/phusion-passenger"
 	fake_docdir = "#{fakeroot}#{NATIVELY_PACKAGED_DOCDIR}"
 	fake_bindir = "#{fakeroot}/usr/bin"
 	fake_sbindir = "#{fakeroot}/usr/sbin"
-	fake_source_root = "#{fakeroot}#{NATIVELY_PACKAGED_SOURCE_ROOT}"
-	fake_apache2_module = "#{fakeroot}#{NATIVELY_PACKAGED_APACHE2_MODULE}"
-	fake_apache2_module_dir = File.dirname(fake_apache2_module)
-	
-	sh "rm -rf #{fakeroot}"
-	sh "mkdir -p #{fakeroot}"
 	
 	sh "mkdir -p #{fake_libdir}"
 	sh "cp #{LIBDIR}/phusion_passenger.rb #{fake_libdir}/"
@@ -153,20 +147,6 @@
 	Packaging::SUPER_USER_EXECUTABLES.each do |exe|
 		sh "cp bin/#{exe} #{fake_sbindir}/"
 	end
-	
-	sh "mkdir -p #{fake_apache2_module_dir}"
-	sh "cp #{APACHE2_MODULE} #{fake_apache2_module_dir}/"
-	
-	sh "mkdir -p #{fake_source_root}"
-	spec.files.each do |filename|
-		next if File.directory?(filename)
-		dirname = File.dirname(filename)
-		if !File.directory?("#{fake_source_root}/#{dirname}")
-			sh "mkdir -p '#{fake_source_root}/#{dirname}'"
-		end
-		puts "cp '#{filename}' '#{fake_source_root}/#{dirname}/'"
-		FileUtils.cp(filename, "#{fake_source_root}/#{dirname}/")
-	end
 end
 
 desc "Create a Debian package"
