diff --git a/Telegram/SourceFiles/core/launcher.cpp b/Telegram/SourceFiles/core/launcher.cpp
index 25473426f..21784ea89 100644
--- a/Telegram/SourceFiles/core/launcher.cpp
+++ b/Telegram/SourceFiles/core/launcher.cpp
@@ -278,6 +278,12 @@ int Launcher::exec() {
 	Platform::start();
 	Ui::DisableCustomScaling();
 
+#if defined(Q_OS_LINUX64)
+	QCoreApplication::addLibraryPath("/usr/lib64/qt5/plugins");
+#else
+	QCoreApplication::addLibraryPath("/usr/lib/qt5/plugins");
+#endif
+
 	auto result = executeApplication();
 
 	DEBUG_LOG(("Telegram finished, result: %1").arg(result));
