--- ppstream/Makefile.orig	2009-11-08 12:16:47.000000000 +0800
+++ ppstream/Makefile	2009-11-08 13:11:17.000000000 +0800
@@ -21,12 +21,12 @@
 	g++ $(CXXFLAGS) -c -o $@ $<
 
 libs:ppswrapper-msgget.c
-	gcc -shared -Wl,-ldl -o libppswrapper-preload.so.0.0.0 ppswrapper-msgget.c
+	gcc -shared -fPIC -Wl,-ldl,-soname,libppswrapper-preload.so.0 -o libppswrapper-preload.so.0.0.0 ppswrapper-msgget.c
 	ln -s libppswrapper-preload.so.0.0.0 libppswrapper-preload.so.0.0
 	ln -s libppswrapper-preload.so.0.0 libppswrapper-preload.so.0
 	ln -s libppswrapper-preload.so.0 libppswrapper-preload.so
 clean:
 	rm -rf *.so* *.o xpps
 install:
-	cp xpps /usr/bin/
-	mv libppswrapper-preload.* /usr/lib/
+	mv xpps "${DESTDIR}"/usr/bin/
+	mv libppswrapper-preload.* "${DESTDIR}"/usr/lib/
