diff -Naur libofetion-2.1.0/fetion_types.h libofetion-2.1.0.new/fetion_types.h
--- libofetion-2.1.0/fetion_types.h	2010-12-26 12:56:54.000000000 +0800
+++ libofetion-2.1.0.new/fetion_types.h	2011-01-04 01:51:25.624000055 +0800
@@ -32,7 +32,7 @@
 #endif
 
 #define FETION_NAME "OpenFetion"
-#define FETION_VERSION "2.0.7"
+#define FETION_VERSION "2.1.0"
 #define PROTO_VERSION "4.0.2510"
 #define NAVIGATION_URI "nav.fetion.com.cn"
 #define PGGROUP_SPACE_URI "http://group.feixin.10086.cn/space/Index/"
