diff --git a/src/libime/core/kenlm/util/double-conversion/utils.h b/src/libime/core/kenlm/util/double-conversion/utils.h
index 2c745f3..e5e8892 100644
--- a/src/libime/core/kenlm/util/double-conversion/utils.h
+++ b/src/libime/core/kenlm/util/double-conversion/utils.h
@@ -71,6 +71,7 @@ inline void abort_noreturn() { abort(); }
     defined(__ARMEL__) || defined(__avr32__) || \
     defined(__hppa__) || defined(__ia64__) || \
     defined(__mips__) || \
+    defined(__loongarch64) || defined(__loongarch__) || \
     defined(__powerpc__) || defined(__ppc__) || defined(__ppc64__) || \
     defined(_POWER) || defined(_ARCH_PPC) || defined(_ARCH_PPC64) || \
     defined(__sparc__) || defined(__sparc) || defined(__s390__) || \
