From b080d6dd50ee2e0ad29d4fc36dfec8a72fe4c97c Mon Sep 17 00:00:00 2001
From: Weng Xuetian <wengxt@gmail.com>
Date: Wed, 2 Feb 2022 09:50:44 -0800
Subject: [PATCH] Fix build when wayland is disabled.

---
 src/modules/dbus/dbusmodule.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/modules/dbus/dbusmodule.cpp b/src/modules/dbus/dbusmodule.cpp
index e308cf8a..906f1b0c 100644
--- a/src/modules/dbus/dbusmodule.cpp
+++ b/src/modules/dbus/dbusmodule.cpp
@@ -553,7 +553,7 @@ class Controller1 : public ObjectVTable<Controller1> {
             return;
         }
 #else
-        FCITX_UNUSED(name);
+        FCITX_UNUSED(fd);
 #endif
         throw dbus::MethodCallError("org.freedesktop.DBus.Error.InvalidArgs",
                                     "Wayland addon is not available.");
