diff -urN audacious-plugins-1.2.2.orig/src/mpg123/fileinfo.c audacious-plugins-1.2.2/src/mpg123/fileinfo.c
--- audacious-plugins-1.2.2.orig/src/mpg123/fileinfo.c	2006-11-11 02:41:10.000000000 +0800
+++ audacious-plugins-1.2.2/src/mpg123/fileinfo.c	2006-11-11 02:42:14.000000000 +0800
@@ -117,7 +117,7 @@
   if (str_has_prefix_nocase(current_filename, "http://"))
     return;
 
-  taglib_set_strings_unicode(1);
+  taglib_set_strings_unicode(0);
 
   /* XXX: Gnome VFS workaround. -nenolod */
   if (str_has_prefix_nocase(current_filename, "file://"))
diff -urN audacious-plugins-1.2.2.orig/src/mpg123/tag_c_hacked.cxx audacious-plugins-1.2.2/src/mpg123/tag_c_hacked.cxx
--- audacious-plugins-1.2.2.orig/src/mpg123/tag_c_hacked.cxx	2006-11-11 02:42:34.000000000 +0800
+++ audacious-plugins-1.2.2/src/mpg123/tag_c_hacked.cxx	2006-09-18 18:14:20.000000000 +0800
@@ -9,6 +9,6 @@
 using namespace ID3v2;
 void taglib_set_id3v2_default_text_encoding()
 {
-  TagLib::ID3v2::FrameFactory::instance()->setDefaultTextEncoding(TagLib::String::UTF8);
+  TagLib::ID3v2::FrameFactory::instance()->setDefaultTextEncoding(TagLib::String::Latin1);
 }
 
