--- src/mplayer.py.old	2013-03-11 05:28:58.000000000 -0400
+++ src/mplayer.py	2013-04-13 20:46:04.488839176 -0400
@@ -43,7 +43,11 @@
 from ini import Config
 from cdrom.cdrom import cdrom_type, get_iso_type, CDROM_ERROR, CDROM_TYPE_DVD, CDROM_TYPE_VCD
 
-
+# get the backend name
+if os.path.exists('/usr/bin/mplayer2'):
+	BACKEND = "mplayer2"
+else:
+	BACKEND = "mplayer"
 
 
 # play list state.
@@ -76,7 +80,7 @@
 # Get play file length.
 def get_length(file_path):
     '''Get media player length.'''
-    cmd = "mplayer -vo null -ao null -frames 0 -identify '%s'" % (file_path)
+    cmd = BACKEND+" -vo null -ao null -frames 0 -identify '%s'" % (file_path)
     fp = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
     cmd_str = fp.communicate()[0]
     length_compile = re.compile(r"ID_LENGTH=([\d|\.]+)")
@@ -89,7 +93,7 @@
 def get_vide_width_height(file_path):
     try:
         if is_valid_video_file(file_path):
-            cmd = "mplayer -vo null -ao null -frames 0 -identify '%s' 2>&1" % (file_path)
+            cmd = BACKEND+" -vo null -ao null -frames 0 -identify '%s' 2>&1" % (file_path)
             fp = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)    
             cmd_str = fp.communicate()[0]
             length_compile = re.compile(r"VIDEO:.+") 
@@ -343,7 +347,7 @@
         if not (self.state): # STOPING_STATE
             self.lenState = 1 
             # -input file.. streme player.
-            command = ['mplayer',
+            command = [BACKEND,
                        #'-vo',
                        #'gl,2,x11,xv',
                        '-zoom',
@@ -946,7 +950,7 @@
         
         if self.state == STARTING_STATE:
             # scrot image.
-            scrot_command = "cd ~/.config/deepin-media-player/buffer/ && mplayer -ss %s -noframedrop -nosound -vo png -frames 1 '%s' >/dev/null 2>&1" % (scrot_pos, self.path)
+            scrot_command = "cd ~/.config/deepin-media-player/buffer/ && "+BACKEND+" -ss %s -noframedrop -nosound -vo png -frames 1 '%s' >/dev/null 2>&1" % (scrot_pos, self.path)
             os.system(scrot_command)
             # modify image name or get image file.
             save_image_path = get_home_path() + "/.config/deepin-media-player/buffer/"        # save image buffer dir.
@@ -974,7 +978,7 @@
             
         if self.state == STARTING_STATE:
             # scrot image.
-            os.system("cd /tmp/buffer/ && mplayer -ss %s -noframedrop -nosound -vo png -frames 1 '%s' >/dev/null 2>&1" % (scrot_pos, self.path))
+            os.system("cd /tmp/buffer/ && "+BACKEND+" -ss %s -noframedrop -nosound -vo png -frames 1 '%s' >/dev/null 2>&1" % (scrot_pos, self.path))
             # modify image name or get image file.
             save_image_path = "/tmp/buffer/"        # save preview image buffer dir.    
             image_list = os.listdir(save_image_path) # get dir all image.
--- src/switch_audio/audio.py.old	2013-04-13 20:49:46.031069517 -0400
+++ src/switch_audio/audio.py	2013-04-13 20:49:26.627312097 -0400
@@ -23,6 +23,11 @@
 import os
 import gobject
 
+if os.path.exists('/usr/bin/mplayer2'):
+        BACKEND = "mplayer2"
+else:
+        BACKEND = "mplayer"
+
 VIDEO_TYPE = 0
 DVD_TYPE = 1
 VCD_TYPE = 2
@@ -43,9 +48,9 @@
         
     def get_video_information(self, video_path, video_type=VIDEO_TYPE):
         if video_type == VIDEO_TYPE:
-            cmd = "mplayer -identify -frames 5 -endpos 0 -vo null '%s'" % (video_path)
+            cmd = BACKEND+" -identify -frames 5 -endpos 0 -vo null '%s'" % (video_path)
         elif video_type == DVD_TYPE:
-            cmd = "mplayer -vo null -ao null -frames 0 -identify "
+            cmd = BACKEND+" -vo null -ao null -frames 0 -identify "
             cmd += "dvd:// -dvd-device '%s'" % (video_path)
 
         pipe = os.popen(str(cmd))
--- src/video_information/gui.py.old	2013-02-15 22:04:24.000000000 -0500
+++ src/video_information/gui.py	2013-04-13 20:51:24.909833362 -0400
@@ -35,7 +35,11 @@
 import gtk
 import gobject
 
-
+if os.path.exists('/usr/bin/mplayer2'):
+	BACKEND = "mplayer2"
+else:
+	BACKEND = "mplayer"
+ 
 VIDEO_TYPE = 0        
 DVD_TYPE = 1
 VCD_TYPE = 2
@@ -98,9 +102,9 @@
 
 def get_video_information(video_path, video_type=VIDEO_TYPE):    
     if video_type == VIDEO_TYPE:
-        cmd = "mplayer -identify -frames 5 -endpos 0 -vo null  '%s'" % (video_path)
+        cmd = BACKEND+" -identify -frames 5 -endpos 0 -vo null  '%s'" % (video_path)
     elif video_type == DVD_TYPE:
-        cmd = "mplayer -vo null -ao null -frames 0 -identify "
+        cmd = BACKEND+" -vo null -ao null -frames 0 -identify "
         cmd += "dvd:// -dvd-device '%s'" % (video_path)
 
     pipe = os.popen(str(cmd))
