--- a/chrome/browser/renderer_preferences_util.cc        2025-06-09 23:43:12.302027536 +0700
+++ b/chrome/browser/renderer_preferences_util.cc        2025-06-09 23:43:38.084028322 +0700
@@ -183,7 +183,8 @@
 #if defined(USE_AURA) && BUILDFLAG(IS_LINUX)
   auto* linux_ui_theme = ui::LinuxUiTheme::GetForProfile(profile);
   if (linux_ui_theme) {
-    if (ThemeServiceFactory::GetForProfile(profile)->UsingSystemTheme()) {
+    if (ThemeServiceFactory::GetForProfile(profile) &&
+        ThemeServiceFactory::GetForProfile(profile)->UsingSystemTheme()) {
       linux_ui_theme->GetFocusRingColor(&prefs->focus_ring_color);
       linux_ui_theme->GetActiveSelectionBgColor(
           &prefs->active_selection_bg_color);
