--- a/third_party/blink/renderer/platform/fonts/font_face_creation_params.h
+++ b/third_party/blink/renderer/platform/fonts/font_face_creation_params.h
@@ -64,7 +64,7 @@
 #endif
   }
 
-  FontFaceCreationParams(const std::string& filename,
+  FontFaceCreationParams(const WTF::String& filename,
                          int fontconfig_interface_id,
                          int ttc_index = 0)
       : creation_type_(kCreateFontByFciIdAndTtcIndex),
@@ -77,7 +77,7 @@
     DCHECK_EQ(creation_type_, kCreateFontByFamily);
     return family_;
   }
-  const std::string& Filename() const {
+  const WTF::String& Filename() const {
     DCHECK_EQ(creation_type_, kCreateFontByFciIdAndTtcIndex);
 #if defined(ADDRESS_SANITIZER)
     DCHECK(filename_.has_value());
@@ -105,7 +105,7 @@
       std::tuple<int, int, unsigned> hash_data = {
           ttc_index_, fontconfig_interface_id_,
           HasFilename() ? StringHasher::HashMemory(
-                              Filename().data(),
+                              Filename().Utf8().data(),
                               static_cast<unsigned>(Filename().length()))
                         : 0};
       return StringHasher::HashMemory(&hash_data, sizeof(hash_data));
@@ -130,7 +130,7 @@
   FontFaceCreationType creation_type_;
   AtomicString family_;
 
-  void SetFilename(std::string& filename) {
+  void SetFilename(WTF::String& filename) {
 #if defined(ADDRESS_SANITIZER)
     *filename_ = filename;
 #else
@@ -167,7 +167,7 @@
   // See crbug.com/346174906.
   std::optional<std::string> filename_;
 #else
-  std::string filename_;
+  WTF::String filename_;
 #endif
   int fontconfig_interface_id_ = 0;
   int ttc_index_ = 0;
--- a/third_party/blink/renderer/platform/fonts/linux/font_cache_linux.cc
+++ b/third_party/blink/renderer/platform/fonts/linux/font_cache_linux.cc
@@ -111,7 +111,7 @@
 
   FontFaceCreationParams creation_params;
   creation_params = FontFaceCreationParams(
-      fallback_font.filepath.value(), fallback_font.fontconfig_interface_id,
+      WTF::String(fallback_font.filepath.value()), fallback_font.fontconfig_interface_id,
       fallback_font.ttc_index);
 
   // Changes weight and/or italic of given FontDescription depends on
--- a/third_party/blink/renderer/platform/fonts/skia/font_cache_skia.cc
+++ b/third_party/blink/renderer/platform/fonts/skia/font_cache_skia.cc
@@ -234,7 +234,7 @@
           creation_params.FontconfigInterfaceId(), creation_params.TtcIndex());
     }
     return SkTypeface_Factory::FromFilenameAndTtcIndex(
-        creation_params.Filename().data(), creation_params.TtcIndex());
+        creation_params.Filename().Utf8().data(), creation_params.TtcIndex());
   }
 #endif
 
