Index: gnome-panel-2.26.3/applets/clock/calendar-window.c
===================================================================
--- gnome-panel-2.26.3.orig/applets/clock/calendar-window.c
+++ gnome-panel-2.26.3/applets/clock/calendar-window.c
@@ -50,6 +50,9 @@
 #ifdef HAVE_LIBECAL
 #include "calendar-client.h"
 #endif
+#ifdef HAVE_LIBLUNAR_GTK
+#include <lunar-gtk/lunar.h>
+#endif
 
 #define CALENDAR_WINDOW_GET_PRIVATE(o) (G_TYPE_INSTANCE_GET_PRIVATE ((o), CALENDAR_TYPE_WINDOW, CalendarWindowPrivate))
 
@@ -1455,13 +1458,19 @@
 	GtkWidget                 *calendar;
 	GtkCalendarDisplayOptions  options;
         struct tm                 *tm;
-
+#ifdef HAVE_LIBLUNAR_GTK
+	calendar = lunar_calendar_new ();
+#else
 	calendar = gtk_calendar_new ();
+#endif
 	options = gtk_calendar_get_display_options (GTK_CALENDAR (calendar));
 	if (calwin->priv->show_weeks)
 		options |= GTK_CALENDAR_SHOW_WEEK_NUMBERS;
 	else
 		options &= ~(GTK_CALENDAR_SHOW_WEEK_NUMBERS);
+#ifdef HAVE_LIBLUNAR_GTK
+		options |= GTK_CALENDAR_SHOW_DETAILS;
+#endif
 	gtk_calendar_set_display_options (GTK_CALENDAR (calendar), options);
 
 	if (calwin->priv->utc_time)
Index: gnome-panel-2.26.3/configure.in
===================================================================
--- gnome-panel-2.26.3.orig/configure.in
+++ gnome-panel-2.26.3/configure.in
@@ -124,6 +124,30 @@ PKG_CHECK_MODULES(CLOCK, pango >= $PANGO
 AC_SUBST(CLOCK_CFLAGS)
 AC_SUBST(CLOCK_LIBS)
 
+dnl *************
+dnl liblunar-gtk
+dnl *************
+AC_ARG_ENABLE(liblunar,
+        AS_HELP_STRING([--enable-liblunar],[Enable chinese lunar calendar for clock applet.]),
+        enable_lunar=$enableval, enable_lunar=auto)
+
+if test "x$enable_lunar" = "xauto"; then
+         PKG_CHECK_EXISTS([liblunar-gtk],enable_lunar=yes,enable_lunar=no)
+fi
+
+if test "x$enable_lunar" = "xyes";then
+        LIBLUNAR_GTK_REQUIRED=2.2.1
+        PKG_CHECK_MODULES(LUNAR_GTK, [liblunar-gtk >= $LIBLUNAR_GTK_REQUIRED])
+        AC_SUBST(LUNAR_GTK_CFLAGS)
+        AC_SUBST(LUNAR_GTK_LIBS)
+        AC_DEFINE([HAVE_LIBLUNAR_GTK], 1, [Defined when liblunar-gtk are detected])
+        have_liblunar_gtk=yes
+else
+        have_liblunar_gtk=no
+fi
+AM_CONDITIONAL(CLOCK_LUNAR, test "x$have_liblunar_gtk" = "xyes")
+
+
 dnl Hmm, we already checked for these above.  Is there a way to avoid checking twice?
 PKG_CHECK_MODULES(CLOCK_MECHANISM, [ gthread-2.0 gio-2.0 dbus-glib-1 ])
 AC_SUBST(CLOCK_MECHANISM_CFLAGS)
Index: gnome-panel-2.26.3/applets/clock/Makefile.am
===================================================================
--- gnome-panel-2.26.3.orig/applets/clock/Makefile.am
+++ gnome-panel-2.26.3/applets/clock/Makefile.am
@@ -77,6 +77,11 @@ BUILT_SOURCES =					\
 	clock-typebuiltins.c			\
 	clock-typebuiltins.h
 
+if CLOCK_LUNAR
+INCLUDES += $(LUNAR_GTK_CFLAGS)
+CLOCK_LDADD += $(LUNAR_GTK_LIBS)
+endif
+
 clock-marshallers.c: clock-marshallers.list
 	$(QUIET_GEN)glib-genmarshal --prefix _clock_marshal --header --body --internal $< > $@
 
