From a7f5c2b455180f14be2a68ac39720b5959dca73b Mon Sep 17 00:00:00 2001
From: liuyujielol <2073201758GD@gmail.com>
Date: Fri, 7 Jun 2024 11:41:14 +0800
Subject: [PATCH 2/2] nostart-stop-gc

---
 crates/zed/build.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/crates/zed/build.rs b/crates/zed/build.rs
index 4a8f2c81f..e9641cb7c 100644
--- a/crates/zed/build.rs
+++ b/crates/zed/build.rs
@@ -23,6 +23,7 @@ fn main() {
         println!("cargo:rustc-link-arg=-Wl,-ObjC");
     }
 
+    println!("cargo:rustc-link-arg=-Wl,-z,nostart-stop-gc");
     // Populate git sha environment variable if git is available
     println!("cargo:rerun-if-changed=../../.git/logs/HEAD");
     if let Ok(output) = Command::new("git").args(["rev-parse", "HEAD"]).output() {
-- 
2.45.2

