diff --git a/crates/sqlez/Cargo.toml b/crates/sqlez/Cargo.toml
index 6f25ba9..dae8c33 100644
--- a/crates/sqlez/Cargo.toml
+++ b/crates/sqlez/Cargo.toml
@@ -13,7 +13,7 @@ anyhow.workspace = true
 collections.workspace = true
 futures.workspace = true
 indoc.workspace = true
-libsqlite3-sys = { version = "0.28", features = ["bundled"] }
+libsqlite3-sys = { version = "0.28" }
 parking_lot.workspace = true
 smol.workspace = true
 thread_local = "1.1.4"
