--- a/ble.pp
+++ b/ble.pp
@@ -76,7 +76,7 @@
 # check --help or --version
 
 {
-  #%[commit_hash = system("git show -s --format=%h")]
+  #%[commit_hash = "1a5c451c8baa71439a6be4ea0f92750de35a7620"]
   #%[ble_version = getenv("FULLVER") + "+" + commit_hash]
   #%expand
   ##%if commit_hash != ""
@@ -1729,7 +1729,7 @@
 }
 
 #%$ pwd=$(pwd) q=\' Q="'\''" bash -c 'echo "_ble_base_repository=$q${pwd//$q/$Q}$q"'
-#%$ echo "_ble_base_branch=$(git rev-parse --abbrev-ref HEAD)"
+#%$ echo "_ble_base_branch=\"v0.4.0-devel3\""
 _ble_base_repository_url=https://github.com/akinomyoga/ble.sh
 #%$ echo "_ble_base_build_git_version=\"$BUILD_GIT_VERSION\""
 #%$ echo "_ble_base_build_make_version=\"$BUILD_MAKE_VERSION\""
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -43,10 +43,10 @@
 
 outfiles+=$(OUTDIR)/ble.sh
 -include $(OUTDIR)/ble.dep
-$(OUTDIR)/ble.sh: ble.pp GNUmakefile | .git $(OUTDIR)
+$(OUTDIR)/ble.sh: ble.pp GNUmakefile | $(OUTDIR)
 	DEPENDENCIES_PHONY=1 DEPENDENCIES_OUTPUT="$(@:%.sh=%.dep)" DEPENDENCIES_TARGET="$@" \
 	  FULLVER=$(FULLVER) \
-	  BUILD_GIT_VERSION="$(shell LANG=C git --version)" \
+	  BUILD_GIT_VERSION="2.39.0" \
 	  BUILD_MAKE_VERSION="$(shell LANG=C $(MAKE) --version | head -1)" \
 	  BUILD_GAWK_VERSION="$(GAWK_VERSION)" \
 	  $(MWGPP) $< >/dev/null
@@ -139,7 +139,6 @@
 
 .PHONY: update-contrib
 update-contrib contrib/contrib.mk:
-	git submodule update --init --recursive
 
 include contrib/contrib.mk
 
--- a/make_command.sh
+++ b/make_command.sh
@@ -79,7 +79,7 @@
 #\
 # Source: /ble.pp'
       local file
-      for file in $(git ls-files src); do
+      for file in $(ls src); do
         [[ $file == *.sh ]] || continue
         script=$script"\\
 # Source: /$file"
