--- a/GNUmakefile
+++ b/GNUmakefile
@@ -28,6 +28,13 @@
 
 MWGPP:=$(GAWK) -f make/mwg_pp.awk
 
+# Note (#D2058): we had used "cp -p xxx out/xxx" to copy files to the build
+# directory, but some filesystem (ecryptfs) has a bug that the subsecond
+# timestamps are truncated causing an issue: make every time copies all the
+# files into the subdirectory `out`.  We give up using `cp -p` and instead copy
+# the file with `cp` with the timestamps being the copy time.
+CP := cp
+
 #------------------------------------------------------------------------------
 # ble.sh
 
@@ -99,10 +114,11 @@
 outfiles += $(OUTDIR)/lib/test-syntax.sh
 outfiles += $(OUTDIR)/lib/test-complete.sh
 outfiles += $(OUTDIR)/lib/util.bgproc.sh
+
 $(OUTDIR)/lib/%.sh: lib/%.sh | $(OUTDIR)/lib
-	cp -p $< $@
+	$(CP) $< $@
 $(OUTDIR)/lib/%.txt: lib/%.txt | $(OUTDIR)/lib
-	cp -p $< $@
+	$(CP) $< $@
 $(OUTDIR)/lib/core-syntax.sh: lib/core-syntax.sh lib/core-syntax-ctx.def | $(OUTDIR)/lib
 	$(MWGPP) $< > $@
 $(OUTDIR)/lib/init-msys1.sh: lib/init-msys1.sh lib/init-msys1-helper.c | $(OUTDIR)/lib
@@ -119,20 +135,45 @@
 #	$(MWGPP) $< > $@
 
 #------------------------------------------------------------------------------
-# documents
+# licenses and documents
+
+outdirs += $(OUTDIR)/licenses $(OUTDIR)/doc
+outfiles-license += $(OUTDIR)/licenses/LICENSE.md
+ifneq ($(USE_DOC),no)
+  outfiles-doc += $(OUTDIR)/doc/README.md
+  outfiles-doc += $(OUTDIR)/doc/README-ja_JP.md
+  outfiles-doc += $(OUTDIR)/doc/CONTRIBUTING.md
+  outfiles-doc += $(OUTDIR)/doc/ChangeLog.md
+  outfiles-doc += $(OUTDIR)/doc/Release.md
+endif
 
-outdirs += $(OUTDIR)/doc
-outfiles-doc += $(OUTDIR)/doc/README.md
-outfiles-doc += $(OUTDIR)/doc/README-ja_JP.md
-outfiles-doc += $(OUTDIR)/doc/LICENSE.md
-outfiles-doc += $(OUTDIR)/doc/CONTRIBUTING.md
-outfiles-doc += $(OUTDIR)/doc/ChangeLog.md
-outfiles-doc += $(OUTDIR)/doc/Release.md
-$(OUTDIR)/doc/%: % | $(OUTDIR)/doc
-	cp -p $< $@
+# Workaround for make-3.81 (#D2065)
+#
+# We want to do something like the following:
+#
+#   $(OUTDIR)/license/%.md: %.md | $(OUTDIR)/license
+#   	$(CP) $< $@
+#   $(OUTDIR)/doc/%.md: %.md | $(OUTDIR)/doc
+#   	$(CP) $< $@
+#
+# However, because of a bug in make-3.81, this rule overrides all the other
+# more detailed patterns such as $(OUTDIR)/doc/contrib/%.md.  As a result, even
+# when we want to apply preprocessing to specific file patterns under
+# $(OUTDIR)/doc/%, $(CP) is always is used to install the files.  To work
+# around this problem in make-3.81, we need to manually filter the target files
+# whose source files are at the top level in the source tree.
+#
+outfiles-doc-toplevel := \
+  $(filter $(outfiles-doc),$(patsubst %,$(OUTDIR)/doc/%,$(wildcard *.md)))
+$(outfiles-doc-toplevel): $(OUTDIR)/doc/%.md: %.md | $(OUTDIR)/doc
+	$(CP) $< $@
+outfiles-license-toplevel := \
+  $(filter $(outfiles-license),$(patsubst %,$(OUTDIR)/licenses/%,$(wildcard *.md)))
+$(outfiles-license-toplevel): $(OUTDIR)/licenses/%.md: %.md | $(OUTDIR)/licenses
+	$(CP) $< $@
 
 $(OUTDIR)/doc/%: docs/% | $(OUTDIR)/doc
-	cp -p $< $@
+	$(CP) $< $@
 
 #------------------------------------------------------------------------------
 # contrib
@@ -149,7 +190,7 @@
 $(outdirs):
 	mkdir -p $@
 
-build: contrib/contrib.mk $(outfiles) $(outfiles-doc)
+build: contrib/contrib.mk $(outfiles) $(outfiles-doc) $(outfiles-license)
 .PHONY: build
 
 all: build
@@ -157,25 +198,27 @@
 #------------------------------------------------------------------------------
 # target "install"
 
+# Users can specify make variables INSDIR, INSDIR_LICENSE, and INSDIR_DOC to
+# control the install locations.  Instead of INSDIR, users may specify DESTDIR
+# and/or PREFIX to automatically set up these variables.
+
 ifneq ($(INSDIR),)
-  ifeq ($(INSDIR_DOC),)
-    INSDIR_DOC := $(INSDIR)/doc
-  endif
+  INSDIR_LICENSE := $(INSDIR)/licenses
+  INSDIR_DOC     := $(INSDIR)/doc
 else
-  ifneq ($(filter-out %/,$(DESTDIR)),)
-    DESTDIR := $(DESTDIR)/
-  endif
-
-  ifneq ($(DESTDIR)$(PREFIX),)
-    DATA_HOME := $(DESTDIR)$(PREFIX)/share
+  ifneq ($(DESTDIR),)
+    DATADIR := $(abspath $(DESTDIR)/$(PREFIX)/share)
+  else ifneq ($(PREFIX),)
+    DATADIR := $(abspath $(PREFIX)/share)
   else ifneq ($(XDG_DATA_HOME),)
-    DATA_HOME := $(XDG_DATA_HOME)
+    DATADIR := $(abspath $(XDG_DATA_HOME))
   else
-    DATA_HOME := $(HOME)/.local/share
+    DATADIR := $(abspath $(HOME)/.local/share)
   endif
 
-  INSDIR = $(DATA_HOME)/blesh
-  INSDIR_DOC = $(DATA_HOME)/doc/blesh
+  INSDIR         := $(DATADIR)/blesh
+  INSDIR_LICENSE := $(DATADIR)/blesh/licenses
+  INSDIR_DOC     := $(DATADIR)/doc/blesh
 endif
 
 ifneq ($(strip_comment),)
@@ -184,23 +227,32 @@
   opt_strip_comment :=
 endif
 
-install: \
-  $(outfiles:$(OUTDIR)/%=$(INSDIR)/%) \
-  $(outfiles-doc:$(OUTDIR)/doc/%=$(INSDIR_DOC)/%) \
+insfiles         := $(outfiles:$(OUTDIR)/%=$(INSDIR)/%)
+insfiles-license := $(outfiles-license:$(OUTDIR)/licenses/%=$(INSDIR_LICENSE)/%)
+insfiles-doc     := $(outfiles-doc:$(OUTDIR)/doc/%=$(INSDIR_DOC)/%)
+
+install-files := \
+  $(insfiles) $(insfiles-license) $(insfiles-doc) \
   $(INSDIR)/cache.d $(INSDIR)/run
-$(INSDIR)/%: $(OUTDIR)/%
+install: $(install-files)
+uninstall:
+	bash make_command.sh uninstall $(install-files)
+.PHONY: install uninstall
+
+$(insfiles): $(INSDIR)/%: $(OUTDIR)/%
 	bash make_command.sh install $(opt_strip_comment) "$<" "$@"
-$(INSDIR_DOC)/%: $(OUTDIR)/doc/%
+$(insfiles-license): $(INSDIR_LICENSE)/%: $(OUTDIR)/licenses/%
+	bash make_command.sh install "$<" "$@"
+$(insfiles-doc): $(INSDIR_DOC)/%: $(OUTDIR)/doc/%
 	bash make_command.sh install "$<" "$@"
 $(INSDIR)/cache.d $(INSDIR)/run:
 	mkdir -p $@ && chmod a+rwxt $@
-.PHONY: install
 
 clean:
-	-rm -rf $(outfiles) $(OUTDIR)/ble.dep
+	-rm -rf $(outfiles) $(outfiles-doc) $(outfiles-license) $(OUTDIR)/ble.dep
 .PHONY: clean
 
-dist: $(outfiles)
+dist: $(outfiles) $(outfiles-doc) $(outfiles-license)
 	FULLVER=$(FULLVER) bash make_command.sh dist $^
 .PHONY: dist
 
--- a/contrib/contrib.mk
+++ b/contrib/contrib.mk
@@ -3,13 +3,17 @@
 contrib-subdirs := airline config integration
 contrib-outdirs := $(OUTDIR)/contrib $(contrib-subdirs:%=$(OUTDIR)/contrib/%)
 contrib-srcfiles := $(wildcard contrib/*.bash $(contrib-subdirs:%=contrib/%/*.bash))
-contrib-outfiles := $(contrib-srcfiles:contrib/%=$(OUTDIR)/contrib/%)
+contrib-datfiles := $(wildcard contrib/*.dat $(contrib-subdirs:%=contrib/%/*.dat))
+contrib-docfiles := $(wildcard contrib/*.md $(contrib-subdirs:%=contrib/%/*.md))
+contrib-outfiles := $(contrib-srcfiles:contrib/%=$(OUTDIR)/contrib/%) $(contrib-datfiles:contrib/%=$(OUTDIR)/contrib/%)
 
 # files
 outdirs += $(contrib-outdirs)
 outfiles += $(contrib-outfiles)
 $(OUTDIR)/contrib/%.bash: contrib/%.bash | $(contrib-outdirs)
-	cp -p $< $@
+	$(CP) $< $@
+$(OUTDIR)/contrib/%.dat: contrib/%.dat | $(contrib-outdirs)
+	$(CP) $< $@
 
 define LinkOldIngteration
 outfiles += $$(OUTDIR)/contrib/$1.bash
@@ -22,16 +26,18 @@
 $(eval $(call LinkOldIngteration,fzf-initialize))
 $(eval $(call LinkOldIngteration,fzf-key-bindings))
 
+# licenses
+
+outdirs += $(OUTDIR)/licenses/contrib
+outfiles-license += $(OUTDIR)/licenses/contrib/LICENSE
+$(OUTDIR)/licenses/contrib/LICENSE: contrib/LICENSE | $(OUTDIR)/licenses/contrib
+	$(CP) $< $@
+
 # docs
-outdirs += $(OUTDIR)/doc/contrib
-outfiles-doc += $(OUTDIR)/doc/contrib/LICENSE
-outfiles-doc += $(OUTDIR)/doc/contrib/README-ja.md
-outfiles-doc += $(OUTDIR)/doc/contrib/README.md
-
-# Note (workaround for make-3.81): 当初 $(OUTDIR)/doc/contrib/% に対してルール
-# を記述していたが make-3.81 に於いて正しく適用されない事が分かった。仕方がない
-# ので LICENSE と %.md の二つの規則に分けて書く事にする。
-$(OUTDIR)/doc/contrib/LICENSE: contrib/LICENSE | $(OUTDIR)/doc/contrib
-	cp -p $< $@
-$(OUTDIR)/doc/contrib/%.md: contrib/%.md | $(OUTDIR)/doc/contrib
-	cp -p $< $@
+
+outdirs += $(OUTDIR)/doc/contrib $(OUTDIR)/doc/contrib/integration
+ifneq ($(USE_DOC),no)
+  outfiles-doc += $(contrib-docfiles:contrib/%=$(OUTDIR)/doc/contrib/%)
+endif
+$(OUTDIR)/doc/contrib/%.md: contrib/%.md | $(OUTDIR)/doc/contrib $(OUTDIR)/doc/contrib/integration
+	$(CP) $< $@
