diff --git a/wechat-universal.sh b/wechat-universal.sh
index 186c6b0..8b47ed0 100644
--- a/wechat-universal.sh
+++ b/wechat-universal.sh
@@ -34,6 +34,7 @@ BWRAP_ENV_APPEND=()
 env_add QT_QPA_PLATFORM xcb
 env_add QT_AUTO_SCREEN_SCALE_FACTOR 1
 env_add PATH "/sandbox:${PATH}"
+env_add LD_LIBRARY_PATH "$LD_LIBRARY_PATH:/opt/wechat-universal/RadiumWMPF/runtime"
 
 case "${XMODIFIERS}" in 
     *@im=fcitx*)
@@ -80,11 +81,11 @@ BWRAP_ARGS=(
 
     # /usr
     --ro-bind /usr{,}
-    --symlink usr/lib /lib
-    --symlink usr/lib /lib64
-    --symlink usr/bin /bin
-    --symlink usr/bin /sbin
-    --bind /usr/bin/{true,lsblk}
+    --ro-bind /lib{,}
+    --ro-bind /lib64{,}
+    --ro-bind /bin{,}
+    --ro-bind /sbin{,}
+    --ro-bind /usr/bin/{true,lsblk}
 
     # /sandbox
     --ro-bind /{usr/lib/flatpak-xdg-utils,sandbox}/xdg-open
@@ -99,12 +100,15 @@ BWRAP_ARGS=(
     --proc /proc
 
     # /etc
-    --ro-bind /etc/machine-id{,}
+    --ro-bind-try /etc/machine-id{,}
     --ro-bind /etc/passwd{,}
     --ro-bind /etc/nsswitch.conf{,}
     --ro-bind /etc/resolv.conf{,}
     --ro-bind /etc/localtime{,}
     --ro-bind-try /etc/fonts{,}
+    --ro-bind /etc/ld.so.cache{,}
+    --ro-bind /etc/ld.so.conf{,}
+    --ro-bind-try /etc/ld.so.conf.d{,}
 
     # /sys
     --dir /sys/dev # hack for Intel / AMD graphics, mesa calling virtual nodes needs /sys/dev being 0755
@@ -134,13 +138,13 @@ BWRAP_ARGS=(
     --ro-bind-try "${HOME}/.local/share/.icons"{,}
 
     # /run
-    --dev-bind /run/dbus{,}
-    --ro-bind /run/systemd/userdb{,}
+    --dev-bind-try /run/dbus{,}
+    --ro-bind-try /run/systemd/userdb{,}
     --ro-bind-try "${XAUTHORITY}"{,}
-    --ro-bind "${XDG_RUNTIME_DIR}/bus"{,}
-    --ro-bind "${XDG_RUNTIME_DIR}/pulse"{,}
+    --ro-bind-try "${XDG_RUNTIME_DIR}/bus"{,}
+    --ro-bind-try "${XDG_RUNTIME_DIR}/pulse"{,}
 )
 
 exec bwrap "${BWRAP_ARGS[@]}" "${BWRAP_CUSTOM_BINDS[@]}" "${BWRAP_DEV_BINDS[@]}" "${BWRAP_ENV_APPEND[@]}" /opt/wechat-universal/wechat "$@"
