diff -Nur gcin-1.4.5.pre2.orig/configure gcin-1.4.5.pre2/configure
--- gcin-1.4.5.pre2.orig/configure	2009-02-01 15:00:40.000000000 +0800
+++ gcin-1.4.5.pre2/configure	2009-04-13 13:35:40.477001801 +0800
@@ -138,7 +138,7 @@
 
 
 QT4_IM='N'
-QT4_IM_DIR=/usr/lib/qt4/plugins/lib/inputmethods
+QT4_IM_DIR=/usr/lib/qt4/plugins/inputmethods
 echo "testing if you have qt4 support"
 if [ -d  $QT4_IM_DIR ]; then
   QT4_IM='Y'
