diff -Nur gcin-1.6.4.pre23.orig/qt4-im/Makefile gcin-1.6.4.pre23/qt4-im/Makefile
--- gcin-1.6.4.pre23.orig/qt4-im/Makefile	2011-06-08 21:03:50.000000000 +0800
+++ gcin-1.6.4.pre23/qt4-im/Makefile	2011-06-14 10:28:40.060082794 +0800
@@ -15,7 +15,7 @@
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 
 moc_gcin-qt.cpp:	gcin-qt.h
-	/usr/$(LIB)/$(QT)/bin/moc $< -o moc_gcin-qt.cpp
+	moc $< -o moc_gcin-qt.cpp
 
 im-gcin.so: $(OBJS)
 	export LD_RUN_PATH=$(gcinlibdir) ;\
