diff -Nur gcin-1.4.6.pre16.orig/qt4-im/Makefile gcin-1.4.6.pre16/qt4-im/Makefile
--- gcin-1.4.6.pre16.orig/qt4-im/Makefile	2009-12-10 07:09:17.000000000 +0800
+++ gcin-1.4.6.pre16/qt4-im/Makefile	2009-12-16 10:07:01.259943371 +0800
@@ -15,7 +15,7 @@
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 
 moc_gcin-qt.cpp:	gcin-qt.h
-	/usr/lib/$(QT)/bin/moc $< -o moc_gcin-qt.cpp
+	/usr/bin/moc $< -o moc_gcin-qt.cpp
 
 im-gcin.so: $(OBJS)
 	export LD_RUN_PATH=$(gcinlibdir) ;\
