diff -Nur gcin-2.6.6.orig//configure gcin-2.6.6/configure
--- gcin-2.6.6.orig//configure	2011-12-20 21:48:03.000000000 +0800
+++ gcin-2.6.6/configure	2011-12-26 18:14:38.000000000 +0800
@@ -245,7 +245,7 @@
   mv gcin.spec gcin.spec.tmp
   grep -v qt4/plugins/inputmethods gcin.spec.tmp > gcin.spec
 else
-  for i in /usr/share/qt4/bin/moc /usr/lib/qt4/bin/moc /usr/lib64/qt4/bin/moc
+  for i in /usr/share/qt4/bin/moc /usr/lib/qt4/bin/moc /usr/lib64/qt4/bin/moc /usr/bin/moc
   do
 	[ -x $i ] && QT4_MOC=$i
   done
