--- pho.c	2006-10-07 12:07:47.000000000 +0800
+++ pho.c.new	2006-10-19 22:44:51.000000000 +0800
@@ -188,7 +188,6 @@
   }
 
   int max_in_idx;
-
   for(max_in_idx=3; max_in_idx>=0 && !typ_pho[max_in_idx]; max_in_idx--);
 
   // try insert mode first
@@ -216,13 +215,13 @@
 
       if (num) {
         inph[typ] = newkey;
+        typ_pho[typ] = num;
         insert = typ;
         break;
       }
     }
   }
 
-
 //  dbg("newkey %c\n", newkey);
 
   int mcount = 0;
