diff -Nur gcin-1.4.3.pre5.orig/configure gcin-1.4.3.pre5/configure
--- gcin-1.4.3.pre5.orig/configure	2008-08-03 17:51:13.000000000 +0800
+++ gcin-1.4.3.pre5/configure	2008-08-07 14:16:47.529004234 +0800
@@ -126,9 +126,9 @@
 
 
 QT4_IM='N'
-QT4_IM_DIR=/usr/lib/qt4/plugins/lib/inputmethods
+QT4_IM_DIR=/usr/lib/qt4/plugins/inputmethods
 echo "testing if you have qt4 support"
-if [ -d  $QT_IM_DIR ]; then
+if [ -d  $QT4_IM_DIR ]; then
   QT4_IM='Y'
 fi
 
diff -Nur gcin-1.4.3.pre5.orig/qt4-im/.depend gcin-1.4.3.pre5/qt4-im/.depend
--- gcin-1.4.3.pre5.orig/qt4-im/.depend	2008-08-04 18:48:30.000000000 +0800
+++ gcin-1.4.3.pre5/qt4-im/.depend	1970-01-01 08:00:00.000000000 +0800
@@ -1,206 +0,0 @@
-gcin-imcontext-qt.o: gcin-imcontext-qt.cpp gcin-imcontext-qt.h \
-  /usr/lib/qt4/include/QtGui/QApplication \
-  /usr/lib/qt4/include/QtGui/qapplication.h \
-  /usr/lib/qt4/include/QtCore/qcoreapplication.h \
-  /usr/lib/qt4/include/QtCore/qobject.h \
-  /usr/lib/qt4/include/QtCore/qobjectdefs.h \
-  /usr/lib/qt4/include/QtCore/qnamespace.h \
-  /usr/lib/qt4/include/QtCore/qglobal.h \
-  /usr/lib/qt4/include/QtCore/qconfig.h \
-  /usr/lib/qt4/include/QtCore/qfeatures.h \
-  /usr/lib/qt4/include/QtCore/qstring.h \
-  /usr/lib/qt4/include/QtCore/qchar.h \
-  /usr/lib/qt4/include/QtCore/qbytearray.h \
-  /usr/lib/qt4/include/QtCore/qatomic.h \
-  /usr/lib/qt4/include/QtCore/qatomic_arch.h \
-  /usr/lib/qt4/include/QtCore/qatomic_i386.h \
-  /usr/lib/qt4/include/QtCore/qlist.h \
-  /usr/lib/qt4/include/QtCore/qiterator.h \
-  /usr/lib/qt4/include/QtCore/qalgorithms.h \
-  /usr/lib/qt4/include/QtCore/qcoreevent.h \
-  /usr/lib/qt4/include/QtCore/qeventloop.h \
-  /usr/lib/qt4/include/QtGui/qwindowdefs.h \
-  /usr/lib/qt4/include/QtCore/qpoint.h \
-  /usr/lib/qt4/include/QtCore/qsize.h \
-  /usr/lib/qt4/include/QtGui/qcursor.h /usr/lib/qt4/include/QtCore/QEvent \
-  /usr/lib/qt4/include/QtCore/qcoreevent.h \
-  /usr/lib/qt4/include/QtGui/QFont /usr/lib/qt4/include/QtGui/qfont.h \
-  /usr/lib/qt4/include/QtGui/QInputContext \
-  /usr/lib/qt4/include/QtGui/qinputcontext.h \
-  /usr/lib/qt4/include/QtGui/qevent.h \
-  /usr/lib/qt4/include/QtGui/qregion.h \
-  /usr/lib/qt4/include/QtCore/qrect.h \
-  /usr/lib/qt4/include/QtGui/qkeysequence.h \
-  /usr/lib/qt4/include/QtGui/qmime.h \
-  /usr/lib/qt4/include/QtCore/qmimedata.h \
-  /usr/lib/qt4/include/QtCore/qvariant.h \
-  /usr/lib/qt4/include/QtCore/qmetatype.h \
-  /usr/lib/qt4/include/QtCore/qdatastream.h \
-  /usr/lib/qt4/include/QtCore/qiodevice.h \
-  /usr/lib/qt4/include/QtCore/qmap.h /usr/lib/qt4/include/QtGui/qdrag.h \
-  /usr/lib/qt4/include/QtGui/qaction.h \
-  /usr/lib/qt4/include/QtGui/qwidget.h \
-  /usr/lib/qt4/include/QtGui/qpaintdevice.h \
-  /usr/lib/qt4/include/QtGui/qpalette.h \
-  /usr/lib/qt4/include/QtGui/qcolor.h /usr/lib/qt4/include/QtGui/qrgb.h \
-  /usr/lib/qt4/include/QtCore/qstringlist.h \
-  /usr/lib/qt4/include/QtCore/qregexp.h \
-  /usr/lib/qt4/include/QtCore/qstringmatcher.h \
-  /usr/lib/qt4/include/QtGui/qbrush.h /usr/lib/qt4/include/QtCore/qpair.h \
-  /usr/lib/qt4/include/QtCore/qvector.h \
-  /usr/lib/qt4/include/QtGui/qmatrix.h \
-  /usr/lib/qt4/include/QtGui/qpolygon.h \
-  /usr/lib/qt4/include/QtCore/qline.h \
-  /usr/lib/qt4/include/QtGui/qtransform.h \
-  /usr/lib/qt4/include/QtGui/qpainterpath.h \
-  /usr/lib/qt4/include/QtGui/qimage.h \
-  /usr/lib/qt4/include/QtGui/qpixmap.h /usr/lib/qt4/include/QtGui/qfont.h \
-  /usr/lib/qt4/include/QtGui/qfontmetrics.h \
-  /usr/lib/qt4/include/QtGui/qfontinfo.h \
-  /usr/lib/qt4/include/QtGui/qsizepolicy.h \
-  /usr/lib/qt4/include/QtGui/qicon.h \
-  /usr/lib/qt4/include/QtGui/qactiongroup.h \
-  /usr/lib/qt4/include/QtGui/QInputMethodEvent \
-  /usr/lib/qt4/include/QtGui/qevent.h /usr/lib/qt4/include/QtCore/QObject \
-  /usr/lib/qt4/include/QtCore/qobject.h \
-  /usr/lib/qt4/include/QtCore/QPoint /usr/lib/qt4/include/QtCore/qpoint.h \
-  /usr/lib/qt4/include/QtGui/QWidget /usr/lib/qt4/include/QtGui/qwidget.h \
-  /usr/lib/qt4/include/QtGui/QX11Info \
-  /usr/lib/qt4/include/QtGui/qx11info_x11.h ../im-client/gcin-im-client.h
-gcin-qt.o: gcin-qt.cpp gcin-qt.h /usr/lib/qt4/include/QtCore/QObject \
-  /usr/lib/qt4/include/QtCore/qobject.h \
-  /usr/lib/qt4/include/QtCore/qobjectdefs.h \
-  /usr/lib/qt4/include/QtCore/qnamespace.h \
-  /usr/lib/qt4/include/QtCore/qglobal.h \
-  /usr/lib/qt4/include/QtCore/qconfig.h \
-  /usr/lib/qt4/include/QtCore/qfeatures.h \
-  /usr/lib/qt4/include/QtCore/qstring.h \
-  /usr/lib/qt4/include/QtCore/qchar.h \
-  /usr/lib/qt4/include/QtCore/qbytearray.h \
-  /usr/lib/qt4/include/QtCore/qatomic.h \
-  /usr/lib/qt4/include/QtCore/qatomic_arch.h \
-  /usr/lib/qt4/include/QtCore/qatomic_i386.h \
-  /usr/lib/qt4/include/QtCore/qlist.h \
-  /usr/lib/qt4/include/QtCore/qiterator.h \
-  /usr/lib/qt4/include/QtCore/qalgorithms.h \
-  /usr/lib/qt4/include/QtCore/QSocketNotifier \
-  /usr/lib/qt4/include/QtCore/qsocketnotifier.h \
-  /usr/lib/qt4/include/QtCore/qobject.h gcin-common-qt.h \
-  /usr/lib/qt4/include/QtCore/QString \
-  /usr/lib/qt4/include/QtCore/qstring.h \
-  /usr/lib/qt4/include/QtCore/QStringList \
-  /usr/lib/qt4/include/QtCore/qstringlist.h \
-  /usr/lib/qt4/include/QtCore/qdatastream.h \
-  /usr/lib/qt4/include/QtCore/qiodevice.h \
-  /usr/lib/qt4/include/QtCore/qregexp.h \
-  /usr/lib/qt4/include/QtCore/qstringmatcher.h
-im-gcin-qt.o: im-gcin-qt.cpp /usr/lib/qt4/include/QtCore/Qt \
-  /usr/lib/qt4/include/QtCore/qnamespace.h \
-  /usr/lib/qt4/include/QtCore/qglobal.h \
-  /usr/lib/qt4/include/QtCore/qconfig.h \
-  /usr/lib/qt4/include/QtCore/qfeatures.h \
-  /usr/lib/qt4/include/QtGui/QInputContextPlugin \
-  /usr/lib/qt4/include/QtGui/qinputcontextplugin.h \
-  /usr/lib/qt4/include/QtCore/qplugin.h \
-  /usr/lib/qt4/include/QtCore/qobject.h \
-  /usr/lib/qt4/include/QtCore/qobjectdefs.h \
-  /usr/lib/qt4/include/QtCore/qnamespace.h \
-  /usr/lib/qt4/include/QtCore/qstring.h \
-  /usr/lib/qt4/include/QtCore/qchar.h \
-  /usr/lib/qt4/include/QtCore/qbytearray.h \
-  /usr/lib/qt4/include/QtCore/qatomic.h \
-  /usr/lib/qt4/include/QtCore/qatomic_arch.h \
-  /usr/lib/qt4/include/QtCore/qatomic_i386.h \
-  /usr/lib/qt4/include/QtCore/qlist.h \
-  /usr/lib/qt4/include/QtCore/qiterator.h \
-  /usr/lib/qt4/include/QtCore/qalgorithms.h \
-  /usr/lib/qt4/include/QtCore/qpointer.h \
-  /usr/lib/qt4/include/QtCore/qfactoryinterface.h \
-  /usr/lib/qt4/include/QtCore/qstringlist.h \
-  /usr/lib/qt4/include/QtCore/qdatastream.h \
-  /usr/lib/qt4/include/QtCore/qiodevice.h \
-  /usr/lib/qt4/include/QtCore/qregexp.h \
-  /usr/lib/qt4/include/QtCore/qstringmatcher.h gcin-qt.h \
-  /usr/lib/qt4/include/QtCore/QObject \
-  /usr/lib/qt4/include/QtCore/qobject.h \
-  /usr/lib/qt4/include/QtCore/QSocketNotifier \
-  /usr/lib/qt4/include/QtCore/qsocketnotifier.h gcin-common-qt.h \
-  /usr/lib/qt4/include/QtCore/QString \
-  /usr/lib/qt4/include/QtCore/qstring.h \
-  /usr/lib/qt4/include/QtCore/QStringList \
-  /usr/lib/qt4/include/QtCore/qstringlist.h gcin-imcontext-qt.h \
-  /usr/lib/qt4/include/QtGui/QApplication \
-  /usr/lib/qt4/include/QtGui/qapplication.h \
-  /usr/lib/qt4/include/QtCore/qcoreapplication.h \
-  /usr/lib/qt4/include/QtCore/qcoreevent.h \
-  /usr/lib/qt4/include/QtCore/qeventloop.h \
-  /usr/lib/qt4/include/QtGui/qwindowdefs.h \
-  /usr/lib/qt4/include/QtCore/qpoint.h \
-  /usr/lib/qt4/include/QtCore/qsize.h \
-  /usr/lib/qt4/include/QtGui/qcursor.h /usr/lib/qt4/include/QtCore/QEvent \
-  /usr/lib/qt4/include/QtCore/qcoreevent.h \
-  /usr/lib/qt4/include/QtGui/QFont /usr/lib/qt4/include/QtGui/qfont.h \
-  /usr/lib/qt4/include/QtGui/QInputContext \
-  /usr/lib/qt4/include/QtGui/qinputcontext.h \
-  /usr/lib/qt4/include/QtGui/qevent.h \
-  /usr/lib/qt4/include/QtGui/qregion.h \
-  /usr/lib/qt4/include/QtCore/qrect.h \
-  /usr/lib/qt4/include/QtGui/qkeysequence.h \
-  /usr/lib/qt4/include/QtGui/qmime.h \
-  /usr/lib/qt4/include/QtCore/qmimedata.h \
-  /usr/lib/qt4/include/QtCore/qvariant.h \
-  /usr/lib/qt4/include/QtCore/qmetatype.h \
-  /usr/lib/qt4/include/QtCore/qmap.h /usr/lib/qt4/include/QtGui/qdrag.h \
-  /usr/lib/qt4/include/QtGui/qaction.h \
-  /usr/lib/qt4/include/QtGui/qwidget.h \
-  /usr/lib/qt4/include/QtGui/qpaintdevice.h \
-  /usr/lib/qt4/include/QtGui/qpalette.h \
-  /usr/lib/qt4/include/QtGui/qcolor.h /usr/lib/qt4/include/QtGui/qrgb.h \
-  /usr/lib/qt4/include/QtGui/qbrush.h /usr/lib/qt4/include/QtCore/qpair.h \
-  /usr/lib/qt4/include/QtCore/qvector.h \
-  /usr/lib/qt4/include/QtGui/qmatrix.h \
-  /usr/lib/qt4/include/QtGui/qpolygon.h \
-  /usr/lib/qt4/include/QtCore/qline.h \
-  /usr/lib/qt4/include/QtGui/qtransform.h \
-  /usr/lib/qt4/include/QtGui/qpainterpath.h \
-  /usr/lib/qt4/include/QtGui/qimage.h \
-  /usr/lib/qt4/include/QtGui/qpixmap.h /usr/lib/qt4/include/QtGui/qfont.h \
-  /usr/lib/qt4/include/QtGui/qfontmetrics.h \
-  /usr/lib/qt4/include/QtGui/qfontinfo.h \
-  /usr/lib/qt4/include/QtGui/qsizepolicy.h \
-  /usr/lib/qt4/include/QtGui/qicon.h \
-  /usr/lib/qt4/include/QtGui/qactiongroup.h \
-  /usr/lib/qt4/include/QtGui/QInputMethodEvent \
-  /usr/lib/qt4/include/QtGui/qevent.h /usr/lib/qt4/include/QtCore/QPoint \
-  /usr/lib/qt4/include/QtCore/qpoint.h /usr/lib/qt4/include/QtGui/QWidget \
-  /usr/lib/qt4/include/QtGui/qwidget.h \
-  /usr/lib/qt4/include/QtGui/QX11Info \
-  /usr/lib/qt4/include/QtGui/qx11info_x11.h ../im-client/gcin-im-client.h
-moc_gcin-qt.o: moc_gcin-qt.cpp gcin-qt.h \
-  /usr/lib/qt4/include/QtCore/QObject \
-  /usr/lib/qt4/include/QtCore/qobject.h \
-  /usr/lib/qt4/include/QtCore/qobjectdefs.h \
-  /usr/lib/qt4/include/QtCore/qnamespace.h \
-  /usr/lib/qt4/include/QtCore/qglobal.h \
-  /usr/lib/qt4/include/QtCore/qconfig.h \
-  /usr/lib/qt4/include/QtCore/qfeatures.h \
-  /usr/lib/qt4/include/QtCore/qstring.h \
-  /usr/lib/qt4/include/QtCore/qchar.h \
-  /usr/lib/qt4/include/QtCore/qbytearray.h \
-  /usr/lib/qt4/include/QtCore/qatomic.h \
-  /usr/lib/qt4/include/QtCore/qatomic_arch.h \
-  /usr/lib/qt4/include/QtCore/qatomic_i386.h \
-  /usr/lib/qt4/include/QtCore/qlist.h \
-  /usr/lib/qt4/include/QtCore/qiterator.h \
-  /usr/lib/qt4/include/QtCore/qalgorithms.h \
-  /usr/lib/qt4/include/QtCore/QSocketNotifier \
-  /usr/lib/qt4/include/QtCore/qsocketnotifier.h \
-  /usr/lib/qt4/include/QtCore/qobject.h gcin-common-qt.h \
-  /usr/lib/qt4/include/QtCore/QString \
-  /usr/lib/qt4/include/QtCore/qstring.h \
-  /usr/lib/qt4/include/QtCore/QStringList \
-  /usr/lib/qt4/include/QtCore/qstringlist.h \
-  /usr/lib/qt4/include/QtCore/qdatastream.h \
-  /usr/lib/qt4/include/QtCore/qiodevice.h \
-  /usr/lib/qt4/include/QtCore/qregexp.h \
-  /usr/lib/qt4/include/QtCore/qstringmatcher.h
diff -Nur gcin-1.4.3.pre5.orig/qt4-im/Makefile gcin-1.4.3.pre5/qt4-im/Makefile
--- gcin-1.4.3.pre5.orig/qt4-im/Makefile	2008-08-04 23:55:57.000000000 +0800
+++ gcin-1.4.3.pre5/qt4-im/Makefile	2008-08-07 14:16:47.529004234 +0800
@@ -7,14 +7,14 @@
 else
 	IMMODULES_LOCAL=/usr/lib/$(QTIM)
 endif
-QTINC=/usr/lib/$(QT)/include
+QTINC=/usr/include/qt4
 
 INCS=-I../im-client -I$(QTINC) -I/usr/include/X11 -I$(QTINC)/QtGui -I$(QTINC)/QtCore
 CXXFLAGS=$(OPTFLAGS) $(INCS) -Wall -D_REENTRANT -fPIC  -DQT4 -DQT_SHARED \
 -DQT_IMMODULE -DPIC
 OBJS= moc_gcin-qt.o gcin-qt.o im-gcin-qt.o gcin-imcontext-qt.o
 .SUFFIXES:	.c .cpp .a .so .E .h
-LDFLAGS=-L/usr/lib/$(QT)/$(LIB) -L/usr/X11R6/$(LIB) -L../im-client -lQtGui -lQtCore -lmng -ljpeg -lpng -lz \
+LDFLAGS=-L/usr/lib/$(QT) -L/usr/X11R6/$(LIB) -L../im-client -lQtGui -lQtCore -lmng -ljpeg -lpng -lz \
 -lXrender -lXrandr  -lXft -lfontconfig -lgcin-im-client \
 -lXext -lX11 -lm -lSM -lICE -ldl -lpthread
 all:    im-gcin.so
@@ -23,7 +23,7 @@
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 
 moc_gcin-qt.cpp:	gcin-qt.h
-	/usr/lib/$(QT)/bin/moc $< -o moc_gcin-qt.cpp
+	/usr/bin/moc $< -o moc_gcin-qt.cpp
 
 im-gcin.so: $(OBJS)
 	export LD_RUN_PATH=$(gcinlibdir) ;\
