--- configure	2007-01-19 09:36:13.000000000 +0800
+++ configure.new	2007-01-26 19:09:00.000000000 +0800
@@ -104,13 +104,14 @@
 fi
 
 QT_IM='N'
-for i in qt3 qt4
+for i in 3 4
 do
-  QT_IM_DIR=/usr/lib/$i/plugins/$LIB/inputmethods
+  QT_IM_DIR=/usr/qt/$i/plugins/inputmethods
   echo "testing if you have QT_IM_MODULE dir:$QT_IM_DIR support"
   if [ -d  $QT_IM_DIR ]; then
     QT_IM='Y'
-    QT=$i
+    QT=qt$i
+    QT_DIR=/usr/qt/$i
     break
   fi
 done
@@ -165,7 +166,7 @@
 echo "datadir=\$(DESTDIR)$datadir" >> config.mak
 echo "datadir_r=$datadir" >> config.mak
 echo "libdir=\$(DESTDIR)$libdir" >> config.mak
-echo "gcinlibdir=\$(DESTDIR)\$(libdir)/gcin" >> config.mak
+echo "gcinlibdir=\$(DESTDIR)$libdir/gcin" >> config.mak
 echo "includedir=\$(DESTDIR)$includedir" >> config.mak
 
 echo "LDFLAGS=$GTKLDFLAGS" >> config.mak
@@ -191,6 +192,8 @@
 echo "LIB=$LIB" >> config.mak
 echo "MAC_OS=$MAC_OS" >> config.mak
 echo "SO_FLAGS=$SO_FLAGS" >> config.mak
+echo "QT_DIR=$QT_DIR" >>config.mak
+echo "QT_DIR_i=\$(DESTDIR)/$QT_DIR" >>config.mak
 echo "USE_TRAY=$use_tray" >> config.mak
 
 
--- qt-im/Makefile	2006-10-14 13:15:36.000000000 +0800
+++ qt-im/Makefile.new	2007-01-26 19:05:19.000000000 +0800
@@ -1,16 +1,16 @@
 include ../config.mak
-QTIM=$(QT)/plugins/inputmethods
-IMMODULES=$(libdir)/$(QTIM)
+QTIM=$(QT_DIR_i)/plugins/inputmethods
+IMMODULES=$(QTIM)
 ifeq ($(LIB), lib64)
-	IMMODULES_LOCAL=/usr/lib/$(QT)/plugins/lib64/inputmethods
+	IMMODULES_LOCAL=$(QT_DIR_i)/plugins/lib64/inputmethods
 else
-	IMMODULES_LOCAL=/usr/lib/$(QTIM)
+	IMMODULES_LOCAL=$(QTIM)
 endif
-INCS=-I../im-client -I/usr/lib/$(QT)/include  -I/usr/include/Xft2/X11/Xft  -I/usr/include/fontconfig -I/usr/include/freetype2
+INCS=-I../im-client -I$(QT_DIR)/include  -I/usr/include/Xft2/X11/Xft  -I/usr/include/fontconfig -I/usr/include/freetype2
 CXXFLAGS=$(OPTFLAGS) $(INCS) -Wall -D_REENTRANT -fPIC  -DQT_THREAD_SUPPORT -DQT_PLUGIN -DQT_SHARED -DQT_NO_DEBUG
 OBJS=qgcininputcontextplugin.o qgcininputcontextplugin_my.o qgcininputcontext.o qgcininputcontext_x11.o
 .SUFFIXES:	.c .cpp .a .so .E .h
-LDFLAGS=-L/usr/lib/$(QT)/$(LIB) -L/usr/X11R6/$(LIB) -L../im-client -lqt-mt -lmng -ljpeg -lpng -lz \
+LDFLAGS=-L$(QT_DIR)/$(LIB) -L/usr/X11R6/$(LIB) -L../im-client -lqt-mt -lmng -ljpeg -lpng -lz \
 -lXrender -lXrandr  -lXft -lfontconfig -lgcin-im-client \
 -lXext -lX11 -lm -lSM -lICE -ldl -lpthread
 all:	libqgcin.so
@@ -18,7 +18,7 @@
 .cpp.E:
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 .h.o:
-	/usr/lib/$(QT)/bin/moc $< -o $<_moc.cpp
+	$(QT_DIR)/bin/moc $< -o $<_moc.cpp
 	$(CXX) -c -pipe $(CXXFLAGS) $(INCS) $<_moc.cpp -o $@
 
 libqgcin.so: $(OBJS)
