--- installer.sh.orig	2008-03-31 13:05:29.000000000 +0800
+++ installer.sh	2008-03-31 13:06:15.000000000 +0800
@@ -35,14 +35,14 @@
 rm "$pkgname"
 
 # Now ask the user where he wants to install our package
-echo -e "Please specify where you want to install $pkgdispname [$definstdir]: \c"
-read instdir
-if [ -z "$instdir" ]; then
-  instdir="$definstdir"
-fi
+#echo -e "Please specify where you want to install $pkgdispname [$definstdir]: \c"
+instdir="${1}"
+#if [ -z "$instdir" ]; then
+#  instdir="$definstdir"
+#fi
 
 # create the installation directory first
-mkdir "$instdir"
+#mkdir "$instdir"
 if [ $? != 0 ]; then
   echo "Installation failed."; exit $?;
 fi
