From cc0b0cf4cceb29e1adddf9a171c9409010bee3c9 Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Wed, 9 Feb 2022 14:20:22 +0800
Subject: [PATCH] x86: fix __arch_exchange_and_add_body

---
 sysdeps/x86/atomic-machine.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sysdeps/x86/atomic-machine.h b/sysdeps/x86/atomic-machine.h
index f24f1c7..c431f86 100644
--- a/sysdeps/x86/atomic-machine.h
+++ b/sysdeps/x86/atomic-machine.h
@@ -165,7 +165,7 @@
      else if (__HAVE_64B_ATOMICS)					      \
        __asm __volatile ("xchgq %q0, %1"				      \
 			 : "=r" (result), "=m" (*mem)			      \
-			 : "0" ((int64_t) cast_to_integer (newvalue)),        \
+			 : "0" (cast_to_integer (newvalue)),        \
 			   "m" (*mem));					      \
      else								      \
        {								      \
@@ -196,7 +196,7 @@
      else if (__HAVE_64B_ATOMICS)					      \
        __asm __volatile (lock "xaddq %q0, %1"				      \
 			 : "=r" (__result), "=m" (*mem)			      \
-			 : "0" ((int64_t) cast_to_integer (__addval)),     \
+			 : "0" (cast_to_integer (__addval)),     \
 			   "m" (*mem),					      \
 			   "i" (offsetof (tcbhead_t, multiple_threads)));     \
      else								      \
-- 
2.34.1

