From 954bbeb3ab0dea87041e08220ea2b1b34301149a Mon Sep 17 00:00:00 2001
From: Fangrui Song <maskray@google.com>
Date: Thu, 7 Oct 2021 18:16:04 -0700
Subject: [PATCH 08/19] Disable __USE_EXTERN_INLINES for clang

---
 include/features.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/features.h b/include/features.h
index d974eabfaf..11cbbf512a 100644
--- a/include/features.h
+++ b/include/features.h
@@ -500,7 +500,7 @@
 /* Decide whether we can define 'extern inline' functions in headers.  */
 #if __GNUC_PREREQ (2, 7) && defined __OPTIMIZE__ \
     && !defined __OPTIMIZE_SIZE__ && !defined __NO_INLINE__ \
-    && defined __extern_inline
+    && defined __extern_inline && !defined __clang__
 # define __USE_EXTERN_INLINES	1
 #endif
 
-- 
2.32.0

