From 365740bf8dc417f279a36672f05627cbd5dc431c Mon Sep 17 00:00:00 2001
From: Fangrui Song <maskray@google.com>
Date: Wed, 3 Nov 2021 22:14:22 -0700
Subject: [PATCH 11/19] nptl/pthread_create.c hack: Work around aliasing an asm
 label

---
 nptl/pthread_create.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/nptl/pthread_create.c b/nptl/pthread_create.c
index bad9eeb52f..6c42edf133 100644
--- a/nptl/pthread_create.c
+++ b/nptl/pthread_create.c
@@ -45,13 +45,13 @@
 
 /* Globally enabled events.  */
 td_thr_events_t __nptl_threads_events;
-libc_hidden_proto (__nptl_threads_events)
-libc_hidden_data_def (__nptl_threads_events)
+// libc_hidden_proto (__nptl_threads_events)
+// libc_hidden_data_def (__nptl_threads_events)
 
 /* Pointer to descriptor with the last event.  */
 struct pthread *__nptl_last_event;
-libc_hidden_proto (__nptl_last_event)
-libc_hidden_data_def (__nptl_last_event)
+// libc_hidden_proto (__nptl_last_event)
+// libc_hidden_data_def (__nptl_last_event)
 
 #ifdef SHARED
 /* This variable is used to access _rtld_global from libthread_db.  If
-- 
2.32.0

