From 2b34350565c11ccc49936a018f3cf3f9f7f95eb3 Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Sat, 4 Dec 2021 01:38:44 +0800
Subject: [PATCH 12/19] fix clang doesn'n inline it

---
 sysdeps/nptl/gai_misc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysdeps/nptl/gai_misc.h b/sysdeps/nptl/gai_misc.h
index 261e24dae6..277d9142ce 100644
--- a/sysdeps/nptl/gai_misc.h
+++ b/sysdeps/nptl/gai_misc.h
@@ -86,7 +86,7 @@ __gai_start_notify_thread (void)
   assert_perror (sigerr);
 }
 
-extern inline int
+extern inline int __attribute__ ((always_inline))
 __gai_create_helper_thread (pthread_t *threadp, void *(*tf) (void *),
 			    void *arg)
 {
-- 
2.32.0

