From cf1441bedf652b586e0d5781dbffb2c8fa007a81 Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Thu, 2 Dec 2021 18:56:47 +0800
Subject: [PATCH 14/19] fix clang find cxx headers

---
 configure | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index bac533649b..8d513a68ef 100755
--- a/configure
+++ b/configure
@@ -5559,8 +5559,20 @@ if test -n "$CXX"; then
     echo "#include <$1>" | $CXX -M -MP -x c++ - 2>/dev/null \
 	 | sed -n "\,$1:,{s/:\$//;p}"
   }
-  CXX_CSTDLIB_HEADER="$(find_cxx_header cstdlib)"
-  CXX_CMATH_HEADER="$(find_cxx_header cmath)"
+  clang_find_cxx_header () {
+    echo "#include <$1>" | $CXX -M -MP -x c++ - 2>/dev/null \
+	 | awk '/^-.o/ {print $2}'
+  }
+  case $CC in
+  *clang*)
+    CXX_CSTDLIB_HEADER="$(clang_find_cxx_header cstdlib)"
+    CXX_CMATH_HEADER="$(clang_find_cxx_header cmath)"
+    ;;
+  *gcc*)
+    CXX_CSTDLIB_HEADER="$(find_cxx_header cstdlib)"
+    CXX_CMATH_HEADER="$(find_cxx_header cmath)"
+    ;;
+  esac
   CXX_BITS_STD_ABS_H="$(find_cxx_header bits/std_abs.h)"
 fi
 
-- 
2.32.0

