From d0c8222321cb317624a15227bc44147681fb22db Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Thu, 2 Dec 2021 18:58:52 +0800
Subject: [PATCH 15/19] work around compiler rt for glibc

---
 Makeconfig | 27 ++++++++++++++++++---------
 1 file changed, 18 insertions(+), 9 deletions(-)

diff --git a/Makeconfig b/Makeconfig
index 3fa2f13003..ee9f5f8763 100644
--- a/Makeconfig
+++ b/Makeconfig
@@ -661,35 +661,44 @@ else
 endif
 libgcc_eh := -Wl,--as-needed -lgcc_s $(libunwind) -Wl,--no-as-needed
 gnulib-arch =
-gnulib = -lgcc $(gnulib-arch)
+#gnulib = -lgcc $(gnulib-arch)
+arch = $(shell $(CC) -dumpmachine | awk -F '-' '{print $$1}')
+gnulib = `$(CC) -print-libgcc-file-name` $(gnulib-arch)
 gnulib-tests := -lgcc $(libgcc_eh)
 static-gnulib-arch =
 # By default, elf/static-stubs.o, instead of -lgcc_eh, is used to
 # statically link programs.  When --disable-shared is used, we use
 # -lgcc_eh since elf/static-stubs.o isn't sufficient.
 ifeq (yes,$(build-shared))
-static-gnulib = -lgcc $(static-gnulib-arch)
+static-gnulib = `$(CC) -print-libgcc-file-name` $(static-gnulib-arch)
 else
 static-gnulib = -lgcc -lgcc_eh $(static-gnulib-arch)
 endif
 static-gnulib-tests := -lgcc -lgcc_eh $(libunwind)
-libc.so-gnulib := -lgcc
+#libc.so-gnulib := -lgcc
+libc.so-gnulib := `$(CC) -print-libgcc-file-name`
 endif
 +preinit = $(addprefix $(csu-objpfx),crti.o)
 +postinit = $(addprefix $(csu-objpfx),crtn.o)
-+prector = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtbegin.o`
-+postctor = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtend.o`
+#+prector = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtbegin.o`
++prector = `$(CC) $(sysdep-LDFLAGS) --print-file-name=clang_rt.crtbegin-$(arch).o`
+#+postctor = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtend.o`
++postctor = `$(CC) $(sysdep-LDFLAGS) -print-file-name=clang_rt.crtend-$(arch).o`
 # Variants of the two previous definitions for linking PIE programs.
-+prectorS = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtbeginS.o`
-+postctorS = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtendS.o`
+#+prectorS = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtbeginS.o`
+#+postctorS = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtendS.o`
++prectorS = `$(CC) $(sysdep-LDFLAGS) --print-file-name=clang_rt.crtbegin-$(arch).o`
++postctorS = `$(CC) $(sysdep-LDFLAGS) --print-file-name=clang_rt.crtend-$(arch).o`
 # Variants of the two previous definitions for statically linking programs.
 ifeq (yes,$(enable-static-pie))
 # Static PIE must use PIE variants.
 +prectorT = $(+prectorS)
 +postctorT = $(+postctorS)
 else
-+prectorT = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtbeginT.o`
-+postctorT = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtend.o`
+#+prectorT = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtbeginT.o`
++prectorT = `$(CC) $(sysdep-LDFLAGS) --print-file-name=clang_rt.crtbegin-$(arch).o`
+#+postctorT = `$(CC) $(sysdep-LDFLAGS) --print-file-name=crtend.o`
++postctorT = `$(CC) $(sysdep-LDFLAGS) -print-file-name=clang_rt.crtend-$(arch).o`
 endif
 csu-objpfx = $(common-objpfx)csu/
 elf-objpfx = $(common-objpfx)elf/
-- 
2.32.0

