From f10257247a5e095a5211ee42bf971906e91d682d Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Thu, 2 Dec 2021 19:00:25 +0800
Subject: [PATCH 17/19] work around libc++ for glibc test program

---
 support/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/support/Makefile b/support/Makefile
index 984ec02dfe..ae362064d2 100644
--- a/support/Makefile
+++ b/support/Makefile
@@ -238,7 +238,8 @@ ifeq (,$(CXX))
 LINKS_DSO_PROGRAM = links-dso-program-c
 else
 LINKS_DSO_PROGRAM = links-dso-program
-LDLIBS-links-dso-program = -lstdc++ -lgcc -lgcc_s $(libunwind)
+#LDLIBS-links-dso-program = -lstdc++ -lgcc -lgcc_s $(libunwind)
+LDLIBS-links-dso-program = -lc++ $(libunwind)
 endif
 
 ifeq (yes,$(have-selinux))
-- 
2.32.0

