From 2589dc1ab56b4c0534e18e6b294f82a8a253ee4e Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Sat, 4 Dec 2021 15:14:39 +0800
Subject: [PATCH 19/19] hardcode ld.lld as linker to build glibc

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 8d513a68ef..5c9471402d 100755
--- a/configure
+++ b/configure
@@ -4554,7 +4554,7 @@ case "$CC" in
     *)             LDNAME=ld;;
 esac
 AS=`$CC -print-prog-name=as`
-LD=`$CC -print-prog-name=$LDNAME`
+LD=`$CC -print-prog-name=ld.lld`
 AR=`$CC -print-prog-name=ar`
 
 OBJDUMP=`$CC -print-prog-name=objdump`
-- 
2.32.0

