diff --git a/Makerules b/Makerules
index 8db0ca1296..7a12c4e8ae 100644
--- a/Makerules
+++ b/Makerules
@@ -1164,8 +1164,8 @@ $(inst_includedir)/%.h: $(common-objpfx)%.h $(+force)
 	$(do-install)
 $(inst_includedir)/%.h: %.h $(+force)
 	$(do-install)
-$(inst_includedir)/%.h: $(..)include/%.h $(+force)
-	$(do-install)
+#$(inst_includedir)/%.h: $(..)include/%.h $(+force)
+#	$(do-install)
 headers-nonh := $(filter-out %.h,$(headers))
 ifdef headers-nonh
 $(addprefix $(inst_includedir)/,$(headers-nonh)): $(inst_includedir)/%: \
