From 244458e9754f62ba1ebca12d54655578294ed03a Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Thu, 10 Feb 2022 14:38:49 +0800
Subject: [PATCH] i386/e_atanh: asseble tfloat

---
 sysdeps/i386/fpu/e_atanh.S | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sysdeps/i386/fpu/e_atanh.S b/sysdeps/i386/fpu/e_atanh.S
index 6e4fef0..dcb9cbf 100644
--- a/sysdeps/i386/fpu/e_atanh.S
+++ b/sysdeps/i386/fpu/e_atanh.S
@@ -33,7 +33,11 @@ one:	.double 1.0
 limit:	.double 0.29
 	ASM_SIZE_DIRECTIVE(limit)
 	.type ln2_2,@object
+#ifdef __clang__
+ln2_2:	.byte 0xac, 0x79, 0xcf, 0xd1, 0xf7, 0x17, 0x72, 0xb1, 0xfd, 0x3f
+#else
 ln2_2:	.tfloat 0.3465735902799726547086160
+#endif
 	ASM_SIZE_DIRECTIVE(ln2_2)
 
 DEFINE_DBL_MIN
-- 
2.34.1

