From c6f1b2f86f14a3d9f3aa91878089e27547764715 Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Thu, 10 Feb 2022 12:03:18 +0800
Subject: [PATCH] i386/s_asinhl: asseble tfloat

---
 sysdeps/i386/fpu/s_asinhl.S | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sysdeps/i386/fpu/s_asinhl.S b/sysdeps/i386/fpu/s_asinhl.S
index bd442c6..9fd2170 100644
--- a/sysdeps/i386/fpu/s_asinhl.S
+++ b/sysdeps/i386/fpu/s_asinhl.S
@@ -23,7 +23,11 @@
 
 	.align ALIGNARG(4)
 	.type huge,@object
+#ifdef __clang__
+huge:	.byte 0x1d, 0xfa, 0x6f, 0x45, 0xe7, 0x34, 0xb6, 0x89, 0xf8, 0x7f
+#else
 huge:	.tfloat 1e+4930
+#endif
 	ASM_SIZE_DIRECTIVE(huge)
 	.align ALIGNARG(4)
 	/* Please note that we use double value for 1.0.  This number
-- 
2.34.1

