From 7b2aedc27576acbbce56d4107833b5c156856656 Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Thu, 10 Feb 2022 14:31:27 +0800
Subject: [PATCH] i386/s_log1pl: asseble tfloat

---
 sysdeps/i386/fpu/s_log1pl.S | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/sysdeps/i386/fpu/s_log1pl.S b/sysdeps/i386/fpu/s_log1pl.S
index f28349f..8414cb7 100644
--- a/sysdeps/i386/fpu/s_log1pl.S
+++ b/sysdeps/i386/fpu/s_log1pl.S
@@ -14,7 +14,11 @@ RCSID("$NetBSD: s_log1p.S,v 1.7 1995/05/09 00:10:58 jtc Exp $")
 		-1 + sqrt(2) / 2 <= x <= 1 - sqrt(2) / 2
 	   0.29 is a safe value.
 	*/
+#ifdef __clang__
+limit: .byte 0xe1, 0x7a, 0x14, 0xae, 0x47, 0xe1, 0x7a, 0x94, 0xfd, 0x3f
+#else
 limit:	.tfloat 0.29
+#endif
 	/* Please note:	 we use a double value here.  Since 1.0 has
 	   an exact representation this does not effect the accuracy
 	   but it helps to optimize the code.  */
-- 
2.34.1

