From ec4e5e2cd4bab4f1f85bbb6d2108a2fab35a5df6 Mon Sep 17 00:00:00 2001
From: lujianhua <lujianhua000@gmail.com>
Date: Thu, 10 Feb 2022 18:14:10 +0800
Subject: [PATCH] remove binutils dependencies

---
 configure | 12 +++++++-----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/configure b/configure
index 85fddae0..4a5eb715 100755
--- a/configure
+++ b/configure
@@ -4591,9 +4591,9 @@ case "$CC" in
     *fuse-ld=lld*) LDNAME=ld.lld;;
     *)             LDNAME=ld;;
 esac
-AS=`$CC -print-prog-name=as`
+AS=$CC
 LD=`$CC -print-prog-name=$LDNAME`
 AR=`$CC -print-prog-name=ar`
 
 OBJDUMP=`$CC -print-prog-name=objdump`
 
@@ -4610,7 +4610,7 @@ if ${libc_cv_prog_as_gnu+:} false; then :
 else
   # Most GNU programs take a -v and spit out some text including
 # the word 'GNU'.  Some try to read stdin, so give them /dev/null.
-if $AS -o conftest -v </dev/null 2>&1 | grep GNU > /dev/null 2>&1; then
+if $AS -o conftest -v </dev/null 2>&1 | grep gnu > /dev/null 2>&1; then
   libc_cv_prog_as_gnu=yes
 else
   libc_cv_prog_as_gnu=no
@@ -4690,10 +4690,12 @@ else
   # Found it, now check the version.
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking version of $AS" >&5
 $as_echo_n "checking version of $AS... " >&6; }
-  ac_prog_version=`$AS --version 2>&1 | sed -n 's/^.*GNU assembler.* \([0-9]*\.[0-9.]*\).*$/\1/p'`
+  #ac_prog_version=`$AS --version 2>&1 | sed -n 's/^.*GNU assembler.* \([0-9]*\.[0-9.]*\).*$/\1/p'`
+  ac_prog_version=`$AS --version 2>&1 | sed -n 's/^.*clang version.* \([0-9]*\.[0-9].[0-9]\)/\1/p'`
   case $ac_prog_version in
     '') ac_prog_version="v. ?.??, bad"; ac_verc_fail=yes;;
-    2.1[0-9][0-9]*|2.2[5-9]*|2.[3-9][0-9]*|[3-9].*|[1-9][0-9]*)
+    #2.1[0-9][0-9]*|2.2[5-9]*|2.[3-9][0-9]*|[3-9].*|[1-9][0-9]*)
+    [1-9][0-9].[0-9].[0-9])
        ac_prog_version="$ac_prog_version, ok"; ac_verc_fail=no;;
     *) ac_prog_version="$ac_prog_version, bad"; ac_verc_fail=yes;;
 
-- 
2.34.1

