diff -uNrp fontconfig-2.2.3.orig/fontconfig/fontconfig.h fontconfig-2.2.3/fontconfig/fontconfig.h
--- fontconfig-2.2.3.orig/fontconfig/fontconfig.h	2004-07-01 02:45:31.000000000 +0800
+++ fontconfig-2.2.3/fontconfig/fontconfig.h	2004-11-27 23:50:31.990682424 +0800
@@ -70,6 +70,7 @@ typedef int		FcBool;
 #define FC_FOUNDRY	    "foundry"		/* String */
 #define FC_ANTIALIAS	    "antialias"		/* Bool (depends) */
 #define FC_HINTING	    "hinting"		/* Bool (true) */
+#define FC_HINT_STYLE	    "hintstyle"		/* Int */
 #define FC_VERTICAL_LAYOUT  "verticallayout"	/* Bool (false) */
 #define FC_AUTOHINT	    "autohint"		/* Bool (false) */
 #define FC_GLOBAL_ADVANCE   "globaladvance"	/* Bool (true) */
@@ -88,6 +89,14 @@ typedef int		FcBool;
 #define FC_CHARSET	    "charset"		/* CharSet */
 #define FC_LANG		    "lang"		/* String RFC 3066 langs */
 #define FC_FONTVERSION	    "fontversion"	/* Int from 'head' table */
+/* Add by Firefly(firefly@firefly.idv.tw)
+ * 讓使用者決定要不要使用 gamma 調整字型深度
+ */
+#define FC_USE_GAMMA	    "usegamma"		/* Bool (false) */
+/* Add by Firefly(firefly@firefly.idv.tw)
+ * 要不要使用內嵌的點陣字體
+ */
+#define FC_EMBEDDED_BITMAP  "embeddedbitmap"    /* Bool (true) */
 
 #define FC_DIR_CACHE_FILE	    "fonts.cache-"FC_CACHE_VERSION
 #define FC_USER_CACHE_FILE	    ".fonts.cache-"FC_CACHE_VERSION
@@ -138,6 +147,12 @@ typedef int		FcBool;
 #define FC_RGBA_VBGR	    4
 #define FC_RGBA_NONE	    5
 
+/* hinting style */
+#define FC_HINT_NONE        0
+#define FC_HINT_SLIGHT      1
+#define FC_HINT_MEDIUM      2
+#define FC_HINT_FULL        3
+
 typedef enum _FcType {
     FcTypeVoid, 
     FcTypeInteger, 
diff -uNrp fontconfig-2.2.3.orig/src/fcdefault.c fontconfig-2.2.3/src/fcdefault.c
--- fontconfig-2.2.3.orig/src/fcdefault.c	2003-03-13 06:16:25.000000000 +0800
+++ fontconfig-2.2.3/src/fcdefault.c	2004-11-27 23:46:08.997663440 +0800
@@ -33,6 +33,9 @@ static struct {
     { FC_VERTICAL_LAYOUT,   FcFalse	},  /* FC_LOAD_VERTICAL_LAYOUT */
     { FC_AUTOHINT,	    FcFalse	},  /* FC_LOAD_FORCE_AUTOHINT */
     { FC_GLOBAL_ADVANCE,    FcTrue	},  /* !FC_LOAD_IGNORE_GLOBAL_ADVANCE_WIDTH */
+    /* Add by Firefly(firefly@firefly.idv.tw) */
+    { FC_USE_GAMMA,	    FcFalse	},
+    { FC_EMBEDDED_BITMAP,   FcTrue	},
 };
 
 #define NUM_FC_BOOL_DEFAULTS	(sizeof FcBoolDefaults / sizeof FcBoolDefaults[0])
diff -uNrp fontconfig-2.2.3.orig/src/fcname.c fontconfig-2.2.3/src/fcname.c
--- fontconfig-2.2.3.orig/src/fcname.c	2003-03-13 06:16:25.000000000 +0800
+++ fontconfig-2.2.3/src/fcname.c	2004-11-27 23:56:24.286125376 +0800
@@ -42,6 +42,10 @@ static const FcObjectType _FcBaseObjectT
 /*    { FC_CORE,		FcTypeBool, }, */
     { FC_ANTIALIAS,	FcTypeBool, },
     { FC_HINTING,	FcTypeBool, },
+/* Add by Firefly(firefly@firefly.idv.tw) */
+    { FC_HINT_STYLE,    FcTypeInteger, },
+    { FC_USE_GAMMA,	FcTypeBool, },
+    { FC_EMBEDDED_BITMAP,   FcTypeBool, },
     { FC_VERTICAL_LAYOUT,   FcTypeBool, },
     { FC_AUTOHINT,	FcTypeBool, },
     { FC_GLOBAL_ADVANCE,    FcTypeBool, },
@@ -175,6 +179,10 @@ static const FcConstant _FcBaseConstants
     { (FcChar8 *) "vrgb",	    "rgba",	    FC_RGBA_VRGB },
     { (FcChar8 *) "vbgr",	    "rgba",	    FC_RGBA_VBGR },
     { (FcChar8 *) "none",	    "rgba",	    FC_RGBA_NONE },
+    { (FcChar8 *) "hintnone",	    "hintstyle",   FC_HINT_NONE },
+    { (FcChar8 *) "hintslight",	    "hintstyle",   FC_HINT_SLIGHT },
+    { (FcChar8 *) "hintmedium",	    "hintstyle",   FC_HINT_MEDIUM },
+    { (FcChar8 *) "hintfull",	    "hintstyle",   FC_HINT_FULL },
 };
 
 #define NUM_FC_CONSTANTS   (sizeof _FcBaseConstants/sizeof _FcBaseConstants[0])
