--- fontconfig-2.2.3.orig/src/fclang.c	2003-06-10 02:25:04.000000000 +0800
+++ fontconfig-2.2.3/src/fclang.c	2004-11-14 19:52:47.745322040 +0800
@@ -61,6 +61,15 @@ FcFreeTypeLangSet (const FcCharSet  *cha
 	return 0;
     for (i = 0; i < NUM_LANG_CHAR_SET; i++)
     {
+/* Modify by Firefly (firefly@firefly.idv.tw)
+ * 有些 CJK 字型檔並不止有一種字集, 所以不能找到一種 CJK 字集,
+ * 便忽略其他的, 要一併找出來才行.
+ *
+ * some asian fonts have other Han languages.
+ * ( e.g. SimSun.tt[f|c] have both zh-cn & zh-tw )
+ */
+	FcBool IsExclusiveLang = FcFreeTypeIsExclusiveLang (fcLangCharSets[i].lang);
+#if 0
 	/*
 	 * Check for Han charsets to make fonts
 	 * which advertise support for a single language
@@ -72,6 +81,7 @@ FcFreeTypeLangSet (const FcCharSet  *cha
 	{
 	    continue;
 	}
+#endif
 	missing = FcCharSetSubtractCount (&fcLangCharSets[i].charset, charset);
         if (FcDebug() & FC_DBG_SCANV)
 	{
