diff -uNrp fontconfig-2.2.3.orig/src/fclang.c fontconfig-2.2.3/src/fclang.c
--- fontconfig-2.2.3.orig/src/fclang.c	2003-06-10 02:25:04.000000000 +0800
+++ fontconfig-2.2.3/src/fclang.c	2004-11-28 00:03:26.302969064 +0800
@@ -106,6 +106,14 @@ FcFreeTypeLangSet (const FcCharSet  *cha
 	}
 	if (!missing)
 	    FcLangSetBitSet (ls, i);
+	/* Add by Firefly (firefly@firefly.idv.tw)
+	 * 有些較舊的 CJK 字型可能漏掉某些字, 所以漏掉 1% 以下
+	 * 仍視為有效.
+	 *
+	 * some old CJK fonts maybe lost char.
+	 */
+	else if (IsExclusiveLang && missing < FcCharSetCount(&fcLangCharSets[i].charset) * 0.01)
+	    FcLangSetBitSet (ls, i);
     }
 
     if (FcDebug() & FC_DBG_SCANV)
