diff -uNrp fontconfig-2.2.3.orig/src/fccache.c fontconfig-2.2.3/src/fccache.c
--- fontconfig-2.2.3.orig/src/fccache.c	2003-03-23 05:25:34.000000000 +0800
+++ fontconfig-2.2.3/src/fccache.c	2004-11-17 09:53:53.000000000 +0800
@@ -299,7 +299,8 @@ FcCacheFontSetAdd (FcFontSet	    *set,
 		frozen = FcPatternFreeze (font);
 		ret = (frozen != 0);
 		if (ret)
-		   ret = FcFontSetAdd (set, frozen);
+		    /* Modify by Firefly(firefly@firefly.idv.tw) */
+		   ret = FcFontSetAddSort (set, frozen);
 	    }
 	    FcPatternDestroy (font);
 	}
diff -uNrp fontconfig-2.2.3.orig/src/fccfg.c fontconfig-2.2.3/src/fccfg.c
--- fontconfig-2.2.3.orig/src/fccfg.c	2004-07-01 02:20:19.000000000 +0800
+++ fontconfig-2.2.3/src/fccfg.c	2004-11-17 09:54:43.000000000 +0800
@@ -95,6 +95,8 @@ FcConfigCreate (void)
 
     config->rescanTime = time(0);
     config->rescanInterval = 30;    
+    /* Add by Firefly (firefly@firefly.idv.tw) */
+    config->familyOutput = FC_FAMILYOUTPUT_AUTO;
     
     return config;
 
@@ -385,6 +387,38 @@ FcConfigGetFonts (FcConfig	*config,
     return config->fonts[set];
 }
 
+/* Add by Firefly(firefly@firefly.idv.tw)
+*/
+static void
+FcMakeFakeStyle(FcFontSet *s,
+		FcPattern *srcpat,
+		int weight,
+		int slant,
+		FcChar8 *style)
+{
+    FcMatrix matrix;
+    matrix.xx = 1;
+    matrix.xy = 0.3;
+    matrix.yx = 0;
+    matrix.yy = 1;
+    FcPattern *duppat = FcPatternDuplicate(srcpat);
+    if (duppat)
+    {
+	FcPatternDel(duppat, FC_WEIGHT);
+	FcPatternAddInteger(duppat, FC_WEIGHT, weight);
+	FcPatternDel(duppat, FC_SLANT);
+	FcPatternAddInteger(duppat, FC_SLANT, slant);
+	FcPatternDel(duppat, FC_STYLE);
+	FcPatternAddString(duppat, FC_STYLE, style);
+	if (slant > FC_SLANT_ROMAN)
+	{
+	    FcPatternAddMatrix(duppat, FC_MATRIX, &matrix);
+	}
+	if (!FcFontSetAddSort(s, duppat))
+	    FcPatternDestroy(duppat);
+    }
+}
+
 void
 FcConfigSetFonts (FcConfig	*config,
 		  FcFontSet	*fonts,
@@ -392,6 +426,101 @@ FcConfigSetFonts (FcConfig	*config,
 {
     if (config->fonts[set])
 	FcFontSetDestroy (config->fonts[set]);
+/*---------------------------------------------------------*/
+/* Add by Firefly(firefly@firefly.idv.tw) */
+    #define	ST_ITALIC      0
+    #define	ST_BOLD        1
+    #define	ST_BOLD_ITALIC 2
+    FcBool    scalable;
+    FcPattern *pat;
+    FcChar8   *family;
+    int slant, weight;
+    int f;
+
+    for (f = 0; f < fonts->nfont; f++)
+    {
+	pat = fonts->fonts[f];
+	if (FcPatternGetString(pat, FC_FAMILY, 0, &family) == FcResultMatch &&
+	    FcPatternGetInteger(pat, FC_SLANT, 0, &slant) == FcResultMatch &&
+	    FcPatternGetInteger(pat, FC_WEIGHT, 0, &weight) == FcResultMatch &&
+	    FcPatternGetBool(pat, FC_SCALABLE, 0, &scalable) == FcResultMatch &&
+	    slant == FC_SLANT_ROMAN && weight <= FC_WEIGHT_MEDIUM && scalable)
+	{
+	    FcChar8 *cmpName;
+    	    FcBool  styles[] = {FcFalse, FcFalse, FcFalse};
+	    int cmpweight, cmpslant;
+	    int i = f + 1;
+	    while (i < fonts->nfont && FcPatternGetString(fonts->fonts[i], FC_FAMILY, 0, &cmpName) == FcResultMatch && strcmp(family, cmpName) == 0)
+	    {
+		if (FcPatternGetInteger(fonts->fonts[i], FC_WEIGHT, 0, &cmpweight) != FcResultMatch)
+		    cmpweight = 0;
+		if (FcPatternGetInteger(fonts->fonts[i], FC_SLANT, 0, &cmpslant) != FcResultMatch)
+		    cmpslant = 0;
+		if (cmpweight <= FC_WEIGHT_MEDIUM && cmpslant >= FC_SLANT_ITALIC)
+		    styles[ST_ITALIC] = FcTrue;
+		else if (cmpweight > FC_WEIGHT_MEDIUM && cmpslant < FC_SLANT_ITALIC)
+		    styles[ST_BOLD] = FcTrue;
+		else if (cmpweight > FC_WEIGHT_MEDIUM && cmpslant >= FC_SLANT_ITALIC)
+		    styles[ST_BOLD_ITALIC] = FcTrue;
+		i++;
+	    }
+
+	    if (!styles[ST_BOLD_ITALIC])
+		FcMakeFakeStyle(fonts, pat, FC_WEIGHT_BOLD, FC_SLANT_ITALIC, "Bold Italic");
+
+	    if (!styles[ST_ITALIC])
+		FcMakeFakeStyle(fonts, pat, weight, FC_SLANT_ITALIC, "Italic");
+
+	    if (!styles[ST_BOLD])
+		FcMakeFakeStyle(fonts, pat, FC_WEIGHT_BOLD, slant, "Bold");
+	}
+    }
+
+    FcChar8 *lang = getenv("LANG");
+    if (!lang) lang = "en_US";
+    FcBool isenglishenv = (strncasecmp(lang, "en", 2) == 0) ? FcTrue : FcFalse;
+
+    if (config->familyOutput == FC_FAMILYOUTPUT_ENGLISHONLY ||
+	(config->familyOutput == FC_FAMILYOUTPUT_AUTO && isenglishenv))
+    {
+	config->fonts[set] = fonts;
+	return;
+    }
+
+    FcChar8 *origName, *otherName;
+    FcPattern *dupPattern;
+    int totFonts = fonts->nfont;
+    for (f=0 ; f < totFonts ; f++)
+    {
+	if (FcPatternGetString(fonts->fonts[f], FC_FAMILY, 0, &origName) != FcResultMatch)
+	    origName = 0;
+	if (FcPatternGetString(fonts->fonts[f], FC_FAMILY, 1, &otherName) != FcResultMatch)
+	    otherName = 0;
+	if (config->familyOutput == FC_FAMILYOUTPUT_AUTO && origName && otherName)
+	{
+	    dupPattern = FcPatternDuplicate(fonts->fonts[f]);
+	    if (dupPattern)
+	    {
+		FcPatternDel(dupPattern, FC_FAMILY);
+		FcPatternAddString(dupPattern, FC_FAMILY, otherName);
+		FcPatternAddString(dupPattern, FC_FAMILY, origName);
+		FcPatternDestroy(fonts->fonts[f]);
+		fonts->fonts[f] = dupPattern;
+	    }
+	}
+	if (config->familyOutput == FC_FAMILYOUTPUT_ANY && otherName)
+	{
+	    dupPattern = FcPatternDuplicate(fonts->fonts[f]);
+	    if (dupPattern)
+	    {
+		FcPatternDel(dupPattern, FC_FAMILY);
+		FcPatternAddString(dupPattern, FC_FAMILY, otherName);
+		if (!FcFontSetAdd(fonts, dupPattern))
+		    FcPatternDestroy(dupPattern);
+	    }
+	}
+    }
+/*---------------------------------------------------------*/
     config->fonts[set] = fonts;
 }
 
diff -uNrp fontconfig-2.2.3.orig/src/fcdir.c fontconfig-2.2.3/src/fcdir.c
--- fontconfig-2.2.3.orig/src/fcdir.c	2003-03-13 06:15:39.000000000 +0800
+++ fontconfig-2.2.3/src/fcdir.c	2004-11-17 09:53:53.000000000 +0800
@@ -139,7 +139,8 @@ FcFileScan (FcFontSet	    *set,
 	 */
 	if (font)
 	{
-	    if (!FcFontSetAdd (set, font))
+	    /* Modify by Firefly(firefly@firefly.idv.tw) */
+	    if (!FcFontSetAddSort (set, font))
 	    {
 		FcPatternDestroy (font);
 		font = 0;
diff -uNrp fontconfig-2.2.3.orig/src/fcfs.c fontconfig-2.2.3/src/fcfs.c
--- fontconfig-2.2.3.orig/src/fcfs.c	2003-03-05 13:51:27.000000000 +0800
+++ fontconfig-2.2.3/src/fcfs.c	2004-11-17 09:53:53.000000000 +0800
@@ -80,3 +80,111 @@ FcFontSetAdd (FcFontSet *s, FcPattern *f
     s->fonts[s->nfont++] = font;
     return FcTrue;
 }
+
+/* Add by Firefly(firefly@firefly.idv.tw) */
+static
+FcChar8 *
+_FcMakePatternString(FcPattern *p)
+{
+    FcChar8 *family, *foundry = 0;
+    int slant, weight;
+    FcBool scalable;
+    double pixel_size = 0;
+
+    FcChar8 mkstr[256];
+
+    if (FcPatternGetString(p, FC_FAMILY, 0, &family) != FcResultMatch ||
+	FcPatternGetInteger(p, FC_SLANT, 0, &slant) != FcResultMatch ||
+	FcPatternGetInteger(p, FC_WEIGHT, 0, &weight) != FcResultMatch ||
+	FcPatternGetBool(p, FC_SCALABLE, 0, &scalable) != FcResultMatch )
+	return 0;
+
+    if (!scalable &&
+	 FcPatternGetDouble(p, FC_PIXEL_SIZE, 0, &pixel_size) != FcResultMatch)
+	return 0;
+
+    FcPatternGetString(p, FC_FOUNDRY, 0, &foundry);
+
+    sprintf(mkstr, "%s,%s,%3d,%3d,%1d,%g", family, foundry, slant, weight, scalable, pixel_size);
+    return FcStrCopy(mkstr);
+}
+
+FcBool
+FcFontSetAddSort (FcFontSet *s, FcPattern *font)
+{
+    FcPattern	**f;
+    int		sfont;
+
+    if (s->nfont == s->sfont)
+    {
+	sfont = s->sfont + 32;
+	if (s->fonts)
+	    f = (FcPattern **) realloc (s->fonts, sfont * sizeof (FcPattern *));
+	else
+	    f = (FcPattern **) malloc (sfont * sizeof (FcPattern *));
+	if (!f)
+	    return FcFalse;
+	if (s->sfont)
+	    FcMemFree (FC_MEM_FONTPTR, s->sfont * sizeof (FcPattern *));
+	FcMemAlloc (FC_MEM_FONTPTR, sfont * sizeof (FcPattern *));
+	s->sfont = sfont;
+	s->fonts = f;
+    }
+
+    FcChar8 *srcstr = _FcMakePatternString(font);
+
+    if (!srcstr)
+	return FcFalse;
+
+    int top = 0;
+    int bottom = s->nfont -1;
+    int middle;
+    int compValue;
+    FcChar8 *cmpstr;
+    while (top <= bottom)
+    {
+	middle = (bottom + top) >> 1;
+	cmpstr = _FcMakePatternString(s->fonts[middle]);
+	if (!cmpstr)
+	{
+	    break;
+	}
+	compValue = strcmp(srcstr, cmpstr);
+	FcStrFree(cmpstr);
+
+	if (compValue == 0)
+	{
+	    break;
+	}
+
+	if (compValue < 0)
+	    bottom = middle - 1;
+	else
+	    top = middle + 1;
+    }
+
+    FcStrFree(srcstr);
+
+    if (top <= bottom)
+    {
+	FcLangSet *langset1, *langset2;
+	FcPatternGetLangSet(s->fonts[middle], FC_LANG, 0, &langset1);
+	FcPatternGetLangSet(font, FC_LANG, 0, &langset2);
+	if (FcLangSetEqual(langset1, langset2))
+	{
+	    FcPatternDestroy(font);
+	    return FcTrue;
+	}
+	top = middle + 1;
+    }
+
+    s->nfont++;
+    int i;
+    for (i= s->nfont; i > top; i--)
+    {
+	s->fonts[i] = s->fonts[i-1];
+    }
+    s->fonts[top] = font;
+
+    return FcTrue;
+}
diff -uNrp fontconfig-2.2.3.orig/src/fcint.h fontconfig-2.2.3/src/fcint.h
--- fontconfig-2.2.3.orig/src/fcint.h	2003-04-16 07:38:06.000000000 +0800
+++ fontconfig-2.2.3/src/fcint.h	2004-11-17 09:53:53.000000000 +0800
@@ -102,6 +102,11 @@ typedef struct _FcSymbolic {
 
 #define FC_MEM_NUM	    29
 
+/* Add by Firefly(firefly@firefly.idv.tw) */
+#define FC_FAMILYOUTPUT_AUTO 0
+#define FC_FAMILYOUTPUT_ENGLISHONLY 1
+#define FC_FAMILYOUTPUT_ANY 2
+
 typedef enum _FcValueBinding {
     FcValueBindingWeak, FcValueBindingStrong, FcValueBindingSame
 } FcValueBinding;
@@ -328,6 +333,9 @@ struct _FcConfig {
      */
     time_t	rescanTime;	    /* last time information was scanned */
     int		rescanInterval;	    /* interval between scans */
+    /* Add by Firefly(firefly@firefly.idv.tw) */
+    /* output family name */
+    int		familyOutput;
 };
  
 extern FcConfig	*_fcConfig;
@@ -497,6 +505,9 @@ const FcCharMap *
 FcFreeTypeGetPrivateMap (FT_Encoding encoding);
     
 /* fcfs.c */
+/* Add by Firefly(firefly@firefly.idv.tw) */
+FcBool
+FcFontSetAddSort(FcFontSet *s, FcPattern *font);
 /* fcgram.y */
 int
 FcConfigparse (void);
diff -uNrp fontconfig-2.2.3.orig/src/fcmatch.c fontconfig-2.2.3/src/fcmatch.c
--- fontconfig-2.2.3.orig/src/fcmatch.c	2003-03-13 06:16:25.000000000 +0800
+++ fontconfig-2.2.3/src/fcmatch.c	2004-11-17 09:53:53.000000000 +0800
@@ -203,7 +203,8 @@ static FcMatcher _FcMatchers [] = {
     { FC_SLANT,		FcCompareNumber,	8, 8 },
 #define MATCH_SLANT	    7
     
-    { FC_WEIGHT,	FcCompareNumber,	9, 9 },
+/* Modify by Firefly(firefly@firefly.idv.tw) */
+    { FC_WEIGHT,	FcCompareNumber,	8, 8 },
 #define MATCH_WEIGHT	    8
     
     { FC_WIDTH,		FcCompareNumber,	10, 10 },
@@ -661,6 +662,11 @@ FcFontSetSort (FcConfig	    *config,
     int		    f;
     int		    i;
 
+    /* Add by Firefly(firefly@firefly.idv.tw)
+	For speed up.
+     */
+    trim = FcTrue;
+
     if (FcDebug () & FC_DBG_MATCH)
     {
 	printf ("Sort ");
diff -uNrp fontconfig-2.2.3.orig/src/fcxml.c fontconfig-2.2.3/src/fcxml.c
--- fontconfig-2.2.3.orig/src/fcxml.c	2004-03-31 00:49:15.000000000 +0800
+++ fontconfig-2.2.3/src/fcxml.c	2004-11-17 09:53:53.000000000 +0800
@@ -311,6 +311,8 @@ typedef enum _FcElement {
 	
     FcElementBlank,
     FcElementRescan,
+    /* Add by Firefly (firefly@firefly.idv.tw) */
+    FcElementFamilyOutput,
 
     FcElementPrefer,
     FcElementAccept,
@@ -367,6 +369,8 @@ FcElementMap (const XML_Char *name)
 	
 	{ "blank",	FcElementBlank },
 	{ "rescan",	FcElementRescan },
+        /* Add by Firefly (firefly@firefly.idv.tw) */
+        { "familyoutput",       FcElementFamilyOutput },
 
 	{ "prefer",	FcElementPrefer },
 	{ "accept",	FcElementAccept },
@@ -894,6 +898,30 @@ FcParseRescan (FcConfigParse *parse)
     }
 }
 
+/* Add by Firefly (firefly@firefly.idv.tw)
+ * ¸ÑªR <familyoutput> ¼ÐÅÒ
+ */
+static void
+FcParseFamilyOutput (FcConfigParse *parse)
+{
+    int n = FcVStackElements (parse);
+    while (n-- > 0)
+    {
+	FcVStack    *v = FcVStackFetch (parse, n);
+	if (v->tag == FcVStackConstant)
+	{
+	    if (strcasecmp(v->u.string, "auto") == 0)
+		parse->config->familyOutput = FC_FAMILYOUTPUT_AUTO;
+	    else if (strcasecmp(v->u.string, "englishonly") == 0)
+		parse->config->familyOutput = FC_FAMILYOUTPUT_ENGLISHONLY;
+	    else if (strcasecmp(v->u.string, "any") == 0)
+		parse->config->familyOutput = FC_FAMILYOUTPUT_ANY;
+	}
+	else
+	    FcConfigMessage (parse, FcSevereWarning, "non-const familyoutput");
+    }
+}
+
 static void
 FcParseInt (FcConfigParse *parse)
 {
@@ -1725,6 +1753,10 @@ FcEndElement(void *userData, const XML_C
     case FcElementRescan:
 	FcParseRescan (parse);
 	break;
+    /* Add by Firefly (firefly@firefly.idv.tw) */
+    case FcElementFamilyOutput:
+	FcParseFamilyOutput (parse);
+	break;
 	
     case FcElementPrefer:
 	FcParseFamilies (parse, FcVStackPrefer);
