--- fontconfig-2.2.3.orig/fonts.conf.in	2003-04-17 06:04:42.000000000 +0800
+++ fontconfig-2.2.3/fonts.conf.in	2004-11-14 16:56:16.134973088 +0800
@@ -24,7 +24,7 @@
 <!-- Font directory list configured on @FC_FONTDATE@ -->
 
 	<dir>@FC_DEFAULT_FONTS@</dir>
-	@FC_FONTPATH@
+	<dir>/usr/X11R6/lib/X11/fonts</dir>
 	<dir>~/.fonts</dir>
 
 <!--
--- fontconfig-2.2.3.orig/local.conf	2003-04-05 04:17:40.000000000 +0800
+++ fontconfig-2.2.3/local.conf	2004-11-14 16:54:28.710304120 +0800
@@ -11,4 +11,82 @@
 		<edit name="rgba" mode="assign"><const>rgb</const></edit>
 	</match>
 -->
+<!--
+  Add by Firefly(firefly@firefly.idv.tw)
+-->
+        <match target="font">
+		<edit name="embeddedbitmap" mod="assign">
+			<bool>true</bool>
+		</edit>
+	</match>
+
+<!--
+  Add by Firefly(firefly@firefly.idv.tw)
+  Most of Asian fonts can't explain by freetype2,
+  so,if these fonts have dual width(half/full) and monospacing,
+  you need to disable globaladvance.
+ -->
+	<match target="font">
+		<test name="spacing" compare="more_eq">
+			<const>mono</const>
+		</test>
+		<edit name="globaladvance" mode="assign">
+			<bool>false</bool>
+		</edit>
+	</match>
+
+<!-- 
+ Add by firefly@firefly.idv.tw
+ Artificial bold for fonts without a bold version.
+ -->
+	<match target="font">
+		<!-- check to see if the pattern requested > "medium" -->
+		<test target="pattern" name="weight" compare="more">
+			<const>medium</const>
+		</test>
+		<!-- pretend the font is bold now -->
+		<edit name="weight" mode="assign">
+			<const>bold</const>
+		</edit>
+	</match>
+<!--
+  Add by firefly@firefly.idv.tw
+-->
+	<match target="pattern">
+		<test name="lang" compare="contains">
+			<string>zh-tw</string>
+			<string>zh-cn</string>
+			<string>zh-hk</string>
+			<string>zh-mo</string>
+			<string>zh-sg</string>
+			<string>ja</string>
+			<string>ko</string>
+		</test>
+		<edit name="hinting" mode="assign">
+			<bool>true</bool>
+		</edit>
+		<edit name="usegamma" mode="assign">
+		    <if>
+			<less>
+				<name>pixelsize</name>
+				<double>20</double>
+			</less>
+		    </if>
+		    <bool>true</bool>
+		    <bool>false</bool>
+		</edit>
+	</match>
+<!--........................................................................-->	
+	<config>
+<!--
+  Add by Firefly (firefly@firefly.idv.tw)
+  Output non English/Latin family name.
+ -->
+		<familyoutput>
+			<const>auto</const>
+			<!--const>englishonly</const-->
+			<!--const>any</const-->
+		</familyoutput>
+	</config>
+
 </fontconfig>
