--- gmlive-0.22/plugins/tvonline/tvonline.py.origin	2009-08-07 14:15:51.993766518 +0800
+++ gmlive-0.22/plugins/tvonline/tvonline.py	2009-08-07 14:16:04.454751992 +0800
@@ -32,11 +32,13 @@
 COL_CLASS
  ) = range(3)
 
-def on_active (tv, path, view_column, totem):
+def on_active (tv, path, view_column, totem_object):
     model = tv.get_model()
     iter = model.get_iter(path)
     url = model.get_value(iter, COL_URL)
-    totem.action_set_mrl_and_play(url)
+#    totem.action_set_mrl_and_play(url)
+    totem_object.action_remote(totem.REMOTE_COMMAND_REPLACE,url)
+    totem_object.action_remote(totem.REMOTE_COMMAND_PLAY,url)
 
 def create_page(totem):
     tv = gtk.TreeView()
