diff --git a/dict/configure.ac b/dict/configure.ac
index fc1a3af..9b1adfa 100644
--- a/dict/configure.ac
+++ b/dict/configure.ac
@@ -425,6 +425,8 @@ fi
 AC_CHECK_LIB([X11], [main], , [AC_MSG_ERROR([X11 lib not found])])
 fi
 
+AC_CHECK_LIB(gmodule-2.0,[main],STARDICT_LIBS="$STARDICT_LIBS -lgmodule-2.0",[AC_MSG_ERROR([gmodule-2.0 not found, or glib not installed])])
+
 AC_SUBST(STARDICT_LIBS)
 AC_SUBST(STARDICT_CFLAGS)
 
