Index: configure.in
===================================================================
--- configure.in	(revision 257)
+++ configure.in	(working copy)
@@ -12,6 +12,7 @@
 
 AC_PATH_PROG(GCONFTOOL, gconftool-2)
 AC_PROG_INTLTOOL([0.22])
+PKG_PROG_PKG_CONFIG
 
 AC_CONFIG_MACRO_DIR(m4)
 AM_PROG_LIBTOOL
@@ -72,13 +73,17 @@
 dnl gucharmap plugin checks.
 dnl ================================================================
 
+GUCHARMAP_REQUIRED=1.4.0
 AC_ARG_ENABLE([gucharmap],
 	AS_HELP_STRING([--disable-gucharmap],[Disable gucharmap plugin (default: enabled)]),
 	[enable_gucharmap=$enableval],
 	[enable_gucharmap=yes])
 
 if test "x$enable_gucharmap" = "xyes" ; then
-	PKG_CHECK_MODULES(GUCHARMAP, [gucharmap >= 1.4.0], have_gucharmap=yes, have_gucharmap=no)
+	PKG_CHECK_MODULES(GUCHARMAP, [gucharmap >= $GUCHARMAP_REQUIRED], have_gucharmap=yes, have_gucharmap=no)
+	if test "x$have_gucharmap" != "xyes"; then
+		PKG_CHECK_MODULES(GUCHARMAP, [gucharmap-2 >= $GUCHARMAP_REQUIRED], have_gucharmap=yes, have_gucharmap=no)
+	fi
 	if test "x$have_gucharmap" = "xyes"; then
 		GUCHARMAP_CFLAGS="${GUCHARMAP_CFLAGS}"
 		GUCHARMAP_LIBS="${GUCHARMAP_LIBS}"
Index: stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp
===================================================================
--- stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp	(revision 257)
+++ stardict-plugins/stardict-gucharmap-plugin/stardict_gucharmap.cpp	(working copy)
@@ -238,7 +238,7 @@
 	/* canonical decomposition */
 	gunichar *decomposition;
 	gsize result_len;
-	decomposition = gucharmap_unicode_canonical_decomposition (uc, &result_len);
+	decomposition = g_unicode_canonical_decomposition (uc, &result_len);
 	if (result_len != 1) {
 		definition += _("Canonical decomposition:");
 		definition += " ";

 	  	 
