diff -rup a/stardict-3.0.4/lib/src/libcommon.h b/stardict-3.0.4/lib/src/libcommon.h
--- a/stardict-3.0.4/lib/src/libcommon.h	2012-01-13 14:21:34.000000000 +0800
+++ b/stardict-3.0.4/lib/src/libcommon.h	2013-01-07 15:03:59.095687537 +0800
@@ -188,7 +188,7 @@ namespace clib {
 }
 
 namespace zip {
-typedef ResourceWrapper<void, void*, int, gzclose> gzFile;
+typedef ResourceWrapper<void, ::gzFile, int, gzclose> gzFile;
 }
 
 /* Create a new temporary file. Return file name in file name encoding.
