--- build-xetex	2008-06-01 01:04:01.000000000 +0800
+++ build-xetex.gentoo	2008-06-02 01:53:27.000000000 +0800
@@ -159,11 +159,11 @@
 fi
 
 # build xdvipdfmx
-(cd "$WORKDIR/texk/xdvipdfmx" && ${GNUMAKE}) || {
-	echo "### failed to build the xdvipdfmx driver"
-	exit 1
-}
-PRODUCTS="${PRODUCTS} $WORKDIR/texk/xdvipdfmx/src/xdvipdfmx"
+#(cd "$WORKDIR/texk/xdvipdfmx" && ${GNUMAKE}) || {
+#	echo "### failed to build the xdvipdfmx driver"
+#	exit 1
+#}
+#PRODUCTS="${PRODUCTS} $WORKDIR/texk/xdvipdfmx/src/xdvipdfmx"
 
 # show the executables we made (assuming the build was successful)
 echo "### Products built successfully:"
