diff --git a/src/lib/data.cpp b/src/lib/data.cpp
index 1b29ec3..cd9b5e9 100644
--- a/src/lib/data.cpp
+++ b/src/lib/data.cpp
@@ -22,6 +22,7 @@
 #  include "config.h"
 #endif
 
+#include <stdio.h>
 #include <cstring>
 #include "kmp.h"
 
diff --git a/src/lib/stardict_client.cpp b/src/lib/stardict_client.cpp
index 12c4aab..8f468b8 100644
--- a/src/lib/stardict_client.cpp
+++ b/src/lib/stardict_client.cpp
@@ -22,6 +22,7 @@
 #  include "config.h"
 #endif
 
+#include <stdio.h>
 #include <cstdlib>
 #include <cstring>
 #include <glib.h>
