--- configure.in.orig	2007-08-13 20:44:31.000000000 +0800
+++ configure.in	2007-08-13 22:12:33.000000000 +0800
@@ -131,7 +131,7 @@
 	else
 		AC_MSG_RESULT(yes)
 		FESTIVAL_LIBS="-lFestival -lestools -lestbase -leststring"
-		FESTIVAL_CFLAGS="-I/usr/include/speech-tools/EST -ffriend-injection -Wno-deprecated"
+		FESTIVAL_CFLAGS="-I/usr/include/speech-tools -ffriend-injection -Wno-deprecated"
 		FESTIVAL_PLUGIN_DIR="stardict-festival-tts-plugin"
 	fi
 else
