--- configure.in.orig	2007-08-13 20:44:31.000000000 +0800
+++ configure.in	2007-08-13 22:12:33.000000000 +0800
@@ -131,7 +131,7 @@
 	else
 		AC_MSG_RESULT(yes)
 		FESTIVAL_LIBS="-lFestival -lestools -lestbase -leststring"
-		FESTIVAL_CFLAGS="-I/usr/include/speech-tools/EST -ffriend-injection -Wno-deprecated"
+		FESTIVAL_CFLAGS="-I/usr/include/speech-tools -ffriend-injection -Wno-deprecated"
 		FESTIVAL_PLUGIN_DIR="stardict-festival-tts-plugin"
 	fi
 else
@@ -192,7 +192,7 @@
 elif test "x${enable_maemo_support}" = "xyes" ; then
   DEP_MODULES="gtk+-2.0 >= 2.6 gconf-2.0 >= 2.6 hildon-libs >= 0.12"
 elif test "x${enable_gnome_support}" = "xno" ; then
-  DEP_MODULES="gtk+-2.0 >= 2.6"
+  DEP_MODULES="gtk+-2.0 >= 2.6 gthread"
 else
   DEP_MODULES="libgnomeui-2.0 >= 2.2.0"
 fi
