--- configure.orig	2007-10-08 16:34:30.000000000 +0800
+++ configure	2007-10-08 16:35:53.000000000 +0800
@@ -23509,7 +23509,7 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; then
-  pkg_cv_MOZILLA_CFLAGS=`$PKG_CONFIG --cflags "$gecko_provider-plugin" 2>/dev/null`
+  pkg_cv_MOZILLA_CFLAGS=`$PKG_CONFIG --cflags "$gecko_provider-plugin $gecko_provider-xpcom" 2>/dev/null`
 else
   pkg_failed=yes
 fi
