--- configure.bak	2008-01-03 11:05:14.000000000 +0800
+++ configure	2008-01-03 11:05:57.000000000 +0800
@@ -27039,8 +27039,7 @@
   { echo "$as_me:$LINENO: result: $gecko_provider found!" >&5
 echo "${ECHO_T}$gecko_provider found!" >&6; }
   if test "$user_specified_mozilla" = "no"; then
-                MOZILLA_INCLUDE_ROOT=`$PKG_CONFIG --cflags-only-I $gecko_provider-plugin | awk '{print $1}' | sed "s/^-I//"`
-    MOZILLA_INCLUDE_ROOT=`dirname $MOZILLA_INCLUDE_ROOT`
+    MOZILLA_INCLUDE_ROOT=`$PKG_CONFIG --cflags-only-I $gecko_provider-plugin $gecko_provider-xpcom`
     MOZILLA_LIB_ROOT=`$PKG_CONFIG --variable=libdir $gecko_provider-plugin`
     MOZILLA_LIB_DIR_SEARCH="$MOZILLA_LIB_ROOT"
   else
