diff -uNrp freetype-2.1.6.orig/include/freetype/config/ftoption.h freetype-2.1.6/include/freetype/config/ftoption.h
--- freetype-2.1.6.orig/include/freetype/config/ftoption.h	2003-09-10 04:11:56.000000000 +0800
+++ freetype-2.1.6/include/freetype/config/ftoption.h	2003-11-10 12:50:13.000000000 +0800
@@ -399,7 +399,7 @@ FT_BEGIN_HEADER
   /*   Do not #undef this macro here, since the build system might         */
   /*   define it for certain configurations only.                          */
   /*                                                                       */
-/* #define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
