diff -urN emesene-orig/emesenelib/MsnOIM.py emesene/emesenelib/MsnOIM.py
--- emesene-orig/emesenelib/MsnOIM.py	2009-04-08 20:13:14.000000000 +0800
+++ emesene/emesenelib/MsnOIM.py	2009-05-19 12:45:24.000000000 +0800
@@ -18,7 +18,8 @@
 
 import email
 import time
-from uuid import uuid4
+from uuid import UUID
+from os import urandom
 from string import Template
 from xml.parsers.expat import ExpatError
 from xml.dom.minidom import parseString
@@ -134,8 +135,8 @@
                  to = to, passport = self._passportid,
                  appid = SignalHandler._PRODUCT_ID,
                  lockKey = self._lockKey, seqNum = self._seq,
-                 runId = uuid4().__str__(), content = msg.encode('base64')\
-                          .strip())
+                 runId = UUID(bytes=urandom(16), version=4).__str__(),\
+                         content = msg.encode('base64').strip())
 
         self._seq=self._seq+1
         self.soapRequest('http://messenger.live.com/ws/2006/09/oim/Store2',
