diff -urN emesene-orig/emesenelib/soap/manager.py emesene/emesenelib/soap/manager.py
--- emesene-orig/emesenelib/soap/manager.py	2008-05-11 02:12:46.000000000 +0800
+++ emesene/emesenelib/soap/manager.py	2008-05-11 02:21:13.000000000 +0800
@@ -35,8 +35,8 @@
 def get():
     global responseQueue
     retval = responseQueue.get()
-    if isinstance(request, SoapRequest):
-        common.debug("soap.manager.get() --> " + retval, "soap")
+    if isinstance(retval, SoapRequest):
+        common.debug("soap.manager.get() --> %s" % retval, "soap")
     return retval
 
 def getNoBlock():
