diff -urN emesene-orig/plugins_base/Sound.py emesene/plugins_base/Sound.py
--- emesene-orig/plugins_base/Sound.py	2008-05-21 00:20:39.000000000 +0800
+++ emesene/plugins_base/Sound.py	2008-05-21 00:24:16.000000000 +0800
@@ -77,7 +77,7 @@
             self.isMac = True
         elif self.is_on_path('aplay'):
             self.canPlay = True
-            self.command = 'aplay'
+            self.command = 'aplay --buffer-size=16'
         elif self.is_on_path('play'):
             self.canPlay = True
             self.command = 'play'
