diff -urN emesene-orig/setup.py emesene-fix/setup.py
--- emesene-orig/setup.py	2008-05-20 00:46:58.984138436 +0800
+++ emesene-fix/setup.py	2008-05-20 01:33:23.608138109 +0800
@@ -26,8 +26,8 @@
       license      = 'GNU GPL 2',
       requires     = ["gtk"],
       platforms    = ["Platform Independent"],
-      packages     = ['', 'abstract', 'emesenelib', 'plugins_base', 'plugins_base.currentSong'],
-      package_data = {'./': ['conversation_themes/*/*', 'smilies/*/*',
+      packages     = ['', 'abstract', 'emesenelib','emesenelib.p2p','emesenelib.soap','plugins_base', 'plugins_base.currentSong'],
+      package_data = {'': ['conversation_themes/*/*', 'smilies/*/*',
                            'sound_themes/*/*', 'themes/*/*', 'hotmlog.htm']},
       scripts      = ['emesene'],
       data_files   = [('share/pixmaps', ['misc/emesene.png']),
