diff -urN emesene-orig/emesenelib/Switchboard.py emesene/emesenelib/Switchboard.py
--- emesene-orig/emesenelib/Switchboard.py	2008-05-11 02:12:46.000000000 +0800
+++ emesene/emesenelib/Switchboard.py	2008-05-11 02:16:56.000000000 +0800
@@ -277,6 +277,10 @@
                 except IndexError:
                     charset = ''
                 
+                if 'P4-Context' in header:
+                    nick += header['P4-Context']
+                    body = "[" + header['P4-Context'] + "]: " + body
+                                
                 self.emit('message', tid, nick, body, format, charset)
                 self.msn.emit('message-received', tid)
 
