diff --git a/examples/FACEBOOK b/examples/FACEBOOK
new file mode 100644
index 0000000..6485d34
--- /dev/null
+++ b/examples/FACEBOOK
@@ -0,0 +1,5 @@
+-N FACEBOOK nethash --hashsize 50 --probes 1
+-A FACEBOOK 69.63.176.0/20
+-A FACEBOOK 66.220.144.0/20
+-A FACEBOOK 204.15.20.0/22
+COMMIT
diff --git a/examples/NOCLIP b/examples/NOCLIP
index 6c2b717..8e41765 100644
--- a/examples/NOCLIP
+++ b/examples/NOCLIP
@@ -1,4 +1,5 @@
 -N NOCLIP setlist --size 4
 -A NOCLIP GOOGLE
 -A NOCLIP YOUTUBE
+-A NOCLIP FACEBOOK
 COMMIT
