diff --git a/examples/WIKIPEDIA b/examples/WIKIPEDIA
new file mode 100644
index 0000000..6485d34
--- /dev/null
+++ b/examples/WIKIPEDIA
@@ -0,0 +1,4 @@
+-N WIKIPEDIA nethash --hashsize 50 --probes 1 --resize 50
+-A WIKIPEDIA 208.80.152.0/22
+-A WIKIPEDIA 91.198.174.0/24
+COMMIT
diff --git a/examples/NOCLIP b/examples/NOCLIP
index 6c2b717..8e41765 100644
--- a/examples/NOCLIP
+++ b/examples/NOCLIP
@@ -2,5 +2,6 @@
 -A NOCLIP GOOGLE
 -A NOCLIP YOUTUBE
 -A NOCLIP FACEBOOK
+-A NOCLIP WIKIPEDIA
 COMMIT
