diff -Naur sdcv-0.4.2.orig/src/lib/lib.cpp sdcv-0.4.2/src/lib/lib.cpp
--- sdcv-0.4.2.orig/src/lib/lib.cpp	2006-04-24 12:17:27.000000000 +0400
+++ sdcv-0.4.2/src/lib/lib.cpp	2008-07-04 18:29:37.000000000 +0400
@@ -512,7 +512,7 @@
 inline const gchar *offset_index::read_first_on_page_key(glong page_idx)
 {
 	fseek(idxfile, wordoffset[page_idx], SEEK_SET);
-	guint32 page_size=wordoffset[page_idx+1]-wordoffset[page_idx];
+	unsigned long page_size=wordoffset[page_idx+1]-wordoffset[page_idx];
 	fread(wordentry_buf, std::min(sizeof(wordentry_buf), page_size), 1, idxfile); //TODO: check returned values, deal with word entry that strlen>255.
 	return wordentry_buf;
 }
