diff -Nur notification-daemon-0.4.0/src/daemon/daemon.c notification-daemon-0.4.0_patched/src/daemon/daemon.c
--- notification-daemon-0.4.0/src/daemon/daemon.c	2008-09-26 04:27:07.000000000 +0300
+++ notification-daemon-0.4.0_patched/src/daemon/daemon.c	2009-05-02 18:50:32.000000000 +0300
@@ -1370,14 +1370,23 @@
 gboolean
 notify_daemon_get_capabilities(NotifyDaemon *daemon, char ***caps)
 {
+#ifdef HAVE_GSTREAMER
+	*caps = g_new0(char *, 7);
+#else
 	*caps = g_new0(char *, 6);
+#endif
 
 	(*caps)[0] = g_strdup("actions");
 	(*caps)[1] = g_strdup("body");
 	(*caps)[2] = g_strdup("body-hyperlinks");
 	(*caps)[3] = g_strdup("body-markup");
 	(*caps)[4] = g_strdup("icon-static");
+#ifdef HAVE_GSTREAMER
+	(*caps)[5] = g_strdup("sound");
+	(*caps)[6] = NULL;
+#else
 	(*caps)[5] = NULL;
+#endif
 
 	return TRUE;
 }
