--- a/tools/upgrade.sh
+++ b/tools/upgrade.sh
@@ -25,18 +25,10 @@
 
   printf '%s\n' "${BLUE}Updating Oh My Bash${NORMAL}"
 
-  # Note: The git option "-C PATH" is only supported from git-1.8.5
-  # (https://github.com/git/git/commit/44e1e4d6 2013-09).  On the other hand,
-  # the synonym "--git-dir=PATH/.git --work-tree=PATH" is supported from
-  # git-1.5.3 (https://github.com/git/git/commit/892c41b9 2007-06).
-  if ! command git --git-dir="$OSH/.git" --work-tree="$OSH" pull --rebase --stat origin master; then
-    # In case it enters the rebasing mode
-    printf '%s\n' "oh-my-bash: running 'git rebase --abort'..."
-    command git --git-dir="$OSH/.git" --work-tree="$OSH" rebase --abort
+  if ! command smart-live-rebuild -f app-shells/oh-my-bash; then
     printf "${RED}%s${NORMAL}\n" \
            'There was an error updating.' \
-           "If you have uncommited changes in '$BOLD$OSH$NORMAL$RED', please commit, stash or discard them and retry updating." \
-           "If you have your own local commits in '$BOLD$OSH$NORMAL$RED' that conflict with the upstream changes, please resolve conflicts and merge the upstream manually."
+           "If app-portage/smart-live-rebuild is not installed, please install and retry updating."
     return 1
   fi
 
