Index: qt-x11-opensource-src-4.5.0-beta1/src/gui/text/qfont.cpp
===================================================================
--- qt-x11-opensource-src-4.5.0-beta1.orig/src/gui/text/qfont.cpp
+++ qt-x11-opensource-src-4.5.0-beta1/src/gui/text/qfont.cpp
@@ -1767,7 +1767,7 @@ static void initFontSubst()
     // default substitutions
     static const char *initTbl[] = {
 
-#if defined(Q_WS_X11)
+#if 0
         "arial",        "helvetica",
         "times new roman", "times",
         "courier new",  "courier",
