diff -up qt-x11-opensource-src-4.5.3/src/corelib/kernel/qeventdispatcher_glib.cpp.than qt-x11-opensource-src-4.5.3/src/corelib/kernel/qeventdispatcher_glib.cpp
--- qt-x11-opensource-src-4.5.3/src/corelib/kernel/qeventdispatcher_glib.cpp.than	2009-10-29 16:15:58.000000000 +0100
+++ qt-x11-opensource-src-4.5.3/src/corelib/kernel/qeventdispatcher_glib.cpp	2009-10-29 16:16:22.000000000 +0100
@@ -263,7 +263,6 @@ QEventDispatcherGlibPrivate::QEventDispa
     (void) new (&timerSource->timerList) QTimerInfoList();
     timerSource->processEventsFlags = QEventLoop::AllEvents;
     g_source_set_can_recurse(&timerSource->source, true);
-    g_source_set_priority(&timerSource->source, G_PRIORITY_DEFAULT_IDLE);
     g_source_attach(&timerSource->source, mainContext);
 }
 
