Bug 220751, Mikko Cal.

--- src/gui/itemviews/qabstractitemview.cpp	2008-04-28 15:11:18.000000000 +0200
+++ src/gui/itemviews/qabstractitemview.cpp	2008-04-10 01:00:58.000000000 +0200
@@ -2980,7 +2982,7 @@
     if (q->isVisible()) {
         q->doItemsLayout();
         const QModelIndex current = q->currentIndex();
-        if (current.isValid())
+        if (current.isValid() && state == QAbstractItemView::EditingState)
             q->scrollTo(current);
     } else {
         doDelayedItemsLayout();
