--- src/tools/designer/src/components/propertyeditor/propertyeditor.cpp	2008-08-14 10:25:28 -0000
+++ src/tools/designer/src/components/propertyeditor/propertyeditor.cpp	2008-08-14 10:25:28 -0000

@@ -656,7 +656,7 @@
 
 void PropertyEditor::updateColors()
 {
-    if (m_currentBrowser == m_treeBrowser) {
+    if (m_treeBrowser && m_currentBrowser == m_treeBrowser) {
         QList<QtBrowserItem *> items = m_treeBrowser->topLevelItems();
         QListIterator<QtBrowserItem *> itItem(items);
         while (itItem.hasNext()) {
