Index: qt-x11-opensource-src-4.5.0/tools/linguist/lrelease/main.cpp
===================================================================
--- qt-x11-opensource-src-4.5.0.orig/tools/linguist/lrelease/main.cpp
+++ qt-x11-opensource-src-4.5.0/tools/linguist/lrelease/main.cpp
@@ -88,6 +88,8 @@ static bool loadTsFile(Translator &tor, 
     bool ok = tor.load(tsFileName, cd, QLatin1String("auto"));
     if (!ok) {
         qWarning("lrelease error: %s\n", qPrintable(cd.error()));
+	// dont fail
+	ok = true;
     } else {
         if (!cd.errors().isEmpty())
             printOut(cd.error());
@@ -102,7 +104,8 @@ static bool loadTsFile(Translator &tor, 
                 if (!msg.comment().isEmpty())
                     qWarning("\n* Comment: %s", qPrintable(msg.comment()));
             }
-            ok = false;
+	    // dont fail
+            //ok = false;
         }
     }
     return ok;
