diff -Nur qt-everywhere-opensource-src-4.8.0.orig/src/gui/text/qfontengine_ft.cpp qt-everywhere-opensource-src-4.8.0/src/gui/text/qfontengine_ft.cpp
--- qt-everywhere-opensource-src-4.8.0.orig/src/gui/text/qfontengine_ft.cpp	2011-12-08 13:06:02.000000000 +0800
+++ qt-everywhere-opensource-src-4.8.0/src/gui/text/qfontengine_ft.cpp	2012-01-31 10:48:47.050071713 +0800
@@ -800,6 +800,7 @@
 
     if (set && set->outline_drawing)
         load_flags = FT_LOAD_NO_BITMAP;
+    load_flags |= FT_LOAD_IGNORE_GLOBAL_ADVANCE_WIDTH;
 
     if (default_hint_style == HintNone || (flags & HB_ShaperFlag_UseDesignMetrics) || (set && set->outline_drawing))
         load_flags |= FT_LOAD_NO_HINTING;
@@ -858,6 +859,7 @@
 
     if (transform)
         load_flags |= FT_LOAD_NO_BITMAP;
+    load_flags |= FT_LOAD_IGNORE_GLOBAL_ADVANCE_WIDTH;
 
     FT_Face face = freetype->face;
 
