--- src/gui/kernel/qapplication_x11.cpp.orig	2007-12-23 15:10:43.000000000 -0500
+++ src/gui/kernel/qapplication_x11.cpp	2007-12-23 15:11:01.000000000 -0500
@@ -1715,9 +1715,9 @@
                 (PtrXineramaQueryScreens) xineramaLib.resolve("XineramaQueryScreens");
         }
 #else
-        ptrXineramaQueryScreens = XineramaQueryScreens;
-        ptrXineramaIsActive = XineramaIsActive;
-        ptrXineramaQueryExtension = XineramaQueryExtension;
+        X11->ptrXineramaQueryScreens = XineramaQueryScreens;
+        X11->ptrXineramaIsActive = XineramaIsActive;
+        X11->ptrXineramaQueryExtension = XineramaQueryExtension;
 #endif // QT_RUNTIME_XINERAMA
 #endif // QT_NO_XINERAMA
 
