diff -uNr gftp-2.0.18/src/gtk/options_dialog.c gftp-2.0.18.patched/src/gtk/options_dialog.c
--- gftp-2.0.18/src/gtk/options_dialog.c	2005-01-04 14:32:20.000000000 +0100
+++ gftp-2.0.18.patched/src/gtk/options_dialog.c	2005-02-14 16:44:10.064677160 +0100
@@ -1002,7 +1002,7 @@
   gtk_widget_show (box);
   
 #if GTK_MAJOR_VERSION == 1
-  tempwid = gtk_label_new (_("Type:"))
+  tempwid = gtk_label_new (_("Type:"));
 #else
   tempwid = gtk_label_new_with_mnemonic (_("_Type:"));
 #endif
diff -uNr gftp-2.0.18/src/text/gftp-text.c gftp-2.0.18.patched/src/text/gftp-text.c
--- gftp-2.0.18/src/text/gftp-text.c	2005-01-25 02:11:00.000000000 +0100
+++ gftp-2.0.18.patched/src/text/gftp-text.c	2005-02-14 16:44:12.299337440 +0100
@@ -185,6 +185,7 @@
   else
     infd = stdin;
 
+#if GLIB_MAJOR_VERSION > 1
   locale_question = g_locale_from_utf8 (question, -1, NULL, NULL, NULL);
   if (locale_question != NULL)
     {
@@ -193,6 +194,7 @@
       g_free (locale_question);
     }
   else
+#endif
     printf ("%s%s%s ", GFTPUI_COMMON_COLOR_BLUE, question,
             GFTPUI_COMMON_COLOR_DEFAULT);
 
