--- freetype-2.3.7/src/cff/cffparse.c.old	2008/08/04 15:54:24	1.43
+++ freetype-2.3.7/src/cff/cffparse.c	2008/11/27 21:55:20	1.44
@@ -355,6 +355,12 @@
       if ( FT_ABS( integer_length ) > 5 )
         goto Exit;
 
+      /* Remove non-significant digits. */
+      if ( integer_length < 0 ) {
+        number          /= power_tens[-integer_length];
+        fraction_length += integer_length;
+      }
+
       /* Convert into 16.16 format. */
       if ( fraction_length > 0 )
       {
