--- freetype-2.1.9-r1.orig/src/raster/ftraster.c	2003-10-26 11:26:52.000000000 -0500
+++ freetype-2.1.9-r1/src/raster/ftraster.c	2003-10-26 11:33:12.000000000 -0500
@@ -2218,7 +2218,7 @@
           /*          bit problem in the `7' of verdana 10pts, but   */
           /*          makes a new one in the `C' of arial 14pts      */
 
-#if 0
+#if 1 
           if ( x2 - x1 < ras.precision_half )
 #endif
           {
@@ -2384,13 +2384,17 @@
           /*                                                        */
 
           /* rightmost stub test */
+#if 1
+          if ( x2 - x1 < ras.precision_half )
+#endif
+	{
           if ( left->next == right && left->height <= 0 )
             return;
 
           /* leftmost stub test */
           if ( right->next == left && left->start == y )
             return;
-
+	}
           /* check that the rightmost pixel isn't set */
 
           e1 = TRUNC( e1 );
