diff -rup freetype-2.2.1.orig/src/autofit/aflatin.c freetype-2.2.1/src/autofit/aflatin.c
--- freetype-2.2.1.orig/src/autofit/aflatin.c	2006-08-22 22:30:15.396263448 +0800
+++ freetype-2.2.1/src/autofit/aflatin.c	2006-08-22 22:32:01.469137920 +0800
@@ -1344,14 +1344,14 @@
      *  We snap the width of vertical stems for the monochrome and
      *  horizontal LCD rendering targets only.
      */
-    if ( mode == FT_RENDER_MODE_MONO || mode == FT_RENDER_MODE_LCD )
+    if ( mode == FT_RENDER_MODE_MONO )
       other_flags |= AF_LATIN_HINTS_HORZ_SNAP;
 
     /*
      *  We snap the width of horizontal stems for the monochrome and
      *  vertical LCD rendering targets only.
      */
-    if ( mode == FT_RENDER_MODE_MONO || mode == FT_RENDER_MODE_LCD_V )
+    if ( mode == FT_RENDER_MODE_MONO )
       other_flags |= AF_LATIN_HINTS_VERT_SNAP;
 
     /*
