diff --git a/src/statusnotifier.cpp b/src/statusnotifier.cpp
index 1e608f2..8d36b09 100644
--- a/src/statusnotifier.cpp
+++ b/src/statusnotifier.cpp
@@ -44,7 +44,7 @@ StatusNotifier::~StatusNotifier()
 #endif
 }
 
-const QStringList StatusNotifier::appIndicatorDE = QStringList() << "Unity" << "XFCE" << "Pantheon" << "GNOME" << "LXDE";
+const QStringList StatusNotifier::appIndicatorDE = QStringList() << "Unity" << "XFCE" << "Pantheon" << "LXDE";
 
 #ifdef Q_OS_UNIX
 void onAppIndicatorActivated(GtkMenuItem *, gpointer data)
