diff -rup a/cnijfilter-3.40/backend/src/cnij_backend_common.c b/cnijfilter-3.40/backend/src/cnij_backend_common.c
--- a/cnijfilter-3.40/backend/src/cnij_backend_common.c	2010-02-23 11:57:49.000000000 +0800
+++ b/cnijfilter-3.40/backend/src/cnij_backend_common.c	2012-06-23 16:08:18.394957019 +0800
@@ -34,6 +34,8 @@
 #include <fcntl.h>
 #include <termios.h>
 #include <sys/stat.h>
+#include <cups/ppd.h>
+
 
 // CUPS Header
 #include <cups/cups.h>
