Only in b/cnijfilter-3.40/cnijfilter/src: *
diff -rup a/cnijfilter-3.40/cnijfilter/src/bjfimage.c b/cnijfilter-3.40/cnijfilter/src/bjfimage.c
--- a/cnijfilter-3.40/cnijfilter/src/bjfimage.c	2010-02-19 13:29:46.000000000 +0800
+++ b/cnijfilter-3.40/cnijfilter/src/bjfimage.c	2012-06-23 16:17:35.780928531 +0800
@@ -1574,7 +1574,7 @@ static short png_image_init( LPBJF_IMAGE
 		goto onErr;
 	}
 
-	if (setjmp (png_p->jmpbuf))
+	if (setjmp ( png_jmpbuf(png_p)))
 	{
 		png_destroy_read_struct(&png_p, &info_p, (png_infopp)NULL);
 		goto onErr;
