diff --git a/deps/zigini/build.zig.zon b/deps/zigini/build.zig.zon
index 3d5d17e..cb50bd3 100644
--- a/deps/zigini/build.zig.zon
+++ b/deps/zigini/build.zig.zon
@@ -3,8 +3,7 @@
     .version = "0.3.0",
     .dependencies = .{
         .ini = .{
-            .url = "https://github.com/ziglibs/ini/archive/e18d36665905c1e7ba0c1ce3e8780076b33e3002.tar.gz",
-            .hash = "1220b0979ea9891fa4aeb85748fc42bc4b24039d9c99a4d65d893fb1c83e921efad8",
+            .path = "./ini",
         },
     },
     .paths = .{
