diff -ru cairo-1.8.8.orig/build/configure.ac.tools cairo-1.8.8/build/configure.ac.tools
--- cairo-1.8.8.orig/build/configure.ac.tools	2009-09-30 13:36:42 +0200
+++ cairo-1.8.8/build/configure.ac.tools	2009-09-30 13:50:50 +0200
@@ -21,5 +21,12 @@
 *) PKGCONFIG_REQUIRES="Requires.private"; ;;
 esac
 
+dnl hmm... on interix, things go really bad with Requires.private, since libpng12
+dnl is missing on the final link commands, so gtk+'s configure checks for cairo
+dnl fail miserably with unresolved symbols to it.
+case "$host_os" in
+interix*) PKGCONFIG_REQUIRES="Requires" ;;
+esac
+
 AC_SUBST(PKGCONFIG_REQUIRES)
 
