From d00095b305d452b6c00c585debfe517c786ce37f Mon Sep 17 00:00:00 2001
From: microcai <microcaicai@gmail.com>
Date: Thu, 14 Jun 2012 23:46:44 +0800
Subject: [PATCH] b

---
 src/cairo-xlib-display.c |    4 ----
 1 files changed, 0 insertions(+), 4 deletions(-)

diff --git a/src/cairo-xlib-display.c b/src/cairo-xlib-display.c
index e685c7c..9e1342f 100644
--- a/src/cairo-xlib-display.c
+++ b/src/cairo-xlib-display.c
@@ -266,11 +266,7 @@ _cairo_xlib_device_create (Display *dpy)
     /* Prior to Render 0.10, there is no protocol support for gradients and
      * we call function stubs instead, which would silently consume the drawing.
      */
-#if RENDER_MAJOR == 0 && RENDER_MINOR < 10
     display->buggy_gradients = TRUE;
-#else
-    display->buggy_gradients = FALSE;
-#endif
     display->buggy_pad_reflect = FALSE;
     display->buggy_repeat = FALSE;
 
-- 
1.7.8.6

