diff --git a/fastboot/fastboot.c b/fastboot/fastboot.c
index 73a6e56..d9c0cfc 100644
--- a/fastboot/fastboot.c
+++ b/fastboot/fastboot.c
@@ -445,7 +445,7 @@ static int setup_requirement_line(char *name)
     char *val[MAX_OPTIONS];
     const char **out;
     char *prod = NULL;
-    unsigned n, count;
+    unsigned n, i, count;
     char *x;
     int invert = 0;
 
@@ -492,7 +492,7 @@ static int setup_requirement_line(char *name)
     for(n = 0; n < count; n++) {
         out[n] = strdup(strip(val[n]));
         if (out[n] == 0) {
-            for(size_t i = 0; i < n; ++i) {
+            for(i = 0; i < n; ++i) {
                 free((char*) out[i]);
             }
             free(out);
