diff -urN a/kernel/power/tuxonice_highlevel.c b/kernel/power/tuxonice_highlevel.c
--- a/kernel/power/tuxonice_highlevel.c	2014-05-24 06:01:34.749222646 +0800
+++ b/kernel/power/tuxonice_highlevel.c	2014-05-24 06:04:01.871226898 +0800
@@ -1188,7 +1188,7 @@
 	 * If the program reports an error, retval = 256. Don't complain
 	 * about that here.
 	 */
-	if (retval && retval != 256)
+	if (retval && retval != 256 && retval != -2)
 		printk(KERN_ERR "Failed to launch userspace program '%s': "
 				"Error %d\n", command, retval);
 
