diff --git a/RedPandaIDE/compiler/compiler.cpp b/RedPandaIDE/compiler/compiler.cpp
--- a/RedPandaIDE/compiler/compiler.cpp
+++ b/RedPandaIDE/compiler/compiler.cpp
@@ -726,8 +726,13 @@ void Compiler::runCommand(const QString &cmd, const QStringList &arguments, cons
     errorOccurred = false;
     process.setProgram(cmd);
     QString cmdDir = extractFileDir(cmd);
+#ifdef Q_OS_WINDOWS
     bool compilerErrorUTF8=compilerSet()->isCompilerUsingUTF8();
     bool outputUTF8=compilerSet()->isCompilerUsingUTF8();
+#else
+    bool compilerErrorUTF8=compilerSet()->isCompilerInfoUsingUTF8();
+    bool outputUTF8=compilerSet()->isCompilerInfoUsingUTF8();
+#endif
     QProcessEnvironment env = QProcessEnvironment::systemEnvironment();
 #ifdef Q_OS_WIN
     QStringList binDirs=compilerSet()->binDirs();
diff --git a/RedPandaIDE/debugger/debugger.cpp b/RedPandaIDE/debugger/debugger.cpp
--- a/RedPandaIDE/debugger/debugger.cpp
+++ b/RedPandaIDE/debugger/debugger.cpp
@@ -98,8 +98,13 @@ bool Debugger::startClient(int compilerSetIndex,
                               tr("No compiler set is configured.")+tr("Can't start debugging."));
         return false;
     }
+#ifdef Q_OS_WINDOWS
     setForceUTF8(compilerSet->isDebuggerUsingUTF8());
     setDebugInfosUsingUTF8(compilerSet->isCompilerUsingUTF8());
+#else
+    setForceUTF8(compilerSet->isDebugInfoUsingUTF8());
+    setDebugInfosUsingUTF8(compilerSet->isCompilerInfoUsingUTF8());
+#endif
     if (compilerSet->debugger().endsWith(LLDB_MI_PROGRAM))
         setDebuggerType(DebuggerType::LLDB_MI);
     else
diff --git a/RedPandaIDE/settings.cpp b/RedPandaIDE/settings.cpp
--- a/RedPandaIDE/settings.cpp
+++ b/RedPandaIDE/settings.cpp
@@ -3010,6 +3010,6 @@ bool Settings::CompilerSet::supportConvertingCharset()
     return mGccSupportConvertingCharset;
 #else
     // Unix: iconv is a part of POSIX standard.
-    return mCompilerType == CompilerType::GCC || mCompilerType == CompilerType::GCC_UTF8;
+    return mCompilerType == CompilerType::GCC;
 #endif
 }
diff --git a/RedPandaIDE/widgets/darkfusionstyle.cpp b/RedPandaIDE/widgets/darkfusionstyle.cpp
--- a/RedPandaIDE/widgets/darkfusionstyle.cpp
+++ b/RedPandaIDE/widgets/darkfusionstyle.cpp
@@ -74,7 +74,7 @@ static QString uniqueName(const QString &key, const QStyleOption *option, const
     const QStyleOptionComplex *complexOption = qstyleoption_cast<const QStyleOptionComplex *>(option);
     QString tmp = QString("%1%2%3%4%5%6%7")
             .arg(key)
-            .arg(option->state)
+            .arg(option->state.toInt())
             .arg(option->direction)
             .arg(complexOption ? uint(complexOption->activeSubControls) : 0u)
             .arg(option->palette.cacheKey())
@@ -85,7 +85,7 @@ static QString uniqueName(const QString &key, const QStyleOption *option, const
     if (const QStyleOptionSpinBox *spinBox = qstyleoption_cast<const QStyleOptionSpinBox *>(option)) {
         tmp = tmp + QString("%1%2%3")
                 .arg(spinBox->buttonSymbols)
-                .arg(spinBox->stepEnabled)
+                .arg(spinBox->stepEnabled.toInt())
                 .arg(QLatin1Char(spinBox->frame ? '1' : '0'));
     }
 #endif // QT_CONFIG(spinbox)
