--- tcl/unix/Makefile.in.orig	2007-03-07 19:36:22.000000000 -0500
+++ tcl/unix/Makefile.in	2007-03-07 19:36:36.000000000 -0500
@@ -1051,7 +1051,7 @@
 # END REDHAT LOCAL
 
 tclUnixInit.o: $(UNIX_DIR)/tclUnixInit.c $(GENERIC_DIR)/tclInitScript.h tclConfig.sh
-	$(CC) -c $(CC_SWITCHES) -DTCL_LIBRARY=\"\" \
+	$(CC) -c $(CC_SWITCHES) -DTCL_LIBRARY=\"${TCL_LIBRARY}\" \
 		-DTCL_PACKAGE_PATH="\"${TCL_PACKAGE_PATH}\"" \
 		$(UNIX_DIR)/tclUnixInit.c
 
