diff -urN qterm-0.5.2.orig/cmake/FindOpenSSL.cmake qterm-0.5.2/cmake/FindOpenSSL.cmake
--- qterm-0.5.2.orig/cmake/FindOpenSSL.cmake	2008-05-12 00:07:32.000000000 +0800
+++ qterm-0.5.2/cmake/FindOpenSSL.cmake	2008-06-06 21:56:41.000000000 +0800
@@ -5,38 +5,6 @@
 #  OPENSSL_INCLUDE_DIR - the OpenSSL include directory
 #  OPENSSL_LIBRARIES - The libraries needed to use OpenSSL
 
-if (OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES)
+INCLUDE( FindPkgConfig )
+pkg_search_module(OPENSSL "openssl")
 
-  # in cache already
-  SET(OPENSSL_FOUND TRUE)
-
-else (OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES)
-
-  FIND_PATH(OPENSSL_INCLUDE_DIR openssl/ssl.h
-     /usr/include/
-     /usr/local/include/
-  )
-  
-  FIND_LIBRARY(OPENSSL_LIBRARIES NAMES ssl ssleay32
-     PATHS
-     /usr/lib
-     /usr/local/lib
-  )
-  
-  if (OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES)
-     set(OPENSSL_FOUND TRUE)
-  endif (OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES)
-  
-  if (OPENSSL_FOUND)
-     if (NOT OpenSSL_FIND_QUIETLY)
-        message(STATUS "Found OpenSSL: ${OPENSSL_LIBRARIES}")
-     endif (NOT OpenSSL_FIND_QUIETLY)
-  else (OPENSSL_FOUND)
-     if (OpenSSL_FIND_REQUIRED)
-        message(FATAL_ERROR "Could NOT find OpenSSL")
-     endif (OpenSSL_FIND_REQUIRED)
-  endif (OPENSSL_FOUND)
-  
-  MARK_AS_ADVANCED(OPENSSL_INCLUDE_DIR OPENSSL_LIBRARIES)
-  
-endif (OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES)
