--- build/common.gypi.orig	2010-04-19 18:19:49.000000000 +0200
+++ build/common.gypi	2010-04-19 18:20:41.000000000 +0200
@@ -881,24 +881,24 @@
             # used during computation does not change depending on how the
             # compiler optimized the code, since the value is always kept
             # in its specified precision.
-            'conditions': [
-              ['branding=="Chromium"', {
-                'cflags': [
-                  '-march=pentium4',
-                  '-msse2',
-                  '-mfpmath=sse',
-                ],
-              }],
-            ],
+            #'conditions': [
+            #  ['branding=="Chromium"', {
+            #    'cflags': [
+            #      '-march=pentium4',
+            #      '-msse2',
+            #      '-mfpmath=sse',
+            #    ],
+            #  }],
+            #],
             # -mmmx allows mmintrin.h to be used for mmx intrinsics.
             # video playback is mmx and sse2 optimized.
-            'cflags': [
-              '-m32',
-              '-mmmx',
-            ],
-            'ldflags': [
-              '-m32',
-            ],
+            #'cflags': [
+            #  '-m32',
+            #  '-mmmx',
+            #],
+            #'ldflags': [
+            #  '-m32',
+            #],
           }],
           ['target_arch=="arm"', {
             'target_conditions': [
