--- third_party/WebKit/WebCore/platform/image-encoders/skia/PNGImageEncoder.cpp.orig	2010-04-03 18:37:29.000000000 +0200
+++ third_party/WebKit/WebCore/platform/image-encoders/skia/PNGImageEncoder.cpp	2010-04-03 18:38:20.000000000 +0200
@@ -130,9 +130,9 @@
     }
 
     png_struct* pngPtr = png_create_write_struct(PNG_LIBPNG_VER_STRING,
-                                                 png_voidp_NULL,
-                                                 png_error_ptr_NULL,
-                                                 png_error_ptr_NULL);
+                                                 0,
+                                                 0,
+                                                 0);
     if (!pngPtr)
         return false;
 
