diff --git a/configure.in b/configure.in
index 6435eb0..194da35 100644
--- a/configure.in
+++ b/configure.in
@@ -26,7 +26,7 @@ AM_PROG_LIBTOOL
 PKG_CHECK_MODULES(PACKAGE, [gtk+-2.0 gdk-2.0])
 AC_SUBST(PACKAGE_CFLAGS)
 AC_SUBST(PACKAGE_LIBS)
-
+AC_HAVE_LIBRARY(pthread, LIBS="$LIBS -lpthread", AC_MSG_ERROR(Can't find detect -library!))
 
 dnl Set PACKAGE_LOCALE_DIR in config.h.
 if test "x${prefix}" = "xNONE"; then
