--- /tmp/llk_linux-2.3/src/main.c	2006-01-09 22:30:03.000000000 +0800
+++ src/main.c	2008-03-04 17:35:58.000000000 +0800
@@ -1962,7 +1962,10 @@
       insert_pos++; 
     }
     if(insert_pos > 10)
-    {insert_pos=0;score_item_tmp=NULL;}
+    {/*insert_pos=0;score_item_tmp=NULL;}*/
+      g_free (path);
+      return;/* If player's score is not within 10,no record it,and no show the score window. */
+    }
     else
     {
       if(insert_pos == 1)
@@ -2048,11 +2051,6 @@
         g_print(_("Open score.dat error.\n"));
       }
     }
-    else
-    {
-	  g_free (path);
-      return;/* If player's score is not within 10,no record it,and no show the score window. */
-    }
   }
   g_free (path);
   /* Create a new window to show scores. */

