diff -urNp chmsee-0.9.5/src/Makefile.am.orig chmsee-0.9.5/src/Makefile.am
--- chmsee-0.9.5/src/Makefile.am.orig	2005-05-20 23:36:15.000000000 +0800
+++ chmsee-0.9.5/src/Makefile.am	2005-05-20 23:37:12.000000000 +0800
@@ -9,11 +9,6 @@ INCLUDES = \
 
 chmsee_CFLAGS =\
 	 -Wall\
-	 -O3	\
-	 -march=i686 \
-	 -mcpu=i686 \
-	 -pipe \
-	 -fomit-frame-pointer \
 	 -DCHM_MT \
 	 -DCHM_USE_PREAD \
 	 -DCHM_USE_IO64 \
@@ -40,12 +35,7 @@ chmsee_SOURCES = \
 		pixmaps.h \
 		main.c
 
-chmsee_LDFLAGS = \
-	-O3	\
-	-march=i686 \
-	-mcpu=i686 \
-	-pipe \
-	-fomit-frame-pointer
+chmsee_LDFLAGS = 
 
 chmsee_LDADD =  \
 	$(GTK_LIBS) \
diff -urNp chmsee-0.9.5/src/Makefile.in.orig chmsee-0.9.5/src/Makefile.in
--- chmsee-0.9.5/src/Makefile.in.orig	2005-05-20 23:37:24.000000000 +0800
+++ chmsee-0.9.5/src/Makefile.in	2005-05-20 23:37:51.000000000 +0800
@@ -211,11 +211,6 @@ INCLUDES = \
 
 chmsee_CFLAGS = \
 	 -Wall\
-	 -O3	\
-	 -march=i686 \
-	 -mcpu=i686 \
-	 -pipe \
-	 -fomit-frame-pointer \
 	 -DCHM_MT \
 	 -DCHM_USE_PREAD \
 	 -DCHM_USE_IO64 \
@@ -240,12 +235,7 @@ chmsee_SOURCES = \
 		pixmaps.h \
 		main.c
 
-chmsee_LDFLAGS = \
-	-O3	\
-	-march=i686 \
-	-mcpu=i686 \
-	-pipe \
-	-fomit-frame-pointer
+chmsee_LDFLAGS = 
 
 chmsee_LDADD = \
 	$(GTK_LIBS) \
