--- gimp-gap-2.2.0/extern_libs/_Makefile.in	2005-10-28 11:39:26.000000000 -0500
+++ gimp-gap-2.2.0/extern_libs/Makefile.in	2005-11-24 13:49:50.000000000 -0500
@@ -451,9 +451,9 @@
 @BUILD_LIBMPEG3_LIB_TRUE@	make
 
 install-data-hook:
-	rm -f $(GAPLIBDIR)/libavformat.a \
-	rm -f $(GAPLIBDIR)/libavcodec.a \
-	rm -f $(GAPLIBDIR)/libmpeg3.a
+	rm -f ${D}$(GAPLIBDIR)/libavformat.a \
+	rm -f ${D}$(GAPLIBDIR)/libavcodec.a \
+	rm -f ${D}$(GAPLIBDIR)/libmpeg3.a
 
 # eof (without this line Makefile generation produces syntax error)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
