From 244e53763bfd3d2a515b88aaa05444ff41832d4d Mon Sep 17 00:00:00 2001
From: microcai <microcai@fedoraproject.org>
Date: Thu, 25 Oct 2012 20:51:36 +0800
Subject: [PATCH] b

---
 frontend/main.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/frontend/main.cpp b/frontend/main.cpp
index 497128b..5af5b96 100644
--- a/frontend/main.cpp
+++ b/frontend/main.cpp
@@ -13,6 +13,8 @@
 
 // This program is a wrapper for `TTF_autohint'.
 
+#include <unistd.h>
+
 #ifdef BUILD_GUI
 #  ifndef _WIN32
 #    define CONSOLE_OUTPUT
-- 
1.7.12.4

