 configure.in |   35 +++++++++++++++++++++++++----------
 1 files changed, 25 insertions(+), 10 deletions(-)

diff --git a/configure.in b/configure.in
index 515f053..03eda8c 100644
--- a/configure.in
+++ b/configure.in
@@ -60,13 +60,6 @@ AC_CHECK_FUNC(XScreenSaverQueryInfo, [
 AC_SUBST(TYPING_BREAK)
 AC_SUBST(SCREENSAVER_LIBS)
 
-PKG_CHECK_MODULES(LIBCANBERRA_GTK, libcanberra-gtk, [
-  CANBERRA_GTK=1
-  AC_SUBST(LIBCANBERRA_GTK_CFLAGS)
-  AC_SUBST(LIBCANBERRA_GTK_LIBS)
-  ], [:])
-AM_CONDITIONAL(HAVE_LIBCANBERRA_GTK, test "x$CANBERRA_GTK" = "x1")
-
 AM_CONDITIONAL(HAVE_TYPING_BREAK, test "x$TYPING_BREAK" = xtyping-break)
 
 dnl keyboard-properties-capplet
@@ -96,6 +89,22 @@ COMMON_MODULES="gtk+-2.0 >= 2.13.1 dnl
  gconf-2.0 dnl
  libglade-2.0 >= 2.0.0 dnl
  gnome-desktop-2.0 >= 2.25.1"
+
+AC_ARG_ENABLE([canberra],
+	AC_HELP_STRING([--enable-canberra],
+	[Enable canberra support @<:@default=yes@:>@]),
+	enable_canberra=$enableval, enable_canberra=yes)
+
+CANBERRA_GTK=0
+if test x$enable_canberra = xyes; then
+   PKG_CHECK_MODULES(LIBCANBERRA_GTK, libcanberra-gtk, 
+   [CANBERRA_GTK=1
+   AC_SUBST(LIBCANBERRA_GTK_CFLAGS)
+   AC_SUBST(LIBCANBERRA_GTK_LIBS)], [])
+fi
+
+AM_CONDITIONAL(HAVE_LIBCANBERRA_GTK, test "x$CANBERRA_GTK" = "x1")
+
 PKG_CHECK_MODULES(LIBSLAB, $COMMON_MODULES libgnomeui-2.0 gnome-desktop-2.0 librsvg-2.0 libgnome-menu pango libgnomeui-2.0)
 PKG_CHECK_MODULES(CAPPLET, $COMMON_MODULES)
 PKG_CHECK_MODULES(GNOMECC, $COMMON_MODULES libgnome-menu >= 2.10.1)
@@ -199,15 +208,21 @@ AC_ARG_ENABLE([aboutme],
    [enable_aboutme=no])
 AC_MSG_RESULT([$enable_aboutme])
 
+AC_ARG_ENABLE([policykit-gnome],
+	AC_HELP_STRING([--enable-policykit-gnome],
+	[Enable PolicyKit-gnome support @<:@default=yes@:>@]),
+	enable_polkit=$enableval, enable_polkit=yes)
+
 if test "x$enable_aboutme" = "xyes"; then
   PKG_CHECK_MODULES(LIBEBOOK, [libebook-1.2 >= 1.7.90],
                     [AC_DEFINE([HAVE_LIBEBOOK], 1,
                     [Define if evolution-data-server libebook-1.2 is available])])
-  dnl PolicyKit-gnome is optional
+  if test "x$enable_polkit" = "xyes"; then
   PKG_CHECK_MODULES(POLKIT_GNOME, [polkit-gnome],
   		    [AC_DEFINE([HAVE_POLKIT], 1,
-  		    [Define if PolicyKit-gnome is available])],
-  		    has_polkit=false)
+  		    [Define if PolicyKit-gnome is available])], 		    
+		    has_polkit=false)
+  fi
 fi
 
 AM_CONDITIONAL(BUILD_ABOUTME, test "x$enable_aboutme" = "xyes")
