--- b/src/html/enmlformatter.cpp
+++ a/src/html/enmlformatter.cpp
@@ -38,8 +38,8 @@
 #include <tidy.h>
 #include <tidybuffio.h>
 #else
-#include <tidy/tidy.h>
-#include <tidy/tidybuffio.h>
+#include <tidy.h>
+#include <tidybuffio.h>
 #endif
 
 #define ENML_MODULE_LOGPREFIX "enml-cleanup: "
--- b/src/main.cpp
+++ a/src/main.cpp
@@ -50,8 +50,8 @@
 #include <tidy.h>
 #include <tidybuffio.h>
 #else
-#include <tidy/tidy.h>
-#include <tidy/tidybuffio.h>
+#include <tidy.h>
+#include <tidybuffio.h>
 #endif
 

