--- panel/candidatepanel.py.orig	2008-08-09 00:01:40.507822433 +0800
+++ panel/candidatepanel.py	2008-08-09 00:01:47.618840593 +0800
@@ -143,8 +143,8 @@
         self.__toplevel.connect("button-release-event", self.__button_release_event_cb)
         self.__toplevel.connect("motion-notify-event", self.__motion_notify_event_cb)
 
-        self.__orientation = gtk.ORIENTATION_HORIZONTAL
         self.__orientation = gtk.ORIENTATION_VERTICAL
+        self.__orientation = gtk.ORIENTATION_HORIZONTAL
         self.__preedit_visible = False
         self.__aux_string_visible = False
         self.__lookup_table_visible = False
