Fixes building documents.
https://bugs.gentoo.org/442904

--- ibus-1.4.2.orig/docs/reference/ibus/ibus-docs.sgml.in
+++ ibus-1.4.2/docs/reference/ibus/ibus-docs.sgml.in
@@ -74,8 +74,10 @@
       <xi:include href="xml/ibusenginedesc.xml"/>
       <xi:include href="xml/ibusfactory.xml"/>
       <xi:include href="xml/ibusinputcontext.xml"/>
+      <xi:include href="xml/ibuskeys.xml"/>
       <xi:include href="xml/ibuskeymap.xml"/>
       <xi:include href="xml/ibuskeysyms.xml"/>
+      <xi:include href="xml/ibusutil.xml"/>
   </chapter>
   <chapter>
       <title>Configuration</title>
@@ -92,9 +94,6 @@
       <title>Communication</title>
       <para>Here lists the service communication functions.</para>
       <xi:include href="xml/ibusbus.xml"/>
-      <xi:include href="xml/ibusconnection.xml"/>
-      <xi:include href="xml/ibusmessage.xml"/>
-      <xi:include href="xml/ibuspendingcall.xml"/>
       <xi:include href="xml/ibusproxy.xml"/>
       <xi:include href="xml/ibusserializable.xml"/>
       <xi:include href="xml/ibusservice.xml"/>
@@ -104,11 +103,18 @@
       <para>Definition and functions for IBus internal use.</para>
       <xi:include href="xml/ibusdebug.xml"/>
       <xi:include href="xml/ibuserror.xml"/>
-      <xi:include href="xml/ibusmainloop.xml"/>
       <xi:include href="xml/ibusobject.xml"/>
-      <xi:include href="xml/ibusserver.xml"/>
       <xi:include href="xml/ibusshare.xml"/>
       <xi:include href="xml/ibustypes.xml"/>
   </chapter>
+  <index id="api-index-full">
+    <title>Index</title>
+    <xi:include href="xml/api-index-full.xml"><xi:fallback /></xi:include>
+  </index>
+  <index id="api-index-deprecated" role="deprecated">
+    <title>Index of deprecated symbols</title>
+    <xi:include href="xml/api-index-deprecated.xml"><xi:fallback /></xi:include>
+  </index>
+  <xi:include href="xml/annotation-glossary.xml"><xi:fallback /></xi:include>
 </book>
 
--- ibus-1.4.2.orig/src/ibusobservedpath.h
+++ ibus-1.4.2/src/ibusobservedpath.h
@@ -75,6 +75,7 @@ struct _IBusObservedPath {
     IBusSerializable parent;
     /* instance members */
 
+    /*< public >*/
     gchar *path;
     glong mtime;
     gboolean is_dir;
--- ibus-1.4.2.orig/src/ibusproperty.h
+++ ibus-1.4.2/src/ibusproperty.h
@@ -124,15 +124,15 @@ typedef struct _IBusPropListClass IBusPr
 
 /**
  * IBusProperty:
- * @key: Unique Identity for the IBusProperty.
- * @icon: Icon file for the IBusProperty.
- * @label: Text shown in UI.
- * @tooltip: Message shown if mouse hovered the  IBusProperty.
- * @sensitive: Whether the IBusProperty is sensitive to keyboard and mouse event.
- * @visible: Whether the IBusProperty is visible.
- * @type: IBusPropType of IBusProperty.
- * @state: IBusPropState of IBusProperty.
- * @sub_props: IBusPropList that contains sub IBusProperties. These IBusProperties are usually
+ * key: Unique Identity for the IBusProperty.
+ * icon: Icon file for the IBusProperty.
+ * label: Text shown in UI.
+ * tooltip: Message shown if mouse hovered the  IBusProperty.
+ * sensitive: Whether the IBusProperty is sensitive to keyboard and mouse event.
+ * visible: Whether the IBusProperty is visible.
+ * type: IBusPropType of IBusProperty.
+ * state: IBusPropState of IBusProperty.
+ * sub_props: IBusPropList that contains sub IBusProperties. These IBusProperties are usually
  * shown as sub menu item.
  *
  * UI component for input method engine property.

