diff --git a/docs/reference/ibus/Makefile.am b/docs/reference/ibus/Makefile.am
index 4fa77a1..75e3815 100644
--- a/docs/reference/ibus/Makefile.am
+++ b/docs/reference/ibus/Makefile.am
@@ -117,7 +117,7 @@ trim-build.stamp: scan-build.stamp
 	$(AM_V_GEN) \
 	    $(SED) -f $(srcdir)/trim.sed -i.bak \
 			$(builddir)/$(DOC_MODULE)-sections.txt && \
-	    $(RM) $(buildir)/$(DOC_MODULE)-sections.txt.bak && \
+	    $(RM) $(builddir)/$(DOC_MODULE)-sections.txt.bak && \
 	    touch trim-build.stamp
 
 tmpl-build.stamp: trim-build.stamp  $(DOC_MODULE)-decl.txt $(SCANOBJ_FILES) $(DOC_MODULE)-overrides.txt
