diff --git a/src/uterm_drm3d_internal.h b/src/uterm_drm3d_internal.h
index 49fcd7e..9a0e4ca 100644
--- a/src/uterm_drm3d_internal.h
+++ b/src/uterm_drm3d_internal.h
@@ -47,6 +47,11 @@
 #include <xf86drmMode.h>
 #include "uterm_video.h"
 
+/* thanks khronos for breaking backwards compatibility.. */
+#if !defined(GL_UNPACK_ROW_LENGTH) && defined(GL_UNPACK_ROW_LENGTH_EXT)
+#  define GL_UNPACK_ROW_LENGTH GL_UNPACK_ROW_LENGTH_EXT
+#endif
+
 struct uterm_drm3d_rb {
 	struct uterm_display *disp;
 	struct gbm_bo *bo;
