--- a/CMakeLists.txt	2026-06-26 18:56:41.501729582 +0800
+++ b/CMakeLists.txt	2026-06-26 19:07:43.675441824 +0800
@@ -5,7 +5,9 @@
         HOMEPAGE_URL "https://github.com/arenekosreal/plasma-ions-china"
         LANGUAGES CXX)
 
-if(NOT PROJECT_VERSION)
+set(PROJECT_VERSION "null" CACHE STRING "plasma-ions-china version")
+
+if(PROJECT_VERSION STREQUAL "null")
     find_package(Git)
     if (Git_FOUND)
         execute_process(COMMAND "${GIT_EXECUTABLE}" describe --tags --always
