--- player_all-bingo-beta-20040615/installer/common/scripts/installer.backup	2004-04-21 09:09:34.000000000 +0900
+++ player_all-bingo-beta-20040615/installer/common/scripts/installer.py	2004-07-11 14:22:35.037901736 +0900
@@ -294,7 +294,7 @@
         else:
             ext = self.archive_type
 
-        dest_arch = "%s%s-%s.%s" % (name, verstr, sysinfo.id, ext)
+        dest_arch = "%s.%s" % ("realplayer", ext)
         project.writeln("copy:")
         project.writeln("\t %s %s %s" %
                         (platform.copy.cmd, self.archive_file,
