--- hxplay-1.0.3/build/umakecf/linux-linker.cf.use-compat	2005-03-03 21:22:01.000000000 -0500
+++ hxplay-1.0.3/build/umakecf/linux-linker.cf	2005-03-03 21:22:10.000000000 -0500
@@ -120,7 +120,7 @@
 
         cmd = "%s%s -o %s %s %s %s %s %s" % (
             cmd, 
-            platform.cc.cmd,
+            platform.cxx.cmd,
             target_path, objects, libpath,
             self.link_paths(),
             string.join(self.extra_arguments),
