--- common/system/pub/platform/unix/pthreadthreads.h_old	2004-11-14 10:54:35.925645344 +0900
+++ common/system/pub/platform/unix/pthreadthreads.h	2004-11-14 10:55:19.110080312 +0900
@@ -102,11 +102,6 @@
 
 #ifndef _MAC_UNIX
 
-struct HXsem_t : public sem_t
-{
-    char padding[64]; /* Flawfinder: ignore */ // different linux versions have different binary reps blechhhh!
-};
-
 //=======================================================================
 //
 //                   HXPthreadSemaphore
@@ -129,7 +124,7 @@
 
 private:
 
-    HXsem_t m_semaphore;
+    sem_t m_semaphore;
     
     //Prevent unintentional ctors
     HXPthreadSemaphore( const HXPthreadSemaphore& ); //copy ctor
