--- HelixPlayer-1.0.beta20040615/build/lib/cvs.py~	2004-05-11 16:55:26.000000000 -0400
+++ HelixPlayer-1.0.beta20040615/build/lib/cvs.py	2004-07-13 12:00:31.301744656 -0400
@@ -594,6 +594,7 @@
              timestamp = None,
              nonrecursive = 0,
              zap = None):
+    return
 
     module_list = listify(module_list)
 
@@ -617,12 +618,15 @@
 
 
 def Status(repository, path, dir = None):
+    return
     return Get(repository).Status(path, dir)
 
 def Commit(repository, path, message, dir = None):
+    return
     return Get(repository).Commit(path, message, dir)
 
 def Tag(repository, path, tag, dir=None):
+    return
     return Get(repository).Tag(path, tag, dir)
 
 def MagicFixDir(dir = "."):
