--- hxplay-1.0.3/audio/fixptutil/pub/math64.h.disable-asm	2004-09-17 14:22:38.000000000 -0400
+++ hxplay-1.0.3/audio/fixptutil/pub/math64.h	2005-03-03 14:56:56.000000000 -0500
@@ -124,7 +124,7 @@
 // GCC / i386
 ///////////////////////////////////////////////////////////////////////////////////////
 
-#if defined(__GNUC__) && defined(__i386__) && !defined(_NO_GNU_AS)
+#if defined(__GNUC__) && defined(__i386__) && !defined(_NO_GNU_AS) && 0
 
 #define HAVE_PLATFORM_MACROS
 
@@ -250,7 +250,7 @@
 #endif
 #endif // defined(__sparc)
 
-#if (defined(__SVR4) && defined(__i386) && (defined(_NO_GNU_AS) || !defined(__GNUC__)) )
+#if (defined(__SVR4) && defined(__i386) && (defined(_NO_GNU_AS) || !defined(__GNUC__))  || 1)
 /* No 64bit, no asm provided in some other file..
  * need normal funcs for sun forte CC + 386 
  * However... forte's inline assembly for MulShift32 is just as good
@@ -414,7 +414,7 @@
 // GCC / PowerPC
 ///////////////////////////////////////////////////////////////////////////////////////
 
-#if defined(__GNUC__) && (defined(__POWERPC__) || defined(__powerpc__))
+#if defined(__GNUC__) && (defined(__POWERPC__) || defined(__powerpc__) && 0)
 
 /*if your compiler can compile 64-bit instructions, and your CPU has them,
  define this. */
