--- hxplay-1.0.0.298/build/lib/sysinfo.py.orig	2004-07-20 02:19:57.000000000 +0100
+++ hxplay-1.0.0.298/build/lib/sysinfo.py	2004-09-21 14:26:08.244116728 +0100
@@ -880,9 +880,9 @@
         else:
             set_platform('linux-2.2-libc6-i586')
 
-    ## Linux PPC Q3
+    ## Fedora PPC
     elif arch == 'ppc':
-        set_platform('linux-2.2-libc6-powerpc')
+        set_platform('linux-2.2-libc6-gcc32-powerpc')
         
     ## Linux/Redhat SPARC
     elif arch == 'sparc64' or arch == 'sparc':
