--- realplay-10.0.4.orig/build/lib/sysinfo.py	2005-04-25 16:38:52.000000000 +0800
+++ realplay-10.0.4/build/lib/sysinfo.py	2005-04-29 10:17:34.000000000 +0800
@@ -969,7 +969,7 @@ def set_platform_linux2():
         gcc_version=os.popen("gcc -dumpversion").read()
         gcc_version=string.strip(gcc_version)
 
-        if gcc_version[0] == "3":
+        if gcc_version[0] == "3" or gcc_version[0] == "4":
             set_platform('linux-2.2-libc6-gcc32-i586')
         else:
             set_platform('linux-2.2-libc6-i586')
