--- realplay-10.0.4.orig/player/app/gtk/realplay	2005-04-25 16:39:03.000000000 +0800
+++ realplay-10.0.4/player/app/gtk/realplay	2005-04-29 21:42:41.000000000 +0800
@@ -67,7 +67,11 @@ if [ -n "$DEBUG" -a -x "$CATCHSEGV" ]; t
 else
     while /bin/true; do
         # Restart the player if exit code is 10
-        $REALPLAYBIN "$@"
+	if which aoss;then
+        	aoss $REALPLAYBIN "$@"
+	else
+		$REALPLAYBIN "$@"
+	fi
         if [ $? -ne 10 ]; then
            break
         fi
