diff -urN linux_fetion-orig/LinuxFetion.desktop linux_fetion/LinuxFetion.desktop
--- linux_fetion-orig/LinuxFetion.desktop	2008-05-04 21:13:52.187401354 +0800
+++ linux_fetion/LinuxFetion.desktop	2008-05-04 21:14:06.878401523 +0800
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Application
 Name=Linux Fetion
-Exec=linux_fetion %u
+Exec=linux-fetion %u
 GenericName=Linux Fetion
 GenericName[zh_CN]=Linux 飞信第三方客户程序
 Comment=Linux Fetion
