From 2f0e599c5600a2497cb9e88fdebc91a0af937582 Mon Sep 17 00:00:00 2001
From: Keeyou <keeyou-cn@outlook.com>
Date: Wed, 22 May 2024 09:58:32 +0800
Subject: [PATCH] cmake: add basic ctest support

---
 CMakeLists.txt | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98942410..4462ddf4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4703,12 +4703,20 @@ if (BUILD_TESTS)
     configure_file("src/core/win32.manifest" "${TEST_MSVC_MANIFEST}" @ONLY)
     target_sources(yass_test PRIVATE ${TEST_MSVC_MANIFEST})
   endif()
+  if (BUILD_TESTS_NO_NETWORK)
+    set(TESTS_FLAGS --no_cares_tests --no_doh_tests --no_dot_tests)
+  endif()
   add_custom_target(check
-    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/yass_test
+    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/yass_test ${TESTS_FLAGS}
     DEPENDS yass_test
     COMMENT "yass unittests"
     USES_TERMINAL
   )
+  enable_testing()
+  add_test(NAME unittests
+    COMMAND ${CMAKE_CURRENT_BINARY_DIR}/yass_test -logtostderr ${TESTS_FLAGS}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+  set_tests_properties(unittests PROPERTIES DEPENDS yass_test)
 
   if (USE_CURL)
     if (MSVC AND MSVC_CRT_LINKAGE STREQUAL "dynamic")
-- 
2.45.1

