From 6a65a9d53cbcc031d96e747578e6a3479ffc2d3e Mon Sep 17 00:00:00 2001
From: Keeyou <keeyou-cn@outlook.com>
Date: Fri, 19 Jul 2024 10:54:45 +0800
Subject: [PATCH] gtk4: no longer blacklist loong arch

---
 CMakeLists.txt | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1c56b38d..87af272b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2097,13 +2097,6 @@ elseif (GUI)
       set(GUI off)
     endif()
 
-    # fix up for loongarch64 gtk support
-    # see https://github.com/microcai/gentoo-zh/pull/4486
-    if (USE_GTK4 AND CMAKE_SYSTEM_PROCESSOR STREQUAL "loongarch64")
-      message(STATUS "Blacklist gtk4 for loongarch64 target")
-      set(USE_GTK4 OFF)
-    endif()
-
     if (GUI AND USE_GTK4)
       pkg_check_modules(GTK4 gtk4)
     endif()
-- 
2.45.2

