diff -uNrbB scim-chewing-0.3.1.orig/src/scim_chewing_imengine.cpp scim-chewing-0.3.1/src/scim_chewing_imengine.cpp
--- scim-chewing-0.3.1.orig/src/scim_chewing_imengine.cpp	2006-05-30 23:06:34.000000000 +0800
+++ scim-chewing-0.3.1/src/scim_chewing_imengine.cpp	2008-08-08 05:21:51.695480377 +0800
@@ -45,6 +45,7 @@
 
 #include <scim.h>
 #include <chewing/chewing.h>
+#include <cstring>
 
 #include "scim_chewing_imengine.h"
 #include "scim_chewing_config_entry.h"
diff -uNrbB scim-chewing-0.3.1.orig/src/scim_chewing_imengine_setup.cpp scim-chewing-0.3.1/src/scim_chewing_imengine_setup.cpp
--- scim-chewing-0.3.1.orig/src/scim_chewing_imengine_setup.cpp	2006-05-30 23:06:34.000000000 +0800
+++ scim-chewing-0.3.1/src/scim_chewing_imengine_setup.cpp	2008-08-08 05:22:18.598249661 +0800
@@ -35,6 +35,8 @@
 
 #include <scim.h>
 #include <gtk/scimkeyselection.h>
+#include <cstring>
+
 #include "scim_chewing_config_entry.h"
 #include "scim_color_button.h"
 
