diff --git a/cmake/trojan.cmake b/cmake/trojan.cmake
index c76b5ee..d102175 100644
--- a/cmake/trojan.cmake
+++ b/cmake/trojan.cmake
@@ -3,7 +3,7 @@ set(ENABLE_MYSQL OFF)
 
 add_definitions(-D_WIN32_WINNT=_WIN32_WINNT_WIN8)
 find_package(Threads REQUIRED)
-find_package(Boost 1.66.0 REQUIRED COMPONENTS system program_options)
+find_package(Boost 1.66.0 REQUIRED COMPONENTS program_options)
 find_package(OpenSSL 1.1.0 REQUIRED)
 
 include_directories(${OPENSSL_INCLUDE_DIR})
