diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/autogen.sh xlyrics-0.4.6.new/autogen.sh
--- xlyrics-0.4.6/autogen.sh	2005-09-04 00:13:00.000000000 +0800
+++ xlyrics-0.4.6.new/autogen.sh	2007-12-02 17:37:25.000000000 +0800
@@ -79,9 +79,10 @@
 	  		test -r $dr/aclocal.m4 && chmod u+w $dr/aclocal.m4
         fi
       fi
+      autopoint -f
       test -d m4 && aclocalinclude="$aclocalinclude -I m4"
       echo "Running aclocal $aclocalinclude ..."
-      aclocal $aclocalinclude
+      aclocal-1.10 $aclocalinclude
       if grep "^AM_CONFIG_HEADER" configure.in >/dev/null; then
 			echo "Running autoheader..."
 			autoheader
@@ -90,6 +91,8 @@
       automake --add-missing --gnu $am_opt
       echo "Running autoconf ..."
       autoconf
+      libtoolize --copy --force
+
     )
   fi
 done
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/config.rpath xlyrics-0.4.6.new/config.rpath
--- xlyrics-0.4.6/config.rpath	1970-01-01 08:00:00.000000000 +0800
+++ xlyrics-0.4.6.new/config.rpath	2007-12-02 17:37:25.000000000 +0800
@@ -0,0 +1,666 @@
+#! /bin/sh
+# Output a system dependent set of variables, describing how to set the
+# run time search path of shared libraries in an executable.
+#
+#   Copyright 1996-2007 Free Software Foundation, Inc.
+#   Taken from GNU libtool, 2001
+#   Originally by Gordon Matzigkeit <gord@gnu.ai.mit.edu>, 1996
+#
+#   This file is free software; the Free Software Foundation gives
+#   unlimited permission to copy and/or distribute it, with or without
+#   modifications, as long as this notice is preserved.
+#
+# The first argument passed to this file is the canonical host specification,
+#    CPU_TYPE-MANUFACTURER-OPERATING_SYSTEM
+# or
+#    CPU_TYPE-MANUFACTURER-KERNEL-OPERATING_SYSTEM
+# The environment variables CC, GCC, LDFLAGS, LD, with_gnu_ld
+# should be set by the caller.
+#
+# The set of defined variables is at the end of this script.
+
+# Known limitations:
+# - On IRIX 6.5 with CC="cc", the run time search patch must not be longer
+#   than 256 bytes, otherwise the compiler driver will dump core. The only
+#   known workaround is to choose shorter directory names for the build
+#   directory and/or the installation directory.
+
+# All known linkers require a `.a' archive for static linking (except MSVC,
+# which needs '.lib').
+libext=a
+shrext=.so
+
+host="$1"
+host_cpu=`echo "$host" | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\1/'`
+host_vendor=`echo "$host" | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\2/'`
+host_os=`echo "$host" | sed 's/^\([^-]*\)-\([^-]*\)-\(.*\)$/\3/'`
+
+# Code taken from libtool.m4's _LT_CC_BASENAME.
+
+for cc_temp in $CC""; do
+  case $cc_temp in
+    compile | *[\\/]compile | ccache | *[\\/]ccache ) ;;
+    distcc | *[\\/]distcc | purify | *[\\/]purify ) ;;
+    \-*) ;;
+    *) break;;
+  esac
+done
+cc_basename=`echo "$cc_temp" | sed -e 's%^.*/%%'`
+
+# Code taken from libtool.m4's AC_LIBTOOL_PROG_COMPILER_PIC.
+
+wl=
+if test "$GCC" = yes; then
+  wl='-Wl,'
+else
+  case "$host_os" in
+    aix*)
+      wl='-Wl,'
+      ;;
+    darwin*)
+      case $cc_basename in
+        xlc*)
+          wl='-Wl,'
+          ;;
+      esac
+      ;;
+    mingw* | cygwin* | pw32* | os2*)
+      ;;
+    hpux9* | hpux10* | hpux11*)
+      wl='-Wl,'
+      ;;
+    irix5* | irix6* | nonstopux*)
+      wl='-Wl,'
+      ;;
+    newsos6)
+      ;;
+    linux* | k*bsd*-gnu)
+      case $cc_basename in
+        icc* | ecc*)
+          wl='-Wl,'
+          ;;
+        pgcc | pgf77 | pgf90)
+          wl='-Wl,'
+          ;;
+        ccc*)
+          wl='-Wl,'
+          ;;
+        como)
+          wl='-lopt='
+          ;;
+        *)
+          case `$CC -V 2>&1 | sed 5q` in
+            *Sun\ C*)
+              wl='-Wl,'
+              ;;
+          esac
+          ;;
+      esac
+      ;;
+    osf3* | osf4* | osf5*)
+      wl='-Wl,'
+      ;;
+    rdos*)
+      ;;
+    solaris*)
+      wl='-Wl,'
+      ;;
+    sunos4*)
+      wl='-Qoption ld '
+      ;;
+    sysv4 | sysv4.2uw2* | sysv4.3*)
+      wl='-Wl,'
+      ;;
+    sysv4*MP*)
+      ;;
+    sysv5* | unixware* | sco3.2v5* | sco5v6* | OpenUNIX*)
+      wl='-Wl,'
+      ;;
+    unicos*)
+      wl='-Wl,'
+      ;;
+    uts4*)
+      ;;
+  esac
+fi
+
+# Code taken from libtool.m4's AC_LIBTOOL_PROG_LD_SHLIBS.
+
+hardcode_libdir_flag_spec=
+hardcode_libdir_separator=
+hardcode_direct=no
+hardcode_minus_L=no
+
+case "$host_os" in
+  cygwin* | mingw* | pw32*)
+    # FIXME: the MSVC++ port hasn't been tested in a loooong time
+    # When not using gcc, we currently assume that we are using
+    # Microsoft Visual C++.
+    if test "$GCC" != yes; then
+      with_gnu_ld=no
+    fi
+    ;;
+  interix*)
+    # we just hope/assume this is gcc and not c89 (= MSVC++)
+    with_gnu_ld=yes
+    ;;
+  openbsd*)
+    with_gnu_ld=no
+    ;;
+esac
+
+ld_shlibs=yes
+if test "$with_gnu_ld" = yes; then
+  # Set some defaults for GNU ld with shared library support. These
+  # are reset later if shared libraries are not supported. Putting them
+  # here allows them to be overridden if necessary.
+  # Unlike libtool, we use -rpath here, not --rpath, since the documented
+  # option of GNU ld is called -rpath, not --rpath.
+  hardcode_libdir_flag_spec='${wl}-rpath ${wl}$libdir'
+  case "$host_os" in
+    aix3* | aix4* | aix5*)
+      # On AIX/PPC, the GNU linker is very broken
+      if test "$host_cpu" != ia64; then
+        ld_shlibs=no
+      fi
+      ;;
+    amigaos*)
+      hardcode_libdir_flag_spec='-L$libdir'
+      hardcode_minus_L=yes
+      # Samuel A. Falvo II <kc5tja@dolphin.openprojects.net> reports
+      # that the semantics of dynamic libraries on AmigaOS, at least up
+      # to version 4, is to share data among multiple programs linked
+      # with the same dynamic library.  Since this doesn't match the
+      # behavior of shared libraries on other platforms, we cannot use
+      # them.
+      ld_shlibs=no
+      ;;
+    beos*)
+      if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
+        :
+      else
+        ld_shlibs=no
+      fi
+      ;;
+    cygwin* | mingw* | pw32*)
+      # hardcode_libdir_flag_spec is actually meaningless, as there is
+      # no search path for DLLs.
+      hardcode_libdir_flag_spec='-L$libdir'
+      if $LD --help 2>&1 | grep 'auto-import' > /dev/null; then
+        :
+      else
+        ld_shlibs=no
+      fi
+      ;;
+    interix[3-9]*)
+      hardcode_direct=no
+      hardcode_libdir_flag_spec='${wl}-rpath,$libdir'
+      ;;
+    gnu* | linux* | k*bsd*-gnu)
+      if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
+        :
+      else
+        ld_shlibs=no
+      fi
+      ;;
+    netbsd*)
+      ;;
+    solaris*)
+      if $LD -v 2>&1 | grep 'BFD 2\.8' > /dev/null; then
+        ld_shlibs=no
+      elif $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
+        :
+      else
+        ld_shlibs=no
+      fi
+      ;;
+    sysv5* | sco3.2v5* | sco5v6* | unixware* | OpenUNIX*)
+      case `$LD -v 2>&1` in
+        *\ [01].* | *\ 2.[0-9].* | *\ 2.1[0-5].*)
+          ld_shlibs=no
+          ;;
+        *)
+          if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
+            hardcode_libdir_flag_spec='`test -z "$SCOABSPATH" && echo ${wl}-rpath,$libdir`'
+          else
+            ld_shlibs=no
+          fi
+          ;;
+      esac
+      ;;
+    sunos4*)
+      hardcode_direct=yes
+      ;;
+    *)
+      if $LD --help 2>&1 | grep ': supported targets:.* elf' > /dev/null; then
+        :
+      else
+        ld_shlibs=no
+      fi
+      ;;
+  esac
+  if test "$ld_shlibs" = no; then
+    hardcode_libdir_flag_spec=
+  fi
+else
+  case "$host_os" in
+    aix3*)
+      # Note: this linker hardcodes the directories in LIBPATH if there
+      # are no directories specified by -L.
+      hardcode_minus_L=yes
+      if test "$GCC" = yes; then
+        # Neither direct hardcoding nor static linking is supported with a
+        # broken collect2.
+        hardcode_direct=unsupported
+      fi
+      ;;
+    aix4* | aix5*)
+      if test "$host_cpu" = ia64; then
+        # On IA64, the linker does run time linking by default, so we don't
+        # have to do anything special.
+        aix_use_runtimelinking=no
+      else
+        aix_use_runtimelinking=no
+        # Test if we are trying to use run time linking or normal
+        # AIX style linking. If -brtl is somewhere in LDFLAGS, we
+        # need to do runtime linking.
+        case $host_os in aix4.[23]|aix4.[23].*|aix5*)
+          for ld_flag in $LDFLAGS; do
+            if (test $ld_flag = "-brtl" || test $ld_flag = "-Wl,-brtl"); then
+              aix_use_runtimelinking=yes
+              break
+            fi
+          done
+          ;;
+        esac
+      fi
+      hardcode_direct=yes
+      hardcode_libdir_separator=':'
+      if test "$GCC" = yes; then
+        case $host_os in aix4.[012]|aix4.[012].*)
+          collect2name=`${CC} -print-prog-name=collect2`
+          if test -f "$collect2name" && \
+            strings "$collect2name" | grep resolve_lib_name >/dev/null
+          then
+            # We have reworked collect2
+            :
+          else
+            # We have old collect2
+            hardcode_direct=unsupported
+            hardcode_minus_L=yes
+            hardcode_libdir_flag_spec='-L$libdir'
+            hardcode_libdir_separator=
+          fi
+          ;;
+        esac
+      fi
+      # Begin _LT_AC_SYS_LIBPATH_AIX.
+      echo 'int main () { return 0; }' > conftest.c
+      ${CC} ${LDFLAGS} conftest.c -o conftest
+      aix_libpath=`dump -H conftest 2>/dev/null | sed -n -e '/Import File Strings/,/^$/ { /^0/ { s/^0  *\(.*\)$/\1/; p; }
+}'`
+      if test -z "$aix_libpath"; then
+        aix_libpath=`dump -HX64 conftest 2>/dev/null | sed -n -e '/Import File Strings/,/^$/ { /^0/ { s/^0  *\(.*\)$/\1/; p; }
+}'`
+      fi
+      if test -z "$aix_libpath"; then
+        aix_libpath="/usr/lib:/lib"
+      fi
+      rm -f conftest.c conftest
+      # End _LT_AC_SYS_LIBPATH_AIX.
+      if test "$aix_use_runtimelinking" = yes; then
+        hardcode_libdir_flag_spec='${wl}-blibpath:$libdir:'"$aix_libpath"
+      else
+        if test "$host_cpu" = ia64; then
+          hardcode_libdir_flag_spec='${wl}-R $libdir:/usr/lib:/lib'
+        else
+          hardcode_libdir_flag_spec='${wl}-blibpath:$libdir:'"$aix_libpath"
+        fi
+      fi
+      ;;
+    amigaos*)
+      hardcode_libdir_flag_spec='-L$libdir'
+      hardcode_minus_L=yes
+      # see comment about different semantics on the GNU ld section
+      ld_shlibs=no
+      ;;
+    bsdi[45]*)
+      ;;
+    cygwin* | mingw* | pw32*)
+      # When not using gcc, we currently assume that we are using
+      # Microsoft Visual C++.
+      # hardcode_libdir_flag_spec is actually meaningless, as there is
+      # no search path for DLLs.
+      hardcode_libdir_flag_spec=' '
+      libext=lib
+      ;;
+    darwin* | rhapsody*)
+      hardcode_direct=no
+      if test "$GCC" = yes ; then
+        :
+      else
+        case $cc_basename in
+          xlc*)
+            ;;
+          *)
+            ld_shlibs=no
+            ;;
+        esac
+      fi
+      ;;
+    dgux*)
+      hardcode_libdir_flag_spec='-L$libdir'
+      ;;
+    freebsd1*)
+      ld_shlibs=no
+      ;;
+    freebsd2.2*)
+      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_direct=yes
+      ;;
+    freebsd2*)
+      hardcode_direct=yes
+      hardcode_minus_L=yes
+      ;;
+    freebsd* | dragonfly*)
+      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_direct=yes
+      ;;
+    hpux9*)
+      hardcode_libdir_flag_spec='${wl}+b ${wl}$libdir'
+      hardcode_libdir_separator=:
+      hardcode_direct=yes
+      # hardcode_minus_L: Not really in the search PATH,
+      # but as the default location of the library.
+      hardcode_minus_L=yes
+      ;;
+    hpux10*)
+      if test "$with_gnu_ld" = no; then
+        hardcode_libdir_flag_spec='${wl}+b ${wl}$libdir'
+        hardcode_libdir_separator=:
+        hardcode_direct=yes
+        # hardcode_minus_L: Not really in the search PATH,
+        # but as the default location of the library.
+        hardcode_minus_L=yes
+      fi
+      ;;
+    hpux11*)
+      if test "$with_gnu_ld" = no; then
+        hardcode_libdir_flag_spec='${wl}+b ${wl}$libdir'
+        hardcode_libdir_separator=:
+        case $host_cpu in
+          hppa*64*|ia64*)
+            hardcode_direct=no
+            ;;
+          *)
+            hardcode_direct=yes
+            # hardcode_minus_L: Not really in the search PATH,
+            # but as the default location of the library.
+            hardcode_minus_L=yes
+            ;;
+        esac
+      fi
+      ;;
+    irix5* | irix6* | nonstopux*)
+      hardcode_libdir_flag_spec='${wl}-rpath ${wl}$libdir'
+      hardcode_libdir_separator=:
+      ;;
+    netbsd*)
+      hardcode_libdir_flag_spec='-R$libdir'
+      hardcode_direct=yes
+      ;;
+    newsos6)
+      hardcode_direct=yes
+      hardcode_libdir_flag_spec='${wl}-rpath ${wl}$libdir'
+      hardcode_libdir_separator=:
+      ;;
+    openbsd*)
+      if test -f /usr/libexec/ld.so; then
+        hardcode_direct=yes
+        if test -z "`echo __ELF__ | $CC -E - | grep __ELF__`" || test "$host_os-$host_cpu" = "openbsd2.8-powerpc"; then
+          hardcode_libdir_flag_spec='${wl}-rpath,$libdir'
+        else
+          case "$host_os" in
+            openbsd[01].* | openbsd2.[0-7] | openbsd2.[0-7].*)
+              hardcode_libdir_flag_spec='-R$libdir'
+              ;;
+            *)
+              hardcode_libdir_flag_spec='${wl}-rpath,$libdir'
+              ;;
+          esac
+        fi
+      else
+        ld_shlibs=no
+      fi
+      ;;
+    os2*)
+      hardcode_libdir_flag_spec='-L$libdir'
+      hardcode_minus_L=yes
+      ;;
+    osf3*)
+      hardcode_libdir_flag_spec='${wl}-rpath ${wl}$libdir'
+      hardcode_libdir_separator=:
+      ;;
+    osf4* | osf5*)
+      if test "$GCC" = yes; then
+        hardcode_libdir_flag_spec='${wl}-rpath ${wl}$libdir'
+      else
+        # Both cc and cxx compiler support -rpath directly
+        hardcode_libdir_flag_spec='-rpath $libdir'
+      fi
+      hardcode_libdir_separator=:
+      ;;
+    solaris*)
+      hardcode_libdir_flag_spec='-R$libdir'
+      ;;
+    sunos4*)
+      hardcode_libdir_flag_spec='-L$libdir'
+      hardcode_direct=yes
+      hardcode_minus_L=yes
+      ;;
+    sysv4)
+      case $host_vendor in
+        sni)
+          hardcode_direct=yes # is this really true???
+          ;;
+        siemens)
+          hardcode_direct=no
+          ;;
+        motorola)
+          hardcode_direct=no #Motorola manual says yes, but my tests say they lie
+          ;;
+      esac
+      ;;
+    sysv4.3*)
+      ;;
+    sysv4*MP*)
+      if test -d /usr/nec; then
+        ld_shlibs=yes
+      fi
+      ;;
+    sysv4*uw2* | sysv5OpenUNIX* | sysv5UnixWare7.[01].[10]* | unixware7* | sco3.2v5.0.[024]*)
+      ;;
+    sysv5* | sco3.2v5* | sco5v6*)
+      hardcode_libdir_flag_spec='`test -z "$SCOABSPATH" && echo ${wl}-R,$libdir`'
+      hardcode_libdir_separator=':'
+      ;;
+    uts4*)
+      hardcode_libdir_flag_spec='-L$libdir'
+      ;;
+    *)
+      ld_shlibs=no
+      ;;
+  esac
+fi
+
+# Check dynamic linker characteristics
+# Code taken from libtool.m4's AC_LIBTOOL_SYS_DYNAMIC_LINKER.
+# Unlike libtool.m4, here we don't care about _all_ names of the library, but
+# only about the one the linker finds when passed -lNAME. This is the last
+# element of library_names_spec in libtool.m4, or possibly two of them if the
+# linker has special search rules.
+library_names_spec=      # the last element of library_names_spec in libtool.m4
+libname_spec='lib$name'
+case "$host_os" in
+  aix3*)
+    library_names_spec='$libname.a'
+    ;;
+  aix4* | aix5*)
+    library_names_spec='$libname$shrext'
+    ;;
+  amigaos*)
+    library_names_spec='$libname.a'
+    ;;
+  beos*)
+    library_names_spec='$libname$shrext'
+    ;;
+  bsdi[45]*)
+    library_names_spec='$libname$shrext'
+    ;;
+  cygwin* | mingw* | pw32*)
+    shrext=.dll
+    library_names_spec='$libname.dll.a $libname.lib'
+    ;;
+  darwin* | rhapsody*)
+    shrext=.dylib
+    library_names_spec='$libname$shrext'
+    ;;
+  dgux*)
+    library_names_spec='$libname$shrext'
+    ;;
+  freebsd1*)
+    ;;
+  freebsd* | dragonfly*)
+    case "$host_os" in
+      freebsd[123]*)
+        library_names_spec='$libname$shrext$versuffix' ;;
+      *)
+        library_names_spec='$libname$shrext' ;;
+    esac
+    ;;
+  gnu*)
+    library_names_spec='$libname$shrext'
+    ;;
+  hpux9* | hpux10* | hpux11*)
+    case $host_cpu in
+      ia64*)
+        shrext=.so
+        ;;
+      hppa*64*)
+        shrext=.sl
+        ;;
+      *)
+        shrext=.sl
+        ;;
+    esac
+    library_names_spec='$libname$shrext'
+    ;;
+  interix[3-9]*)
+    library_names_spec='$libname$shrext'
+    ;;
+  irix5* | irix6* | nonstopux*)
+    library_names_spec='$libname$shrext'
+    case "$host_os" in
+      irix5* | nonstopux*)
+        libsuff= shlibsuff=
+        ;;
+      *)
+        case $LD in
+          *-32|*"-32 "|*-melf32bsmip|*"-melf32bsmip ") libsuff= shlibsuff= ;;
+          *-n32|*"-n32 "|*-melf32bmipn32|*"-melf32bmipn32 ") libsuff=32 shlibsuff=N32 ;;
+          *-64|*"-64 "|*-melf64bmip|*"-melf64bmip ") libsuff=64 shlibsuff=64 ;;
+          *) libsuff= shlibsuff= ;;
+        esac
+        ;;
+    esac
+    ;;
+  linux*oldld* | linux*aout* | linux*coff*)
+    ;;
+  linux* | k*bsd*-gnu)
+    library_names_spec='$libname$shrext'
+    ;;
+  knetbsd*-gnu)
+    library_names_spec='$libname$shrext'
+    ;;
+  netbsd*)
+    library_names_spec='$libname$shrext'
+    ;;
+  newsos6)
+    library_names_spec='$libname$shrext'
+    ;;
+  nto-qnx*)
+    library_names_spec='$libname$shrext'
+    ;;
+  openbsd*)
+    library_names_spec='$libname$shrext$versuffix'
+    ;;
+  os2*)
+    libname_spec='$name'
+    shrext=.dll
+    library_names_spec='$libname.a'
+    ;;
+  osf3* | osf4* | osf5*)
+    library_names_spec='$libname$shrext'
+    ;;
+  rdos*)
+    ;;
+  solaris*)
+    library_names_spec='$libname$shrext'
+    ;;
+  sunos4*)
+    library_names_spec='$libname$shrext$versuffix'
+    ;;
+  sysv4 | sysv4.3*)
+    library_names_spec='$libname$shrext'
+    ;;
+  sysv4*MP*)
+    library_names_spec='$libname$shrext'
+    ;;
+  sysv5* | sco3.2v5* | sco5v6* | unixware* | OpenUNIX* | sysv4*uw2*)
+    library_names_spec='$libname$shrext'
+    ;;
+  uts4*)
+    library_names_spec='$libname$shrext'
+    ;;
+esac
+
+sed_quote_subst='s/\(["`$\\]\)/\\\1/g'
+escaped_wl=`echo "X$wl" | sed -e 's/^X//' -e "$sed_quote_subst"`
+shlibext=`echo "$shrext" | sed -e 's,^\.,,'`
+escaped_libname_spec=`echo "X$libname_spec" | sed -e 's/^X//' -e "$sed_quote_subst"`
+escaped_library_names_spec=`echo "X$library_names_spec" | sed -e 's/^X//' -e "$sed_quote_subst"`
+escaped_hardcode_libdir_flag_spec=`echo "X$hardcode_libdir_flag_spec" | sed -e 's/^X//' -e "$sed_quote_subst"`
+
+LC_ALL=C sed -e 's/^\([a-zA-Z0-9_]*\)=/acl_cv_\1=/' <<EOF
+
+# How to pass a linker flag through the compiler.
+wl="$escaped_wl"
+
+# Static library suffix (normally "a").
+libext="$libext"
+
+# Shared library suffix (normally "so").
+shlibext="$shlibext"
+
+# Format of library name prefix.
+libname_spec="$escaped_libname_spec"
+
+# Library names that the linker finds when passed -lNAME.
+library_names_spec="$escaped_library_names_spec"
+
+# Flag to hardcode \$libdir into a binary during linking.
+# This must work even if \$libdir does not exist.
+hardcode_libdir_flag_spec="$escaped_hardcode_libdir_flag_spec"
+
+# Whether we need a single -rpath flag with a separated argument.
+hardcode_libdir_separator="$hardcode_libdir_separator"
+
+# Set to yes if using DIR/libNAME.so during linking hardcodes DIR into the
+# resulting binary.
+hardcode_direct="$hardcode_direct"
+
+# Set to yes if using the -LDIR flag during linking hardcodes DIR into the
+# resulting binary.
+hardcode_minus_L="$hardcode_minus_L"
+
+EOF
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/configure.in xlyrics-0.4.6.new/configure.in
--- xlyrics-0.4.6/configure.in	2006-05-06 22:38:07.000000000 +0800
+++ xlyrics-0.4.6.new/configure.in	2007-12-02 17:37:25.000000000 +0800
@@ -93,18 +93,11 @@
 PKG_CHECK_MODULES(GTK2, gtk+-2.0 >= 2.4.0 gdk-2.0 >= 2.4.0  glib-2.0 >= 2.4.0, ,exit)
 AC_SUBST(GTK2_CFLAGS)
 AC_SUBST(GTK2_LIBS)
-dnl check xmms
-AM_PATH_XMMS(1.0.0, , AC_MSG_RESULT([*** XMMS >= 1.0.0 not installed - the plugin not included***]))
-AC_SUBST(XMMS_CFLAGS)
-AC_SUBST(XMMS_LIBS)
-AC_SUBST(XMMS_GENERAL_PLUGIN_DIR)
-AM_CONDITIONAL(HAVE_XMMS, test X$XMMS_GENERAL_PLUGIN_DIR != X)
-dnl check bmp
-PKG_CHECK_MODULES(BMP, bmp >= 0.9.7, BMP_GENERAL_PLUGIN_DIR=`pkg-config --variable general_plugin_dir bmp`, echo "*** BMP beep media player not found - the plugin not incluede***")
-AC_SUBST(BMP_LIBS)
-AC_SUBST(BMP_CFLAGS)
-AC_SUBST(BMP_GENERAL_PLUGIN_DIR)
-AM_CONDITIONAL(HAVE_BMP, test X$BMP_GENERAL_PLUGIN_DIR != X)
+dnl check dbus 
+PKG_CHECK_MODULES(DBUS, dbus-glib-1, ,exit)
+AC_SUBST(DBUS_LIBS)
+AC_SUBST(DBUS_CFLAGS)
+AC_SUBST(DBUS_GENERAL_PLUGIN_DIR)
 dnl check audacious
 PKG_CHECK_MODULES(AUDACIOUS, audacious, AUDACIOUS_GENERAL_PLUGIN_DIR=`pkg-config --variable general_plugin_dir audacious`, echo "*** audacious media player not found - the plugin not incluede***")
 AC_SUBST(AUDACIOUS_LIBS)
@@ -112,7 +105,7 @@
 AC_SUBST(AUDACIOUS_GENERAL_PLUGIN_DIR)
 AM_CONDITIONAL(HAVE_AUDACIOUS, test X$AUDACIOUS_GENERAL_PLUGIN_DIR != X)
 dnl check amarok
-AM_CONDITIONAL(HAVE_AMAROK, which amarok)
+AM_CONDITIONAL(HAVE_AMAROK, which amarok >/dev/null 2>&1)
 KDE_PREFIX=`kde-config --prefix`
 QT_LDFLAGS=-L${QTDIR}/lib
 QT_CXXFLAGS=-I${QTDIR}/include
@@ -128,8 +121,6 @@
 AC_OUTPUT(Makefile \
 		src/Makefile \
 		src/plugins/Makefile \
-		src/xmmsplugin/Makefile \
-		src/bmpplugin/Makefile \
 		src/audaciousplugin/Makefile \
 		intl/Makefile \
 		po/Makefile.in )
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/po/Makefile.in.in xlyrics-0.4.6.new/po/Makefile.in.in
--- xlyrics-0.4.6/po/Makefile.in.in	2005-09-04 00:13:00.000000000 +0800
+++ xlyrics-0.4.6.new/po/Makefile.in.in	2007-12-02 17:37:25.000000000 +0800
@@ -29,7 +29,7 @@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in ..*) echo "$(MKINSTALLDIRS)" ;; /*) echo "$(MKINSTALLDIRS)";; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
+mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
 
 CC = @CC@
 GMSGFMT = @GMSGFMT@
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/audaciousplugin/audacious_xlyrics.c xlyrics-0.4.6.new/src/audaciousplugin/audacious_xlyrics.c
--- xlyrics-0.4.6/src/audaciousplugin/audacious_xlyrics.c	2006-05-06 22:38:07.000000000 +0800
+++ xlyrics-0.4.6.new/src/audaciousplugin/audacious_xlyrics.c	2007-12-02 17:37:25.000000000 +0800
@@ -5,9 +5,10 @@
 #include <signal.h>
 
 #include <audacious/plugin.h>
-#include <audacious/beepctrl.h>
+#include <audacious/dbus.h>
+#include <audacious/audctrl.h>
 
-int child;
+static int child;
 void start_plugin(void);
 void cleanup(void);
 
@@ -22,33 +23,35 @@
 	cleanup
 };
 
-	GeneralPlugin *
-get_gplugin_info(void)
+GeneralPlugin * get_gplugin_info(void)
 {
 	return &gp;
 }
 
-	void
-start_plugin(void)
+void start_plugin(void)
 {
-	int i, session;
-
 	child = fork();
 	if(child == 0)
 	{
-		for(i=0; i<10; i++)
-		{
-			usleep( 500000 ); 
-			for( session = 0 ; session < 16 ; session++ )
-				if ( xmms_remote_is_running( session ) )
-					execlp("xlyrics", "xlyrics", "liblyrics_audacious.so", 0);
-		}
+		execlp("xlyrics", "xlyrics", "liblyrics_audacious.so", 0);
+	} else if(child > 0){
+		waitpid(child, NULL, WNOHANG);
 	}
 }
 
-void
-cleanup(void)
+void cleanup(void)
 {
 	if(child != 0)
 		kill(child, SIGKILL);
 }
+
+static GeneralPlugin xlyrics_plugin =
+{
+	.description = "Xlryics",
+	.init = start_plugin,
+	.cleanup = cleanup,
+};
+
+GeneralPlugin *xlyrics_gplist[] = { &xlyrics_plugin, NULL };
+SIMPLE_GENERAL_PLUGIN(xlyrics, xlyrics_gplist);
+
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/audaciousplugin/Makefile.am xlyrics-0.4.6.new/src/audaciousplugin/Makefile.am
--- xlyrics-0.4.6/src/audaciousplugin/Makefile.am	2006-05-06 22:38:07.000000000 +0800
+++ xlyrics-0.4.6.new/src/audaciousplugin/Makefile.am	2007-12-02 17:37:25.000000000 +0800
@@ -6,8 +6,8 @@
 
 # configure the audacious options
 libaudacious_xlyrics_la_SOURCES = audacious_xlyrics.c
-libaudacious_xlyrics_la_LIBADD = ${AUDACIOUS_LIBS}
-libaudacious_xlyrics_la_CFLAGS = ${AUDACIOUS_CFLAGS}
+libaudacious_xlyrics_la_LIBADD = ${AUDACIOUS_LIBS} ${DBUS_LIBS}
+libaudacious_xlyrics_la_CFLAGS = ${AUDACIOUS_CFLAGS} ${DBUS_CFLAGS}
 libaudacious_xlyrics_la_LDFLAGS = -module -avoid-version  -nostdlib
 
 libdir = ${AUDACIOUS_GENERAL_PLUGIN_DIR}
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/bmpplugin/bmp_xlyrics.c xlyrics-0.4.6.new/src/bmpplugin/bmp_xlyrics.c
--- xlyrics-0.4.6/src/bmpplugin/bmp_xlyrics.c	2005-09-04 00:13:00.000000000 +0800
+++ xlyrics-0.4.6.new/src/bmpplugin/bmp_xlyrics.c	1970-01-01 08:00:00.000000000 +0800
@@ -1,53 +0,0 @@
-#include <unistd.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <sys/types.h>
-#include <signal.h>
-
-#include <bmp/plugin.h>
-#include <bmp/beepctrl.h>
-
-int child;
-void start_plugin(void);
-void cleanup(void);
-
-GeneralPlugin gp = {
-	NULL,
-	NULL,
-	-1,
-	"Xlyrics Plugin",
-	start_plugin,
-	NULL,
-	NULL,
-	cleanup
-};
-	GeneralPlugin *
-get_gplugin_info(void)
-{
-	return &gp;
-}
-
-	void
-start_plugin(void)
-{
-	int i, session;
-
-	child = fork();
-	if(child == 0)
-	{
-		for(i=0; i<10; i++)
-		{
-			usleep( 500000 ); 
-			for( session = 0 ; session < 16 ; session++ )
-				if ( xmms_remote_is_running( session ) )
-					execlp("xlyrics", "xlyrics", "liblyrics_beep.so", 0);
-		}
-	}
-}
-
-void
-cleanup(void)
-{
-	if(child != 0)
-		kill(child, SIGKILL);
-}
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/bmpplugin/Makefile.am xlyrics-0.4.6.new/src/bmpplugin/Makefile.am
--- xlyrics-0.4.6/src/bmpplugin/Makefile.am	2005-09-04 00:13:00.000000000 +0800
+++ xlyrics-0.4.6.new/src/bmpplugin/Makefile.am	1970-01-01 08:00:00.000000000 +0800
@@ -1,11 +0,0 @@
-##chose the plugin to complie
-if HAVE_BMP
-lib_LTLIBRARIES  = libbmp_xlyrics.la
-endif
-##configure the bmp options
-libbmp_xlyrics_la_SOURCES = bmp_xlyrics.c
-libbmp_xlyrics_la_LIBADD = ${BMP_LIBS}
-libbmp_xlyrics_la_CFLAGS = ${BMP_CFLAGS}
-libbmp_xlyrics_la_LDFLAGS = -module -avoid-version  -nostdlib
-
-libdir = ${BMP_GENERAL_PLUGIN_DIR}
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/conf.c xlyrics-0.4.6.new/src/conf.c
--- xlyrics-0.4.6/src/conf.c	2005-09-04 00:13:00.000000000 +0800
+++ xlyrics-0.4.6.new/src/conf.c	2007-12-02 17:55:15.000000000 +0800
@@ -12,6 +12,7 @@
  *  GNU General Public License for more details.
  */
 
+#include<string.h>
 #include<gtk/gtk.h>
 #include<gdk/gdk.h>
 #include<glib.h>
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/find.c xlyrics-0.4.6.new/src/find.c
--- xlyrics-0.4.6/src/find.c	2006-05-05 20:08:57.000000000 +0800
+++ xlyrics-0.4.6.new/src/find.c	2007-12-02 17:55:37.000000000 +0800
@@ -12,15 +12,16 @@
  *  GNU General Public License for more details.
  */
 
-#include <sys/types.h>
-#include <sys/stat.h>
-#include <dirent.h>
-#include <stdlib.h>
-#include <glib.h>
-#include <stdio.h>
+#include<string.h>
+#include<sys/types.h>
+#include<sys/stat.h>
+#include<dirent.h>
+#include<stdlib.h>
+#include<glib.h>
+#include<stdio.h>
 
-#include "find.h"
-#include "id3.h"
+#include"find.h"
+#include"id3.h"
 
 /* get the content after " - "*/
 /* the return var must be freed*/
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/id3.c xlyrics-0.4.6.new/src/id3.c
--- xlyrics-0.4.6/src/id3.c	2006-05-05 20:08:57.000000000 +0800
+++ xlyrics-0.4.6.new/src/id3.c	2007-12-02 17:54:36.000000000 +0800
@@ -9,6 +9,7 @@
  */ 
 
 #include "id3.h"
+#include <string.h>
 #include <wchar.h>
 unsigned char *blob;
 size_t filesize;
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/internal.c xlyrics-0.4.6.new/src/internal.c
--- xlyrics-0.4.6/src/internal.c	2005-09-04 00:13:00.000000000 +0800
+++ xlyrics-0.4.6.new/src/internal.c	2007-12-02 17:55:07.000000000 +0800
@@ -12,6 +12,7 @@
  *  GNU General Public License for more details.
  */
 #include <glib.h>
+#include <string.h>
 #include "internal.h"
 /* check the code and convert locale code to utf8 */
 gchar* locale2utf8(const gchar* data)
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/lyrics_download.c xlyrics-0.4.6.new/src/lyrics_download.c
--- xlyrics-0.4.6/src/lyrics_download.c	2005-09-29 13:00:23.000000000 +0800
+++ xlyrics-0.4.6.new/src/lyrics_download.c	2007-12-02 17:54:02.000000000 +0800
@@ -11,14 +11,16 @@
  *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  *  GNU General Public License for more details.
  */
-#include <gtk/gtk.h>
-#include <stdlib.h>
-#include <stdio.h>
-#include <sys/types.h>
-#include <sys/wait.h>
-#include <signal.h>
-#include "lyrics_download.h"
-#include "internal.h"
+#include<unistd.h>
+#include<string.h>
+#include<gtk/gtk.h>
+#include<stdlib.h>
+#include<stdio.h>
+#include<sys/types.h>
+#include<sys/wait.h>
+#include<signal.h>
+#include"lyrics_download.h"
+#include"internal.h"
 
 static GtkListStore *store; /* the lyrics files list store */
 static GtkWidget *tree; /* the tree view of the list store */
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/Makefile.am xlyrics-0.4.6.new/src/Makefile.am
--- xlyrics-0.4.6/src/Makefile.am	2006-05-06 22:38:07.000000000 +0800
+++ xlyrics-0.4.6.new/src/Makefile.am	2007-12-02 17:37:25.000000000 +0800
@@ -1,6 +1,5 @@
-SUBDIRS = plugins xmmsplugin bmpplugin audaciousplugin scripts
-INCLUDES = \
-	$(GTK2_CFLAGS)
+SUBDIRS = plugins audaciousplugin scripts
+INCLUDES = $(GTK2_CFLAGS) $(DBUS_CFLAGS) 
 
 bin_PROGRAMS = xlyrics
 
@@ -24,10 +23,8 @@
 	xlyrics.h \
 	xlyrics.c 
 
-xlyrics_LDADD =  \
-	$(GTK2_LIBS)
+xlyrics_LDADD = $(GTK2_LIBS)
 
 xlyricsmenudir = ${prefix}/share/xlyrics
-xlyricsmenu_DATA = \
-	stylerc
+xlyricsmenu_DATA = stylerc
 EXTRA_DIST = $(xlyricsmenu_DATA)
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/plugins/liblyrics_xmms.la xlyrics-0.4.6.new/src/plugins/liblyrics_xmms.la
--- xlyrics-0.4.6/src/plugins/liblyrics_xmms.la	2005-09-04 00:13:00.000000000 +0800
+++ xlyrics-0.4.6.new/src/plugins/liblyrics_xmms.la	1970-01-01 08:00:00.000000000 +0800
@@ -1,35 +0,0 @@
-# liblyrics_xmms.la - a libtool library file
-# Generated by ltmain.sh - GNU libtool 1.5.2 (1.1220.2.60 2004/01/25 12:25:08)
-#
-# Please DO NOT delete this file!
-# It is necessary for linking the library.
-
-# The name that we can dlopen(3).
-dlname='liblyrics_xmms.so'
-
-# Names of this library.
-library_names='liblyrics_xmms.so liblyrics_xmms.so liblyrics_xmms.so'
-
-# The name of the static archive.
-old_library='liblyrics_xmms.a'
-
-# Libraries that this one depends upon.
-dependency_libs=' -L/usr/lib -L/usr/X11R6/lib /usr/lib/libxmms.la /usr/lib/libgtk.la /usr/lib/libgdk.la /usr/lib/libgmodule.la /usr/lib/libgthread.la /usr/lib/libglib.la -lpthread -ldl -lXi -lXext -lX11 -lm'
-
-# Version information for liblyrics_xmms.
-current=0
-age=0
-revision=0
-
-# Is this an already installed library?
-installed=no
-
-# Should we warn about portability when linking against -modules?
-shouldnotlink=yes
-
-# Files to dlopen/dlpreopen
-dlopen=''
-dlpreopen=''
-
-# Directory that this library needs to be installed in:
-libdir='/usr/lib/xlyrics'
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/plugins/lyrics_audacious.c xlyrics-0.4.6.new/src/plugins/lyrics_audacious.c
--- xlyrics-0.4.6/src/plugins/lyrics_audacious.c	2006-05-06 22:47:05.000000000 +0800
+++ xlyrics-0.4.6.new/src/plugins/lyrics_audacious.c	2007-12-02 17:37:25.000000000 +0800
@@ -2,13 +2,32 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
-#include <audacious/beepctrl.h>  /* provided by audacious */
+#include <audacious/audctrl.h>  /* provided by audacious */
+#include <audacious/dbus.h>  /* provided by audacious */
 
-void set_time(int session, int time)
+DBusGProxy *dbus_proxy = NULL;
+static DBusGConnection *connection = NULL;
+
+static void audtool_connect(void)
+{
+        GError *error = NULL;
+
+	g_type_init();
+        connection = dbus_g_bus_get(DBUS_BUS_SESSION, &error);
+
+        if (connection == NULL)
+                fprintf(stderr, "D-Bus Error: %s", error->message);
+
+        dbus_proxy = dbus_g_proxy_new_for_name(connection, AUDACIOUS_DBUS_SERVICE,
+                                           AUDACIOUS_DBUS_PATH,
+                                           AUDACIOUS_DBUS_INTERFACE);
+}
+
+void set_time(DBusGProxy *session, int time)
 {
   gint ptime = time * 1000;
-  gint pos = xmms_remote_get_playlist_pos( session );
-  gint length = xmms_remote_get_playlist_time( session, pos );
+  gint pos = audacious_remote_get_playlist_pos( session );
+  gint length = audacious_remote_get_playlist_time( session, pos );
 
   /* check bounds */
   if ( ptime < 0 ) {
@@ -17,61 +36,58 @@
 	ptime = length;
   }
   
-  xmms_remote_jump_to_time( session, ptime );
+  audacious_remote_jump_to_time( session, ptime );
 }
 
 /*
- * launch a new xmms and return the session number
+ * launch a new audacious and return the session number
  * exit if error
  */
-int launch(void)
+DBusGProxy *launch(void)
 {
-  gint session;
-  unsigned int tries;
-
-  switch( fork() ) {
+	unsigned int tries;
 
-  case -1:
-    exit(-1);
-
-  case 0:
-    execlp("audacious", "audacious", NULL);
-    exit(1);
-
-  default:
-    for( tries = 0 ; tries < 10 ; tries++ ) {
-      usleep( 500000 ); /* in usec */
-      for( session = 0 ; session < 16 ; session++ )
-		if ( xmms_remote_is_running( session ) )
-		  return session;
-    }
-    exit(-2); /* if no session found, abort */
-  }
+	switch( fork() ) {
+		case -1:
+			exit(-1);
+
+		case 0:
+			execlp("audacious", "audacious", NULL);
+			exit(1);
+
+		default:
+			for( tries = 0 ; tries < 10 ; tries++ ) {
+				usleep( 500000 ); /* in usec */
+				dbus_proxy = dbus_g_proxy_new_for_name(connection, AUDACIOUS_DBUS_SERVICE,
+						AUDACIOUS_DBUS_PATH,
+						AUDACIOUS_DBUS_INTERFACE);
+				if ( audacious_remote_is_running( dbus_proxy ) )
+					return dbus_proxy;
+			}
+			exit(-2); /* if no session found, abort */
+	}
 }
 
-int is_on()
+DBusGProxy *is_on()
 {
-	gint session;
-	for(session=0; session<16; session++)
-	{
-		if(xmms_remote_is_running( session ))
-			return session;
-	}
-	return -1;
+	if(dbus_proxy == NULL)
+		audtool_connect();
+	if(audacious_remote_is_running(dbus_proxy ))
+		return dbus_proxy;
+	return NULL;
 }
 
-int get_time(int session)
+int get_time(DBusGProxy *session)
 {
-	return xmms_remote_get_output_time(session)/1000 ;
+	return audacious_remote_get_output_time(session)/1000 ;
 }
 
-char* get_song(int session)
+char* get_song(DBusGProxy *session)
 {
 	  char *song;
 	  int posi;
 
-	  posi = xmms_remote_get_playlist_pos( session );
-	  song  = xmms_remote_get_playlist_file(session,posi);
+	  posi = audacious_remote_get_playlist_pos( session );
+	  song  = audacious_remote_get_playlist_file(session,posi);
 	  return song;
 }
-
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/plugins/lyrics_beep.c xlyrics-0.4.6.new/src/plugins/lyrics_beep.c
--- xlyrics-0.4.6/src/plugins/lyrics_beep.c	2005-09-28 18:38:35.000000000 +0800
+++ xlyrics-0.4.6.new/src/plugins/lyrics_beep.c	1970-01-01 08:00:00.000000000 +0800
@@ -1,76 +0,0 @@
-#include <stdio.h>
-#include <stdlib.h>
-#include <unistd.h>
-#include <string.h>
-#include <bmp/beepctrl.h>  /* provided by beep */
-
-
-void set_time(int session, int time) {
-  gint ptime = time * 1000;
-  gint pos = xmms_remote_get_playlist_pos( session );
-  gint length = xmms_remote_get_playlist_time( session, pos );
-
-  /* check bounds */
-  if ( ptime < 0 ) {
-    ptime = 0;
-  } else if ( ptime > length ) {
-	ptime = length;
-  }
-  
-  xmms_remote_jump_to_time( session, ptime );
-}
-
-/*
- * launch a new xmms and return the session number
- * exit if error
- */
-int launch(void) {
-  gint session;
-  unsigned int tries;
-
-  switch( fork() ) {
-
-  case -1:
-    exit(-1);
-
-  case 0:
-    execlp("beep-media-player", "beep-media-player", NULL);
-    exit(1);
-
-  default:
-    for( tries = 0 ; tries < 10 ; tries++ ) {
-      usleep( 500000 ); /* in usec */
-      for( session = 0 ; session < 16 ; session++ )
-		if ( xmms_remote_is_running( session ) )
-		  return session;
-    }
-    exit(-2); /* if no session found, abort */
-  }
-}
-
-int is_on()
-{
-	gint session;
-	for(session=0; session<16; session++)
-	{
-		if(xmms_remote_is_running( session ))
-			return session;
-	}
-	return -1;
-}
-
-int get_time(int session)
-{
-	return xmms_remote_get_output_time(session)/1000 ;
-}
-
-char* get_song(int session)
-{
-	  char *song;
-	  int posi;
-
-	  posi = xmms_remote_get_playlist_pos( session );
-	  song  = xmms_remote_get_playlist_file(session,posi);
-	  return song;
-}
-
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/plugins/lyrics_xmms.c xlyrics-0.4.6.new/src/plugins/lyrics_xmms.c
--- xlyrics-0.4.6/src/plugins/lyrics_xmms.c	2005-09-28 18:38:35.000000000 +0800
+++ xlyrics-0.4.6.new/src/plugins/lyrics_xmms.c	1970-01-01 08:00:00.000000000 +0800
@@ -1,73 +0,0 @@
-#include <stdio.h>
-#include <stdlib.h>
-#include <unistd.h>
-#include <string.h>
-#include <xmms/xmmsctrl.h>  /* provided by xmms */
-
-void set_time(int session, int time) {
-  gint ptime = time * 1000;
-  gint pos = xmms_remote_get_playlist_pos( session );
-  gint length = xmms_remote_get_playlist_time( session, pos );
-  
-  /* check bounds */
-  if ( ptime < 0 )
-    ptime = 0;
-  else if ( ptime > length )
-    ptime = length;
-  
-  xmms_remote_jump_to_time( session, ptime );
-}
-
-/*
- * launch a new xmms and return the session number
- * exit if error
- */
-int launch(void) {
-  gint session;
-  unsigned int tries;
-
-  switch( fork() ) {
-
-  case -1:
-    exit(-1);
-
-  case 0:
-    execlp("xmms", "xmms", NULL);
-    exit(1);
-
-  default:
-    for( tries = 0 ; tries < 10 ; tries++ ) {
-      usleep( 500000 ); /* in usec */
-      for( session = 0 ; session < 16 ; session++ )
-		if ( xmms_remote_is_running( session ) )
-		  return session;
-    }
-    exit(-2); /* if no session found, abort */
-  }
-}
-
-int is_on()
-{
-	gint session;
-	for(session=0; session<16; session++)
-	{
-		if(xmms_remote_is_running( session ))
-			return session;
-	}
-	return -1;
-}
-
-int get_time(int session)
-{
-	return xmms_remote_get_output_time(session)/1000 ;
-}
-char* get_song(int session)
-{
-	  char *song;
-	  int posi;
-
-	  posi = xmms_remote_get_playlist_pos( session );
-	  song  = xmms_remote_get_playlist_file(session,posi);
-	  return song;
-}
-
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/plugins/Makefile.am xlyrics-0.4.6.new/src/plugins/Makefile.am
--- xlyrics-0.4.6/src/plugins/Makefile.am	2006-05-06 22:38:08.000000000 +0800
+++ xlyrics-0.4.6.new/src/plugins/Makefile.am	2007-12-02 17:37:25.000000000 +0800
@@ -1,16 +1,6 @@
 ##makefile.am by xiaosuo<xiaosuo@mail.nankai.edu.cn>
 ##chose the plugin to complie
 
-if HAVE_XMMS
-tXMMS = liblyrics_xmms.la
-else
-tXMMS = 
-endif
-if HAVE_BMP
-tBMP = liblyrics_beep.la
-else
-tBMP =
-endif
 if HAVE_AUDACIOUS
 tAUDACIOUS = liblyrics_audacious.la
 else
@@ -22,24 +12,12 @@
 tAMAROK =
 endif
 
-lib_LTLIBRARIES  = ${tXMMS} ${tBMP} ${tAUDACIOUS} ${tAMAROK}
-
-##configure the xmms options
-liblyrics_xmms_la_SOURCES = lyrics_xmms.c
-liblyrics_xmms_la_LIBADD = ${XMMS_LIBS}
-liblyrics_xmms_la_CFLAGS = ${XMMS_CFLAGS}
-liblyrics_xmms_la_LDFLAGS = -module -avoid-version -nostdlib
-
-##configure the beep media player options
-liblyrics_beep_la_SOURCES = lyrics_beep.c
-liblyrics_beep_la_LIBADD = ${BMP_LIBS}
-liblyrics_beep_la_CFLAGS = ${BMP_CFLAGS}
-liblyrics_beep_la_LDFLAGS = -module -avoid-version -nostdlib
+lib_LTLIBRARIES  = ${tAUDACIOUS} ${tAMAROK}
 
 ##configure the audacious media player options
 liblyrics_audacious_la_SOURCES = lyrics_audacious.c
-liblyrics_audacious_la_LIBADD = ${AUDACIOUS_LIBS}
-liblyrics_audacious_la_CFLAGS = ${AUDACIOUS_CFLAGS}
+liblyrics_audacious_la_LIBADD = ${AUDACIOUS_LIBS} ${DBUS_LIBS}
+liblyrics_audacious_la_CFLAGS = ${AUDACIOUS_CFLAGS} ${DBUS_CFLAGS}
 liblyrics_audacious_la_LDFLAGS = -module -avoid-version -nostdlib
 
 ##configure the amarok media player options
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/xlyrics.c xlyrics-0.4.6.new/src/xlyrics.c
--- xlyrics-0.4.6/src/xlyrics.c	2006-05-06 11:25:58.000000000 +0800
+++ xlyrics-0.4.6.new/src/xlyrics.c	2007-12-02 17:53:12.000000000 +0800
@@ -12,14 +12,18 @@
  *  GNU General Public License for more details.
  */
 
+#include<stdlib.h>
+#include<string.h>
 #include<gtk/gtk.h>
 #include<gdk/gdk.h>
 #include<gdk/gdkx.h>
 #include<glib.h>
 #include<gmodule.h>
 #include<locale.h>
-#include <X11/Xlib.h>
-#include <X11/Xatom.h>
+#include<X11/Xlib.h>
+#include<X11/Xatom.h>
+#include<audacious/audctrl.h>  /* provided by audacious */
+#include<audacious/dbus.h>  /* provided by audacious */
 
 #define OPAQUE	0xffffffff
 
@@ -68,13 +72,13 @@
 
 typedef int   (*is_players_on) (void);
 typedef int   (*launch_players) (void);
-typedef char* (*get_players_song) (int session);
-typedef int   (*get_players_time) (int session);
-typedef int   (*set_players_time) (int session, int time);
+typedef char* (*get_players_song) (DBusGProxy *session);
+typedef int   (*get_players_time) (DBusGProxy *session);
+typedef int   (*set_players_time) (DBusGProxy *session, int time);
 
 GModule *module; /*the plugin module*/
 int timer; /*the timer*/
-int session; /*the current player session*/
+DBusGProxy *session; /*the current player session*/
 int is_downloading = 0; /*downloading lyrics*/
 is_players_on is_player_on; /*if player on or not*/
 launch_players launch_player; /*launch a new player session*/
@@ -497,7 +501,7 @@
 	GtkTreeIter iter;
 	gint line_number;
 	struct LyricsLine *line = NULL;
-	int session;
+	DBusGProxy *session;
 
 	if(song && gtk_tree_model_get_iter(GTK_TREE_MODEL(list_store), &iter, path))
 	{
@@ -505,7 +509,7 @@
 		for(line = song->head; line != NULL; line = line->next)
 			if(line->line_number == line_number)
 				break;
-		if((session = is_player_on()) > -1 && line != NULL)
+		if((session = is_player_on()) != NULL && line != NULL)
 			set_player_time(session, line->line_time);
 	}
 }
@@ -716,7 +720,7 @@
 		redraw_list();
 	}
 
-	if((session = is_player_on()) < 0)
+	if((session = is_player_on()) == NULL )
 	{
 		quit(window, "nop");
 		return FALSE;
@@ -822,8 +826,10 @@
 		return 0;
 	}
 
-	if((session = is_player_on()) < 0)
+	if((session = is_player_on()) == NULL ) {
 		session = launch_player();
+	}
+
 
 	gtk_set_locale();
 
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/xmmsplugin/libxmms_xlyrics.la xlyrics-0.4.6.new/src/xmmsplugin/libxmms_xlyrics.la
--- xlyrics-0.4.6/src/xmmsplugin/libxmms_xlyrics.la	2005-09-04 00:13:01.000000000 +0800
+++ xlyrics-0.4.6.new/src/xmmsplugin/libxmms_xlyrics.la	1970-01-01 08:00:00.000000000 +0800
@@ -1,35 +0,0 @@
-# libxmms_xlyrics.la - a libtool library file
-# Generated by ltmain.sh - GNU libtool 1.5.2 (1.1220.2.60 2004/01/25 12:25:08)
-#
-# Please DO NOT delete this file!
-# It is necessary for linking the library.
-
-# The name that we can dlopen(3).
-dlname='libxmms_xlyrics.so'
-
-# Names of this library.
-library_names='libxmms_xlyrics.so libxmms_xlyrics.so libxmms_xlyrics.so'
-
-# The name of the static archive.
-old_library='libxmms_xlyrics.a'
-
-# Libraries that this one depends upon.
-dependency_libs=' -L/usr/lib -L/usr/X11R6/lib /usr/lib/libxmms.la /usr/lib/libgtk.la /usr/lib/libgdk.la /usr/lib/libgmodule.la /usr/lib/libgthread.la /usr/lib/libglib.la -lpthread -ldl -lXi -lXext -lX11 -lm'
-
-# Version information for libxmms_xlyrics.
-current=0
-age=0
-revision=0
-
-# Is this an already installed library?
-installed=no
-
-# Should we warn about portability when linking against -modules?
-shouldnotlink=yes
-
-# Files to dlopen/dlpreopen
-dlopen=''
-dlpreopen=''
-
-# Directory that this library needs to be installed in:
-libdir='/usr/lib/xmms/General'
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/xmmsplugin/Makefile.am xlyrics-0.4.6.new/src/xmmsplugin/Makefile.am
--- xlyrics-0.4.6/src/xmmsplugin/Makefile.am	2005-09-04 00:13:00.000000000 +0800
+++ xlyrics-0.4.6.new/src/xmmsplugin/Makefile.am	1970-01-01 08:00:00.000000000 +0800
@@ -1,11 +0,0 @@
-##chose the plugin to complie
-if HAVE_XMMS
-lib_LTLIBRARIES  = libxmms_xlyrics.la
-endif
-##configure the xmms options
-libxmms_xlyrics_la_SOURCES = xmms_xlyrics.c
-libxmms_xlyrics_la_LIBADD = ${XMMS_LIBS}
-libxmms_xlyrics_la_CFLAGS = ${XMMS_CFLAGS}
-libxmms_xlyrics_la_LDFLAGS = -module -avoid-version  -nostdlib
-
-libdir = ${XMMS_GENERAL_PLUGIN_DIR}
diff -urN -X /home/zhangle/exec/dontdiff xlyrics-0.4.6/src/xmmsplugin/xmms_xlyrics.c xlyrics-0.4.6.new/src/xmmsplugin/xmms_xlyrics.c
--- xlyrics-0.4.6/src/xmmsplugin/xmms_xlyrics.c	2005-09-04 00:13:01.000000000 +0800
+++ xlyrics-0.4.6.new/src/xmmsplugin/xmms_xlyrics.c	1970-01-01 08:00:00.000000000 +0800
@@ -1,53 +0,0 @@
-#include <unistd.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <sys/types.h>
-#include <signal.h>
-
-#include <xmms/plugin.h>
-#include <xmms/xmmsctrl.h>
-
-int child;
-void start_plugin(void);
-void cleanup(void);
-
-GeneralPlugin gp = {
-	NULL,
-	NULL,
-	-1,
-	"Xlyrics Plugin",
-	start_plugin,
-	NULL,
-	NULL,
-	cleanup
-};
-	GeneralPlugin *
-get_gplugin_info(void)
-{
-	return &gp;
-}
-
-	void
-start_plugin(void)
-{
-	int i, session;
-
-	child = fork();
-	if(child == 0)
-	{
-		for(i=0; i<10; i++)
-		{
-			usleep( 500000 ); 
-			for( session = 0 ; session < 16 ; session++ )
-				if ( xmms_remote_is_running( session ) )
-					execlp("xlyrics", "xlyrics", "liblyrics_xmms.so", 0);
-		}
-	}
-}
-
-void
-cleanup(void)
-{
-	if(child != 0)
-		kill(child, SIGKILL);
-}
