--- oxim/configure.orig	2006-06-22 23:23:35.000000000 +0800
+++ oxim/configure	2006-06-22 23:24:01.988607000 +0800
@@ -29632,7 +29632,7 @@
     TABE_DATA="$tabe_data"
 
 
-    for f in db db3; do
+    for f in db db3 db-3; do
         as_ac_Lib=`echo "ac_cv_lib_$f''_db_create" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for db_create in -l$f" >&5
 echo $ECHO_N "checking for db_create in -l$f... $ECHO_C" >&6

