--- oxim/src/qt-immodule/Makefile.orig	2006-09-13 18:15:08.000000000 +0800
+++ oxim/src/qt-immodule/Makefile	2006-09-13 18:15:59.000000000 +0800
@@ -234,7 +234,7 @@
 user_dir = .oxim
 module_LTLIBRARIES = qt-im-oxim.la
 moduledir = $(IM_MODULES_DIR)
-INCLUDES = -I$(top_srcdir)/src/include -I$(QTDIR)/include -I$(QTDIR)/mkspecs/default $(XFT_CFLAGS) -DQT_NO_DEBUG -DQT_PLUGIN -DQT_SHARED -DQT_THREAD_SUPPORT
+INCLUDES = -I$(top_srcdir)/src/include -I$(QTDIR)/include -I$(QTDIR)/mkspecs/default -I$(QTDIR)/mkspecs/linux-g++ $(XFT_CFLAGS) -DQT_NO_DEBUG -DQT_PLUGIN -DQT_SHARED -DQT_THREAD_SUPPORT
 qt_im_oxim_la_SOURCES = moc_qoximinputcontext.cpp \
 			moc_qoximinputcontextplugin.cpp \
 			qoximinputcontext.h \
