diff --color -ur zenmonitor-1.4.2/data/org.pkexec.zenmonitor.policy.in zenmonitor-1.4.2-fix/data/org.pkexec.zenmonitor.policy.in
--- zenmonitor-1.4.2/data/org.pkexec.zenmonitor.policy.in	2020-06-15 05:29:01.000000000 +0800
+++ zenmonitor-1.4.2-fix/data/org.pkexec.zenmonitor.policy.in	2021-12-14 12:14:34.197436633 +0800
@@ -11,7 +11,7 @@
 			<allow_inactive>auth_admin</allow_inactive>
 			<allow_active>auth_admin</allow_active>
 		</defaults>
-		<annotate key="org.freedesktop.policykit.exec.path">@APP_EXEC@</annotate>
+		<annotate key="org.freedesktop.policykit.exec.path">/usr/bin/zenmonitor</annotate>
 		<annotate key="org.freedesktop.policykit.exec.allow_gui">true</annotate>
 	</action>
 
diff --color -ur zenmonitor-1.4.2/data/zenmonitor.desktop.in zenmonitor-1.4.2-fix/data/zenmonitor.desktop.in
--- zenmonitor-1.4.2/data/zenmonitor.desktop.in	2020-06-15 05:29:01.000000000 +0800
+++ zenmonitor-1.4.2-fix/data/zenmonitor.desktop.in	2021-12-14 12:16:08.932245655 +0800
@@ -1,8 +1,9 @@
 [Desktop Entry]
 Name=Zenmonitor
 Comment=Monitoring software for AMD Zen-based CPUs
-Exec=@APP_EXEC@
+Exec=zenmonitor
 Type=Application
 Categories=GTK;System;
 Terminal=false
 Keywords=CPU;AMD;zen;system;core;speed;clock;temperature;voltage;
+Icon=utilities-system-monitor
diff --color -ur zenmonitor-1.4.2/data/zenmonitor-root.desktop.in zenmonitor-1.4.2-fix/data/zenmonitor-root.desktop.in
--- zenmonitor-1.4.2/data/zenmonitor-root.desktop.in	2020-06-15 05:29:01.000000000 +0800
+++ zenmonitor-1.4.2-fix/data/zenmonitor-root.desktop.in	2021-12-14 12:16:02.778323020 +0800
@@ -1,8 +1,9 @@
 [Desktop Entry]
 Name=Zenmonitor (root)
 Comment=Monitoring software for AMD Zen-based CPUs
-Exec=pkexec @APP_EXEC@
+Exec=pkexec zenmonitor
 Type=Application
 Categories=GTK;System;
 Terminal=false
 Keywords=CPU;AMD;zen;system;core;speed;clock;temperature;voltage;
+Icon=utilities-system-monitor
diff --color -ur zenmonitor-1.4.2/makefile zenmonitor-1.4.2-fix/makefile
--- zenmonitor-1.4.2/makefile	2020-06-15 05:29:01.000000000 +0800
+++ zenmonitor-1.4.2-fix/makefile	2021-12-14 12:11:35.143687643 +0800
@@ -3,7 +3,7 @@
 endif
 
 build:
-	cc -Isrc/include `pkg-config --cflags gtk+-3.0` src/*.c src/ss/*.c -o zenmonitor `pkg-config --libs gtk+-3.0` -lm -no-pie -Wall
+	cc -Isrc/include `pkg-config --cflags gtk+-3.0` src/*.c src/ss/*.c -o zenmonitor `pkg-config --libs gtk+-3.0` -lm -no-pie -Wall -fcommon ${CFLAGS} ${LDFLAGS}
 
 install:
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
