---
 data/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/data/meson.build b/data/meson.build
index b9cb54a..cb5f3ff 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -29,7 +29,7 @@ appstream_file = i18n.merge_file(
   output: 'com.gitee.gmg137.NeteaseCloudMusicGtk4.appdata.xml',
   po_dir: '../po',
   install: true,
-  install_dir: join_paths(get_option('datadir'), 'appdata')
+  install_dir: join_paths(get_option('datadir'), 'metainfo')
 )

 appstream_util = find_program('appstream-util', required: false)
