--- psprint/inc/psprint/fontmanager.hxx	2005-09-09 00:33:51.000000000 +0800
+++ psprint/inc/psprint/fontmanager.hxx	2005-09-19 13:59:48.000000000 +0800
@@ -54,6 +54,11 @@
 #include <com/sun/star/lang/Locale.hpp>
 #endif
 
+#ifndef _STRING_HXX
+#include <tools/string.hxx>
+#endif
+
+
 #define ATOM_FAMILYNAME                     2
 #define ATOM_PSNAME                         3
 
@@ -424,6 +429,9 @@ class PrintFontManager
     int getDirectoryAtom( const rtl::OString& rDirectory, bool bCreate = false );
 
     void getServerDirectories(); // get font server directories on e.g. redhat
+    // Add by Firefly(firefly@ossii.com.tw)
+    // 掃描指定目錄下的所有子目錄
+    void getServerDirectories(const ByteString);
 
     /* try to initialize fonts from libfontconfig
 
--- psprint/source/fontmanager/fontmanager.cxx	2005-09-09 00:37:23.000000000 +0800
+++ psprint/source/fontmanager/fontmanager.cxx	2005-09-19 13:59:48.000000000 +0800
@@ -1974,6 +1974,47 @@ bool PrintFontManager::analyzeTrueTypeFi
 }
 
 // -------------------------------------------------------------------------
+// Add by Firefly(firefly@ossii.com.tw)
+// -------------------------------------------------------------------------
+void PrintFontManager::getServerDirectories(const ByteString dirName)
+{
+    DIR  *dir;
+    struct dirent *dirent;
+    struct stat statb;
+
+    if ((dir = opendir(dirName.GetBuffer())) == NULL)
+    {
+        return;
+    }
+
+    bool bDupDir = false;
+    ::std::list< OString >::iterator dir_it;
+    for( dir_it = m_aFontDirectories.begin(); dir_it != m_aFontDirectories.end(); ++dir_it )
+    {
+	if (strcmp((*dir_it).getStr(), dirName.GetBuffer()) == 0)
+	{
+	    bDupDir = true;
+	    break;
+	}
+    }
+
+    if (!bDupDir)
+	m_aFontDirectories.push_back( dirName );
+
+    while ((dirent = readdir(dir)) != NULL)
+    {
+        if (strcmp(dirent->d_name, ".") == 0 ||
+            strcmp(dirent->d_name, "..") == 0 )
+            continue;
+
+        ByteString fullName = dirName + "/" + dirent->d_name;
+        if (stat(fullName.GetBuffer(), &statb) == 0 && S_ISDIR(statb.st_mode) )
+        {
+            getServerDirectories(fullName);
+        }
+    }
+}
+// -------------------------------------------------------------------------
 
 void PrintFontManager::getServerDirectories()
 {
@@ -2164,12 +2205,10 @@ void PrintFontManager::initialize( void*
                 XCloseDisplay( pDisplay );
         }
 
-        // insert some standard directories
-        m_aFontDirectories.push_back( "/usr/openwin/lib/X11/fonts/TrueType" );
-        m_aFontDirectories.push_back( "/usr/openwin/lib/X11/fonts/Type1" );
-        m_aFontDirectories.push_back( "/usr/openwin/lib/X11/fonts/Type1/sun" );
-        m_aFontDirectories.push_back( "/usr/X11R6/lib/X11/fonts/truetype" );
-        m_aFontDirectories.push_back( "/usr/X11R6/lib/X11/fonts/Type1" );
+	// Add by Firefly(firefly@ossii.com.tw)
+	getServerDirectories("/usr/X11R6/lib/X11/fonts");
+	getServerDirectories("/usr/share/fonts");
+	getServerDirectories("/usr/openwin/lib/X11/fonts");
     
 #ifdef SOLARIS
         /* cde specials, from /usr/dt/bin/Xsession: here are the good fonts,
