--- desktop/scripts/soffice.sh	2005-09-09 01:02:48.000000000 +0800
+++ desktop/scripts/soffice.sh	2005-10-07 11:31:59.000000000 +0800
@@ -219,6 +219,14 @@
   PATH="$sd_prog"
 fi
 export PATH
+# Add by Firefly(firefly@ossii.com.tw)
+# 選單及介面字型大小(單位:pt)
+export OOO_INTERFACE_FONT_SIZE=12
+# 其他應用功能字型大小(單位:pt)
+export OOO_DEFAULT_FONT_SIZE=10
+# 螢幕 dpi(預設為 96dpi)
+export OOO_SCREEN_DPI=96
+
 
 # execute soffice binary
 "$sd_prog/$sd_binary" "$@"
