--- vcl/unx/source/app/saldisp.cxx	2005-09-09 21:00:41.000000000 +0800
+++ vcl/unx/source/app/saldisp.cxx	2005-09-19 13:57:09.000000000 +0800
@@ -2794,6 +2794,18 @@ void SalDisplay::PrintInfo() const
 
 void SalDisplay::GetScreenFontResolution( sal_Int32& rDPIX, sal_Int32& rDPIY ) const
 {
+//  Add by Firefly(firefly@firefly.idv.tw)
+    char *pEnv = ::getenv("OOO_SCREEN_DPI");
+    if (pEnv)
+    {
+	int setDPI = atoi(pEnv);
+	if (setDPI > 0)
+	{
+	    rDPIX = rDPIY = setDPI;
+	    return;
+	}
+    }
+//--------------------------------------------------
     rDPIX = aResolution_.A();
     rDPIY = aResolution_.B();
     if( mbExactResolution )
