--- svtools/source/control/ctrltool.cxx	2005-09-08 23:00:34.000000000 +0800
+++ svtools/source/control/ctrltool.cxx	2005-09-19 14:12:18.000000000 +0800
@@ -61,6 +61,21 @@
 
 // =======================================================================
 
+// Add by Firefly(firefly@firefly.idv.tw)
+static bool isEnglishName(XubString& rStr)
+{
+    const char *aName = ByteString(rStr, RTL_TEXTENCODING_UTF8).GetBuffer();
+    ULONG aNameLength = strlen(aName);
+    for (ULONG j=0; j<aNameLength ; j++)
+    {
+	if (aName[j] < 0x20 || aName[j] > 0x7e)
+	{
+	    return false;
+	}
+    }
+    return true;
+}
+
 // Standard Fontgroessen fuer scalierbare Fonts
 static long aStdSizeAry[] =
 {
@@ -192,6 +207,8 @@ ImplFontListNameInfo* FontList::ImplFind
 			*pIndex = LIST_APPEND;
 		return NULL;
 	}
+// Modify by Firefly(firefly@firefly.idv.tw)
+#if 0
 	else
 	{
 		ImplFontListNameInfo* pCmpData = (ImplFontListNameInfo*)List::GetObject( nCount-1 );
@@ -205,12 +222,50 @@ ImplFontListNameInfo* FontList::ImplFind
 		else if ( eComp == COMPARE_EQUAL )
 			return pCmpData;
 	}
+#endif
 
 	// Fonts in der Liste suchen
 	ImplFontListNameInfo*	pCompareData;
 	ImplFontListNameInfo*	pFoundData = NULL;
 	ULONG					nLow = 0;
 	ULONG					nHigh = nCount-1;
+	// Add by Firefly(firefly@firefly.idv.tw)
+	// Find first English name
+	long nFirstEnglish = -1;
+	XubString aSearchName = rSearchName;
+	bool bIsEnglish = isEnglishName(aSearchName);
+	for (ULONG i=0 ; i < nCount ; i++)
+	{
+	    pCompareData = (ImplFontListNameInfo*)List::GetObject(i);
+	    if (isEnglishName(pCompareData->maSearchName))
+	    {
+		nFirstEnglish = i;
+		break;
+	    }
+	}
+	if (!bIsEnglish)
+	{
+	    if (nFirstEnglish == 0)
+	    {
+		if (pIndex)
+		    *pIndex = 0;
+		return NULL;
+	    }
+	    nHigh = nFirstEnglish > 0 ? nFirstEnglish - 1 : 0;
+	}
+	else
+	{
+	    if (nFirstEnglish < 0)
+	    {
+		if (pIndex)
+		{
+		    *pIndex = LIST_APPEND;
+		}
+		return NULL;
+	    }
+	    nLow = nFirstEnglish;
+	}
+	//--------------------------------------------------
 	ULONG					nMid;
 
 	do
@@ -239,7 +294,9 @@ ImplFontListNameInfo* FontList::ImplFind
 
 	if ( pIndex )
 	{
-		eComp = rSearchName.CompareTo( pCompareData->maSearchName );
+		// Remove by Firefly(firefly@firefly.idv.tw)
+		// spilth code
+		//eComp = rSearchName.CompareTo( pCompareData->maSearchName );
 		if ( eComp == COMPARE_GREATER )
 			*pIndex = (nMid+1);
 		else
