--- desktop/source/app/userinstall.cxx	2005-09-09 01:14:01.000000000 +0800
+++ desktop/source/app/userinstall.cxx	2005-10-29 18:32:44.000000000 +0800
@@ -256,6 +256,9 @@
         {
             // copy single file - foldback
             err = File::copy( srcUnqPath,dstUnqPath );
+	    // Add by Firefly(firefly@ossii.com.tw)
+	    // 設為可讀寫，避免從 CD/DVD 複製唯讀屬性
+	    err = File::setAttributes(dstUnqPath, Attribute_OwnRead|Attribute_OwnWrite);
         }
         return err;
     }
