--- vcl/unx/source/app/i18n_ic.cxx	2005-09-09 20:54:58.000000000 +0800
+++ vcl/unx/source/app/i18n_ic.cxx	2005-09-19 13:52:13.000000000 +0800
@@ -536,15 +536,16 @@ SalI18N_InputContext::GetWeightingOfIMSt
     };
 
     StyleWeightingT const *pWeightPtr;
+    // Modify by Firefly(firefly@ossii.com.tw)
     const StyleWeightingT pWeight[] = {
-        { XIMPreeditCallbacks, 0x10000000 },
-        { XIMPreeditPosition,  0x02000000 },
+        { XIMPreeditCallbacks, 0x02000000 },
+        { XIMPreeditPosition,  0x10000000 },
         { XIMPreeditArea,      0x01000000 },
         { XIMPreeditNothing,   0x00100000 },
         { XIMPreeditNone,      0x00010000 },
-        { XIMStatusCallbacks,      0x1000 },
+        { XIMStatusCallbacks,      0x0200 },
         { XIMStatusArea,           0x0100 },
-        { XIMStatusNothing,        0x0010 },
+        { XIMStatusNothing,        0x1000 },
         { XIMStatusNone,           0x0001 },
         { 0, 0x0 }
     };
