--- officecfg/registry/data/org/openoffice/Setup.xcu	2005-09-08 22:46:10.000000000 +0800
+++ officecfg/registry/data/org/openoffice/Setup.xcu	2005-09-19 13:55:57.000000000 +0800
@@ -53,6 +53,12 @@
 		</prop>
 	</node>
 	<node oor:name="Office">
+		<prop oor:name="LicenseAcceptDate" oor:type="xs:string">
+			<value>3005-01-21T19:56:39</value>
+		</prop>
+		<prop oor:name="FirstStartWizardCompleted" oor:type="xs:boolean">
+			<value>true</value>
+		</prop>
 		<prop oor:name="ooSetupInstCompleted">
 			<value>false</value>
 		</prop>
