diff --git a/utils/Makefile.am b/utils/Makefile.am
index 214694e..5bc7e09 100644
--- a/utils/Makefile.am
+++ b/utils/Makefile.am
@@ -51,7 +51,7 @@ libbus_log_la_LDFLAGS += $(WIN32_LD_COMMON)
 libdisasm_la_LDFLAGS += $(WIN32_LD_COMMON) -Xlinker -lopcodes -Xlinker -lopcodes -Xlinker -liberty -Xlinker /mingw/bin/libbfd-2-18-50-20070907.dll
 libos_supp_la_LDFLAGS += $(WIN32_LD_COMMON)
 else
-libdisasm_la_LDFLAGS += -L./$(top_builddir)/third-party/opcodes/.libs/ -L./$(top_builddir)/third-party/bfd/.libs/ -L./$(top_builddir)/third-party/libiberty/pic/ -lopcodes -lbfd -liberty
+libdisasm_la_LDFLAGS += -lopcodes -lbfd -liberty
 endif
 
 pkglib_LTLIBRARIES = libcodecov.la libgdbserver.la libdisasm.la liblog.la libpmon.la libbus_log.la libshutdown.la libos_supp.la libpycli.la
diff --git a/utils/Makefile.in b/utils/Makefile.in
index d49aded..952d65f 100644
--- a/utils/Makefile.in
+++ b/utils/Makefile.in
@@ -317,9 +317,9 @@ libbus_log_la_LDFLAGS = -module
 libshutdown_la_SOURCES = $(utils_shutdown)
 libshutdown_la_LDFLAGS = -module
 libdisasm_la_SOURCES = $(utils_disassembler)
-libdisasm_la_LDFLAGS = -module -L./$(top_builddir)/third-party/opcodes/.libs/ -L./$(top_builddir)/third-party/bfd/.libs/ -L./$(top_builddir)/third-party/libiberty/pic/ -lopcodes -lbfd -liberty
+libdisasm_la_LDFLAGS = -module -lopcodes -lbfd -liberty
 pkglib_LTLIBRARIES = libcodecov.la libgdbserver.la libdisasm.la liblog.la libpmon.la libbus_log.la libshutdown.la
-INCLUDES = -I$(top_srcdir)/third-party/include/ -I$(top_srcdir)/third-party/bfd/ -I$(top_builddir)/third-party/bfd/ -I$(top_srcdir)/common @COMMON_INCLUDES@ 
+INCLUDES = -I$(top_srcdir)/third-party/include/ -I$(top_srcdir)/common @COMMON_INCLUDES@ 
 all: all-am
 
 .SUFFIXES:
