diff --git a/Makefile.am b/Makefile.am
index 33f8fd1..4334ea8 100755
--- a/Makefile.am
+++ b/Makefile.am
@@ -137,16 +137,13 @@ clean_lib:
 	cd $(top_builddir)/third-party/opcodes/ && make distclean
 	cd $(top_builddir)/libltdl/ && make distclean;
 install-data-hook:
-	rm -f -r $(prefix)/testsuite && mkdir $(prefix)/testsuite
 if WIN32
-	mkdir -p $(prefix)/testsuite/arm_hello && mkdir -p $(prefix)/bin
 	cp -a $(top_builddir)/common/.libs/libcommon-0.dll $(prefix)/lib/skyeye/
 	cp -a $(top_builddir)/common/.libs/libcommon-0.dll $(prefix)/bin/
 	cp -a $(top_builddir)/android/objs/emulator_common/.libs/libemulator_common-0.dll $(prefix)/bin/
 	cp -a $(top_builddir)/android/objs/emulator_libui/.libs/libemulator_libui-0.dll $(prefix)/bin/
 endif
-	cp -a $(top_srcdir)/testsuite/arm_hello $(prefix)/testsuite/arm_hello/
-	cp -a $(top_srcdir)/utils/pycli/*.py $(prefix)/bin/
+	cp -a $(top_srcdir)/utils/pycli/*.py ${DESTDIR}/$(prefix)/bin/
 #	rm -f -r $(prefix)/conf && mkdir $(prefix)/conf
 #	cp -a $(top_srcdir)/conf/* $(prefix)/conf
 
