diff --git a/common/Makefile.am b/common/Makefile.am
index 62bac04..85a0b3d 100644
--- a/common/Makefile.am
+++ b/common/Makefile.am
@@ -48,7 +48,7 @@ if WIN32
 libcommon_la_LDFLAGS += -no-undefined -shared -Xlinker /mingw/bin/libbfd-2-18-50-20070907.dll -Xlinker /lib/libltdlc.a -Xlinker -lkernel32 -Xlinker -lmsvcrt -Xlinker -lgw32c -Xlinker -lkernel32 -Xlinker --export-all-symbol
 else
 #libcommon_la_LIBADD += -L$(top_builddir)/third-party/readline/
-libcommon_la_LIBADD += -L$(top_builddir)/third-party/bfd/.libs/ -lbfd -L$(top_builddir)/third-party/libiberty/pic -liberty -L${top_builddir}/libltdl/.libs -lltdl
+libcommon_la_LIBADD += -lbfd -liberty -lltdl
 libcommon_la_LDFLAGS += `python-config --libs`
 endif
 # we will link the common module with main
