diff --git a/arch/ppc/Makefile.am b/arch/ppc/Makefile.am
index 296200e..f715907 100644
--- a/arch/ppc/Makefile.am
+++ b/arch/ppc/Makefile.am
@@ -26,4 +26,4 @@ DEFS += -DLLVM_EXIST
 endif
 
 #noinst_LIBRARIES =      libppc.a
-INCLUDES = -I$(top_srcdir)/common -I$(top_srcdir)/dyncom/ -I$(top_srcdir)/arch/ppc/common -I$(top_srcdir)/arch/ppc/linux_syscall `/usr/local/bin/llvm-config --cflags` -I$(top_srcdir)/utils/debugger @COMMON_INCLUDES@
+INCLUDES = -I$(top_srcdir)/common -I$(top_srcdir)/dyncom/ -I$(top_srcdir)/arch/ppc/common -I$(top_srcdir)/arch/ppc/linux_syscall `llvm-config --cflags` -I$(top_srcdir)/utils/debugger @COMMON_INCLUDES@
