diff --git a/device/flash/skyeye_flash.c b/device/flash/skyeye_flash.c
index 1834ec7..b5175d0 100644
--- a/device/flash/skyeye_flash.c
+++ b/device/flash/skyeye_flash.c
@@ -23,6 +23,7 @@
  * 09/22/2005 	initial version
  *			walimis <wlm@student.dlut.edu.cn>
  */
+#include <stdlib.h>
 
 #include "portable/portable.h"
 #include <sys/time.h>
diff --git a/device/sound/skyeye_sound.c b/device/sound/skyeye_sound.c
index 50163fc..2ceac2d 100644
--- a/device/sound/skyeye_sound.c
+++ b/device/sound/skyeye_sound.c
@@ -21,7 +21,7 @@
 /*
  * 03/25/2007	initial version by Anthony Lee
  */
-
+#include <stdlib.h>
 #include "skyeye_device.h"
 #include "skyeye_options.h"
 #include "skyeye.h"
diff --git a/device/net/skyeye_net.c b/device/net/skyeye_net.c
index 9ff7250..9e06e65 100644
--- a/device/net/skyeye_net.c
+++ b/device/net/skyeye_net.c
@@ -23,7 +23,7 @@
  * 05/01/2005 	initial version
  *			walimis <wlm@student.dlut.edu.cn>
  */
-
+#include <stdlib.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
diff --git a/device/touchscreen/skyeye_touchscreen.c b/device/touchscreen/skyeye_touchscreen.c
index 3d75434..2af9500 100644
--- a/device/touchscreen/skyeye_touchscreen.c
+++ b/device/touchscreen/skyeye_touchscreen.c
@@ -21,7 +21,7 @@
 /*
  * 03/19/2007	initial version by Anthony Lee
  */
-
+#include <stdlib.h>
 #include "skyeye_device.h"
 #include "skyeye_options.h"
 #include "skyeye.h"
