diff --git a/arch/arm/Makefile.am b/arch/arm/Makefile.am
index 3e10ae1..f950a92 100644
--- a/arch/arm/Makefile.am
+++ b/arch/arm/Makefile.am
@@ -29,5 +29,4 @@ endif
 libarm_la_LIBADD = armemu32.lo
 INCLUDES = -I$(top_srcdir)/arch/arm/common -I$(top_srcdir)/arch/arm/mach/ -I$(top_srcdir)/common/dyncom/ `llvm-config --cflags` @COMMON_INCLUDES@ -I$(top_srcdir)/third-party/include/
 armemu32.lo:common/armemu.c armemu.lo
-	$(LIBTOOL) --tag=CC --mode=compile $(CC) -DMODE32 $(DEFS) $(INCLUDES) -g -O2 -MT armemu32.lo -MD -MP -MF .deps/armemu32.Tpo -c -o armemu32.lo `test -f '$(srcdir)/common/armemu.c' || echo `$(srcdir)/common/armemu.c
-#	$(CC) -DMODE32 $(DEFS) $(INCLUDES) common/armemu.c -c -o $@
+	$(LIBTOOL) --mode=compile $(COMPILE) -c -DMODE32 $(DEFS) $(INCLUDES) $(srcdir)/common/armemu.c -o $@
