diff --git a/misc/dde-daemon/gesture.json b/misc/dde-daemon/gesture.json
index 2fc0ce60..9548d8f2 100644
--- a/misc/dde-daemon/gesture.json
+++ b/misc/dde-daemon/gesture.json
@@ -53,15 +53,6 @@
             "Action": "ShowWindow"
         }
     },
-    {
-        "Name": "tap",
-        "Direction": "none",
-        "Fingers": 3,
-        "Action": {
-            "Type": "built-in",
-            "Action": "MoveWindow"
-        }
-    },
     {
         "Name": "swipe",
         "Direction": "up",
diff --git a/system/gesture/core.c b/system/gesture/core.c
index 750baabb..066c4fd9 100644
--- a/system/gesture/core.c
+++ b/system/gesture/core.c
@@ -294,16 +294,16 @@ handle_gesture_events(struct libinput_event *ev, int type)
         }
         raw_event_reset(raw);
         break;
-    case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
-        break;
-    case LIBINPUT_EVENT_GESTURE_TAP_END:
-        if (libinput_event_gesture_get_cancelled(gesture)) {
-			break;
-        }
-        raw->fingers = libinput_event_gesture_get_finger_count(gesture);
-        g_debug("[Tap] fingers: %d", raw->fingers);
-        handleGestureEvent(GESTURE_TYPE_TAP, GESTURE_DIRECTION_NONE, raw->fingers);
-        break;
+    /*case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:*/
+        /*break;*/
+    /*case LIBINPUT_EVENT_GESTURE_TAP_END:*/
+        /*if (libinput_event_gesture_get_cancelled(gesture)) {*/
+			/*break;*/
+        /*}*/
+        /*raw->fingers = libinput_event_gesture_get_finger_count(gesture);*/
+        /*g_debug("[Tap] fingers: %d", raw->fingers);*/
+        /*handleGestureEvent(GESTURE_TYPE_TAP, GESTURE_DIRECTION_NONE, raw->fingers);*/
+        /*break;*/
     }
 }
 
@@ -410,10 +410,10 @@ handle_events(struct libinput *li)
         case LIBINPUT_EVENT_GESTURE_PINCH_END:
         case LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN:
         case LIBINPUT_EVENT_GESTURE_SWIPE_UPDATE:
-        case LIBINPUT_EVENT_GESTURE_SWIPE_END:
-        case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
-        case LIBINPUT_EVENT_GESTURE_TAP_UPDATE:
-        case LIBINPUT_EVENT_GESTURE_TAP_END:{
+        case LIBINPUT_EVENT_GESTURE_SWIPE_END:{
+        /*case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:*/
+        /*case LIBINPUT_EVENT_GESTURE_TAP_UPDATE:*/
+        /*case LIBINPUT_EVENT_GESTURE_TAP_END:{*/
             handle_gesture_events(ev, type);
             break;
         }
