diff -ru gpick_0.2.4_old/source/version/SConscript gpick_0.2.4/source/version/SConscript
--- gpick_0.2.4_old/source/version/SConscript	2011-11-01 16:22:24.175790113 +0800
+++ gpick_0.2.4/source/version/SConscript	2011-11-01 16:22:34.702658509 +0800
@@ -7,7 +7,7 @@
 local_env = env.Clone()
 
 local_env.Append (CPPDEFINES = ['BUILD_DATE="%s"'%(env['GPICK_BUILD_DATE']),
-				'BUILD_REVISION="%s"'%(env['GPICK_BUILD_REVISION']),
+				'BUILD_REVISION="%s"'%(env['GPICK_BUILD_VERSION']),
 				'BUILD_PLATFORM="%s"'%(sys.platform),
 				'BUILD_VERSION="%s"'%(env['GPICK_BUILD_VERSION']),
 ])
