From d113968149fabca90d45ef92a68605112b17c53f Mon Sep 17 00:00:00 2001
From: Weng Xuetian <wengxt@gmail.com>
Date: Wed, 13 Feb 2013 10:11:59 -0500
Subject: [PATCH 1/4] [rime] use load-local for rime

---
 src/fcitx-rime.conf.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/fcitx-rime.conf.in b/src/fcitx-rime.conf.in
index 2ecd1ff..9eaa3e4 100644
--- a/src/fcitx-rime.conf.in
+++ b/src/fcitx-rime.conf.in
@@ -7,3 +7,4 @@ Enabled=True
 Library=fcitx-rime.so
 Type=SharedLibrary
 IMRegisterMethod=ConfigFile
+LoadLocal=True
-- 
1.8.1.5

