From 0eb1fd90d2b8c6135e81acca1fb5066012bbba4d Mon Sep 17 00:00:00 2001
From: Weng Xuetian <wengxt@gmail.com>
Date: Sun, 24 Feb 2013 10:04:31 -0500
Subject: [PATCH 3/4] [rime] fix preedit

---
 src/fcitx-rime.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/fcitx-rime.c b/src/fcitx-rime.c
index 2820d62..b515d89 100644
--- a/src/fcitx-rime.c
+++ b/src/fcitx-rime.c
@@ -270,9 +270,6 @@ INPUT_RETURN_VALUE FcitxRimeGetCandWords(void* arg)
     /* remaining input to convert */
     if (context.composition.sel_end < strlen(context.composition.preedit)) {
         FcitxMessagesAddMessageAtLast(msgPreedit, MSG_CODE, "%s", &context.composition.preedit[context.composition.sel_end]);
-        if (context.commit_text_preview) {
-            FcitxMessagesAddMessageAtLast(msgClientPreedit, MSG_DONOT_COMMIT_WHEN_UNFOCUS, "%s", &context.composition.preedit[context.composition.sel_end]);
-        }
     }
 
     if (context.menu.num_candidates)
-- 
1.8.1.5

