diff --git a/src/fcitx-rime.c b/src/fcitx-rime.c
index f95fdca..e598bb5 100644
--- a/src/fcitx-rime.c
+++ b/src/fcitx-rime.c
@@ -40,7 +40,7 @@ static void* FcitxRimeCreate(FcitxInstance* instance)
     char* shared_data_dir = fcitx_utils_get_fcitx_path_with_filename("pkgdatadir", "rime");
 
     RimeTraits ibus_rime_traits;
-    ibus_rime_traits.shared_data_dir = shared_data_dir;
+    ibus_rime_traits.shared_data_dir = "/usr/share/rime/data";
     ibus_rime_traits.user_data_dir = user_path;
     ibus_rime_traits.distribution_name = "Rime";
     ibus_rime_traits.distribution_code_name = "fcitx-rime";
