diff -Nur hime-0.9.orig/qt4-im/Makefile hime-0.9/qt4-im/Makefile
--- hime-0.9.orig/qt4-im/Makefile	2011-12-08 21:03:50.000000000 +0800
+++ hime-0.9/qt4-im/Makefile	2011-12-14 10:28:40.060082794 +0800
@@ -15,7 +15,7 @@
 	$(CXX) -E $(CFLAGS) $(INCS) $< > $@
 
 moc_hime-qt.cpp:	hime-qt.h
-	/usr/$(LIB)/$(QT)/bin/moc $< -o moc_hime-qt.cpp
+	moc $< -o moc_hime-qt.cpp
 
 im-hime.so: $(OBJS)
 	export LD_RUN_PATH=$(himelibdir) ;\
