diff -Nurw vte-0.22.5.orig/src/vte.c vte-0.22.5/src/vte.c
--- vte-0.22.5.orig/src/vte.c	2009-11-18 01:43:20.000000000 +0800
+++ vte-0.22.5/src/vte.c	2010-01-17 11:00:50.424089248 +0800
@@ -4768,36 +4768,6 @@
 			default:
 				break;
 			}
-			if (terminal->pvt->modifiers & VTE_META_MASK) {
-				steal = TRUE;
-			}
-			switch (keyval) {
-			case GDK_Multi_key:
-			case GDK_Codeinput:
-			case GDK_SingleCandidate:
-			case GDK_MultipleCandidate:
-			case GDK_PreviousCandidate:
-			case GDK_Kanji:
-			case GDK_Muhenkan:
-			case GDK_Henkan:
-			case GDK_Romaji:
-			case GDK_Hiragana:
-			case GDK_Katakana:
-			case GDK_Hiragana_Katakana:
-			case GDK_Zenkaku:
-			case GDK_Hankaku:
-			case GDK_Zenkaku_Hankaku:
-			case GDK_Touroku:
-			case GDK_Massyo:
-			case GDK_Kana_Lock:
-			case GDK_Kana_Shift:
-			case GDK_Eisu_Shift:
-			case GDK_Eisu_toggle:
-				steal = FALSE;
-				break;
-			default:
-				break;
-			}
 		}
 	}
 
