diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -9,6 +9,9 @@ CXXFLAGS += -DUSE_YEARCACHE
 # If the compiler doesn't use namespace, uncomment the following line
 #CXXFLAGS += -DNO_NAMESPACE
 
+# ldflags
+LDFLAGS +=
+
 # Binary installation directory
 BINDIR=/usr/local/bin
 
@@ -43,7 +46,7 @@ CFLAGS += -O2
 CXXFLAGS += -O2
 
 ccal:	ccal.o $(OBJS)
-	$(CXX) $(CXXFLAGS) -o $@ $^ -lm
+	$(CXX) $(LDFLAGS) $(CXXFLAGS) -o $@ $^ -lm
 
 install:	ccal
 	./mkinstalldirs $(BINDIR)
