diff -urN gtkdatabox-0.9.1.1-orig/examples/Makefile.am gtkdatabox-0.9.1.1/examples/Makefile.am
--- gtkdatabox-0.9.1.1-orig/examples/Makefile.am	2010-12-16 09:03:33.604999997 +0800
+++ gtkdatabox-0.9.1.1/examples/Makefile.am	2010-12-16 09:05:16.568000001 +0800
@@ -49,10 +49,6 @@
 AM_CFLAGS			=\
 			-O2\
 			-W -Wall -pedantic\
-			-DG_DISABLE_DEPRECATED\
-			-DGDK_DISABLE_DEPRECATED\
-			-DGDK_PIXBUF_DISABLE_DEPRECATED\
-			-DGTK_DISABLE_DEPRECATED\
 			@LIBGLADE_CFLAGS@ \
 			`pkg-config gtk+-2.0 --cflags`
 
diff -urN gtkdatabox-0.9.1.1-orig/gtk/Makefile.am gtkdatabox-0.9.1.1/gtk/Makefile.am
--- gtkdatabox-0.9.1.1-orig/gtk/Makefile.am	2010-12-16 09:03:33.609999997 +0800
+++ gtkdatabox-0.9.1.1/gtk/Makefile.am	2010-12-16 09:04:34.680999997 +0800
@@ -52,9 +52,5 @@
 AM_CFLAGS			=\
 			-O2\
 			-W -Wall -pedantic\
-			-DG_DISABLE_DEPRECATED\
-			-DGDK_DISABLE_DEPRECATED\
-			-DGDK_PIXBUF_DISABLE_DEPRECATED\
-			-DGTK_DISABLE_DEPRECATED\
 			-DGTK_MULTIHEAD_SAFE=1\
 			`pkg-config gtk+-2.0 --cflags`
