--- oxim/configure.orig	2006-06-22 22:43:14.000000000 +0800
+++ oxim/configure	2006-06-22 23:13:52.776064000 +0800
@@ -28749,7 +28749,7 @@
 
 
     if test "$enable_bimsphone_module" = "yes" ; then
-        for f in db db3; do
+        for f in db db3 db-3; do
             as_ac_Lib=`echo "ac_cv_lib_$f''_db_create" | $as_tr_sh`
 echo "$as_me:$LINENO: checking for db_create in -l$f" >&5
 echo $ECHO_N "checking for db_create in -l$f... $ECHO_C" >&6

