diff --git a/dde-file-manager-lib/shutil/dsqlitehandle.h b/dde-file-manager-lib/shutil/dsqlitehandle.h
index ae7b5e94..ec0eed6e 100644
--- a/dde-file-manager-lib/shutil/dsqlitehandle.h
+++ b/dde-file-manager-lib/shutil/dsqlitehandle.h
@@ -21,7 +21,7 @@
 #include <QtSql/QSqlDatabase>
 
 
-
+#if QT_VERSION < QT_VERSION_CHECK(5,14,0)
 namespace std
 {
 
@@ -35,7 +35,7 @@ struct hash<QString>
 };
 
 }
-
+#endif
 
 
 class DSqliteHandle : public QObject
