diff -ru a/Source/WebKit/gtk/GNUmakefile.am b/Source/WebKit/gtk/GNUmakefile.am
--- a/Source/WebKit/gtk/GNUmakefile.am
+++ b/Source/WebKit/gtk/GNUmakefile.am
@@ -269,7 +269,7 @@
 WEBKIT_GIRSOURCES += WebKit-@WEBKITGTK_API_VERSION@.gir
 
 $(WEBKIT_GIRSOURCES): $(G_IR_SCANNER) $(JSCORE_GIRSOURCES) libdeepin_webkit-@WEBKITGTK_API_MAJOR_VERSION@.@WEBKITGTK_API_MINOR_VERSION@.la
-	$(AM_V_GEN)$(G_IR_SCANNER) -v --warn-all \
+	$(AM_V_GEN)LIBTOOL="$(LIBTOOL)" $(G_IR_SCANNER) -v --warn-all \
 	     --symbol-prefix=webkit \
 	     --identifier-prefix=WebKit \
 	     --namespace=WebKit \
@@ -280,7 +280,7 @@
 	     --include=Soup-2.4 \
 	     --library=webkitgtk-@WEBKITGTK_API_VERSION@ \
 	     --library=javascriptcoredeepin-@WEBKITGTK_API_VERSION@ \
-	     --libtool="$(LIBTOOL)" \
+	     --libtool="bash $(top_srcdir)/gir-paxctl-lt-wrapper m" \
 	     --pkg=gobject-2.0 \
 	     --pkg=gtk+-@GTK_API_VERSION@ \
 	     --pkg=libsoup-2.4 \
