--- patches/OOO_2_0/apply 
+++ patches/OOO_2_0/apply
@@ -742,6 +742,9 @@
 default-system-fpicker.diff
 # Fix for build breakage on PPC
 stlport-ppc-buildfix.diff
+# Fix for broken filters without java i#56619
+nojava-fix-stringparam.diff
+buildfix-new-xslt.diff
 
 #
 # NOTE: 64 bit OOo is work in progress!
--- bin/package-ooo 
+++ bin/package-ooo 
@@ -50,7 +50,7 @@
 	mkdir -p $MANDIR/man1
 	cp -f $OOBUILDDIR/ooo-wrapper$BINSUFFIX $PREFIX/bin
 	chmod +x $PREFIX/bin/ooo-wrapper$BINSUFFIX
-	for app in base calc draw fromtemplate html impress math web writer ffice; do
+	for app in base calc draw fromtemplate impress math web writer ffice; do
 		ln -sf ooo-wrapper${BINSUFFIX} $PREFIX/bin/oo${app}${BINSUFFIX}
 		if test "z$VENDORNAME" = "zNovell" -o "z$VENDORNAME" = "zSUSE" -o "z$VENDORNAME" = "zPLD" \
 		        -o "z$VENDORNAME" = "zDebian"; then
@@ -124,11 +124,11 @@
   	# Menu icons are currently made in debian/rules
 	;;
   *)
-	mkdir -p $OODESTDIR/opt/gnome/share/applications
+	mkdir -p $OODESTDIR/usr/share/applications
 	cd $TOOLSDIR/desktop
 	for source in *.desktop; do
 	    dest=`echo $source | sed "s|.desktop\$||"`
-	    dest="$OODESTDIR/opt/gnome/share/applications/$dest$BINSUFFIX.desktop"
+	    dest="$OODESTDIR/usr/share/applications/$dest$BINSUFFIX.desktop"
 	    add_version=
 	    test "z$VENDORNAME" = "zNovell" -a "z$BINSUFFIX" != "z" && add_version=" ($VERSION)" || :
 	    sed -e "s|@BINSUFFIX@|$BINSUFFIX|g
@@ -215,8 +215,10 @@
 
 echo "Done";
 
-# Java-disabled builds do not create help
-if test -d $OOINSTDIR/help; then
+# Check if the English help is installed and is in the main package (is first on the list)
+# Note that Java-disabled builds do not create help at all.
+if test -f $OOINSTDIR/help/en/sbasic.cfg -a \
+        "`for lang in $OOO_LANGS_LIST ; do echo $lang ; break ; done`" = "en-US" ; then
     echo "Removing duplicated English help..."
   
     for lang in $OOO_LANGS_LIST ; do
