diff --git a/src/core/qcef_dialog_handler.cpp b/src/core/qcef_dialog_handler.cpp
index feaab23..a952296 100644
--- a/src/core/qcef_dialog_handler.cpp
+++ b/src/core/qcef_dialog_handler.cpp
@@ -106,7 +106,7 @@ bool QCefDialogHandler::OnFileDialog(
 
   QFileDialog* dialog = new QFileDialog();
   dialog->setWindowTitle(title_str);
-  dialog->setConfirmOverwrite(mode & FILE_DIALOG_OVERWRITEPROMPT_FLAG);
+  dialog->setOption(QFileDialog::DontConfirmOverwrite, !(mode & FILE_DIALOG_OVERWRITEPROMPT_FLAG));
   const QString path = default_file_path.ToString().c_str();
   QFileInfo info;
   if (path.isEmpty()) {
