--- pidgin-2.4.3/libpurple/protocols/qq/im.h	2008-07-02 08:46:17.000000000 +0800
+++ pidgin-2.4.3.1/libpurple/protocols/qq/im.h	2008-07-15 20:22:57.202246120 +0800
@@ -29,7 +29,7 @@
 #include "connection.h"
 #include "group.h"
 
-#define QQ_MSG_IM_MAX               500	/* max length of IM */
+#define QQ_MSG_IM_MAX               1500	/* max length of IM */
 #define QQ_SEND_IM_BEFORE_MSG_LEN   53
 #define QQ_SEND_IM_AFTER_MSG_LEN    13	/* there is one 0x00 at the end */
 
