diff --git a/src/utils/lib/CryptoPP.cc b/src/utils/lib/CryptoPP.cc
index be0b04a..1b3b2bd 100644
--- a/src/utils/lib/CryptoPP.cc
+++ b/src/utils/lib/CryptoPP.cc
@@ -1700,7 +1700,7 @@ static const char s_RunAtStartup = (AssignIntToInteger = FunctionAssignIntToInte
 template <class T>
 CPP_TYPENAME AllocatorBase<T>::pointer AlignedAllocator<T>::allocate(size_type n, const void *)
 {
-	CheckSize(n);
+	this->CheckSize(n);
 	if (n == 0)
 		return NULL;
 	if (n >= 4)
