diff --git a/config/Makefile.in b/config/Makefile.in
index e10cadf..9659e06 100644
--- a/config/Makefile.in
+++ b/config/Makefile.in
@@ -554,7 +554,7 @@ CORE_SRCS += $(OPENFT_SRCS)
 endif
 
 ifeq ("$(GD)", "yes")
-  GD_LIBS_flags=-cclib "-lgd $(GD_LIBS)" -ccopt "$(GD_LDFLAGS)"
+  GD_LIBS_flags=-cclib "-lgd -lpng $(GD_LIBS)" -ccopt "$(GD_LDFLAGS)"
   ifneq ("$(GD_STATIC_LIBS)", "")
     GD_STATIC_LIBS_opt=-cclib "-lgd $(GD_STATIC_LIBS)" -ccopt "$(GD_LDFLAGS)"
   endif
