diff --git a/src/utilities/Util.hpp b/src/utilities/Util.hpp
index 73f9a3f..d52ec05 100644
--- a/src/utilities/Util.hpp
+++ b/src/utilities/Util.hpp
@@ -3,6 +3,7 @@
 #include "../defines.hpp"
 #include <fstream>
 #include <math.h>
+#include <array>
 
 // For precise colors
 class CFloatingColor {
@@ -62,4 +63,4 @@ CFloatingColor parabolicColor(CFloatingColor from, CFloatingColor to, double inc
 
 double parabolic(double from, double to, double incline);
 
-std::vector<std::string> splitString(std::string, char);
\ No newline at end of file
+std::vector<std::string> splitString(std::string, char);
