diff --git a/Telegram/SourceFiles/pspecific_linux.cpp b/Telegram/SourceFiles/pspecific_linux.cpp
index d0af098..1961d1f 100644
--- a/Telegram/SourceFiles/pspecific_linux.cpp
+++ b/Telegram/SourceFiles/pspecific_linux.cpp
@@ -1163,6 +1163,7 @@ namespace {
 }
 
 void psRegisterCustomScheme() {
+    #if 0
     QString home(_psHomeDir());
     if (home.isEmpty()) return;
 
@@ -1247,6 +1248,7 @@ void psRegisterCustomScheme() {
             LOG(("App Error: Could not open '%1' for write").arg(file));
         }
     }
+    #endif
 }
 
 bool _execUpdater(bool update = true) {
