diff -uNrbB xcin.bak/configure xcin/configure
--- xcin.bak/configure  2003-09-21 01:20:50.000000000 +0800
+++ xcin/configure      2003-09-21 01:21:01.000000000 +0800
@@ -7869,7 +7869,7 @@
 fi
 
 LIBS="$OLD_LIBS"
-for f in db db3; do
+for f in db db3 db-3; do
     echo $ac_n "checking for db_create in -l$f""... $ac_c" 1>&6
 echo "configure:7875: checking for db_create in -l$f" >&5
 ac_lib_var=`echo $f'_'db_create | sed 'y%./+-%__p_%'`

