--- configure	2002-10-10 01:25:18.000000000 +0900
+++ configure.new	2002-10-10 01:27:57.000000000 +0900
@@ -938,9 +938,9 @@
 if test x$with_extra_prefix = x; then
     with_extra_prefix=$prefix
 fi
-if test x$with_extra_inc = x; then
-    with_extra_inc="-I$with_extra_prefix/include"
-fi
+#if test x$with_extra_inc = x; then
+#    with_extra_inc="-I$with_extra_prefix/include"
+#fi
 if test x$with_extra_lib = x; then
     with_extra_lib="$with_extra_prefix/lib"
 fi
