--- a/libs/enigo/Cargo.toml
+++ b/libs/enigo/Cargo.toml
@@ -20,7 +20,7 @@ serde = { version = "1.0", optional = true }
 serde_derive = { version = "1.0", optional = true }
 log = "0.4"
 rdev = { git = "https://github.com/rustdesk-org/rdev" }
-tfc = { git = "https://github.com/rustdesk-org/The-Fat-Controller", branch = "history/rebase_upstream_20240722" }
+tfc = { git = "https://github.com/rustdesk-org/The-Fat-Controller", branch = "history/rebase_upstream_20240722", default-features = false }
 hbb_common = { path = "../hbb_common" }
 
 [features]
