--- srecite.rc	2004-10-17 19:26:26.000000000 +0800
+++ srecite.rc.new	2007-06-08 14:23:33.000000000 +0800
@@ -1,3 +1,5 @@
+gtk_color_scheme = "fg_color:#000\nbg_color:#ede9e3\nbase_color:#fff\ntext_color:#000\nselected_bg_color:#5598d7\nselected_fg_color:#fff"
+
 style "default"
 {
 
@@ -19,10 +21,27 @@
 }
 style "label"
 {
-
-	fg[NORMAL] = {1.0,1.0, 1.0}
-	bg[NORMAL] = {0,0.3,0}
-
+  fg[NORMAL]        = @fg_color #"#000000" # black
+  fg[PRELIGHT]      = @fg_color #"#000000" # black
+  fg[SELECTED]      = @selected_fg_color #"#ffffff" # white 
+  fg[ACTIVE]        = @fg_color #"#000000" # black
+  fg[INSENSITIVE]   = darker (@bg_color) #"#b5b3ac" # dark beige
+  bg[NORMAL]        = @bg_color # "#ede9e3"
+  bg[PRELIGHT]      = shade (1.02, @bg_color) #"#f9f7f3" # very light beige
+  bg[SELECTED]	    = @selected_bg_color # "#5598d7" # deepsky
+  bg[INSENSITIVE]   = @bg_color # "#efebe5" # beige
+  bg[ACTIVE]        = shade (0.9, @bg_color) #"#dcd4c9" #"#d7d3ca" # dark beige
+  base[NORMAL]      = @base_color # "#ffffff" # white 
+  base[PRELIGHT]    = shade (0.95, @bg_color) # "#5f8ec4" # dark beige
+  base[ACTIVE]      = shade (0.9, @selected_bg_color) # "#a69f91" # darker deepsky
+  base[SELECTED]    = @selected_bg_color # "#5598d7" # deepsky
+  base[INSENSITIVE] = @bg_color # "#e8e5de" # beige
+
+  text[NORMAL]      = @text_color # "#000000" # black
+  text[PRELIGHT]    = @text_color # "#000000" # black
+  text[ACTIVE]      = @selected_fg_color # "#ffffff" # white
+  text[SELECTED]    = @selected_fg_color # "#ffffff" # white
+  text[INSENSITIVE] = darker (@bg_color) # "#b5b3ac" # dark beige
 }
 style "toolbar"
 {
