diff --git a/Makefile b/Makefile
index 66e8bea..e94aa3f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,7 @@
 
 CC=gcc
-CFLAG=-O2 -g -Wall -Iinclude
+# CFLAG=-O2 -g -Wall -Iinclude
+CFLAGS+= -Wall -Iinclude
 LIBS= -Llib -lhz
 
 .PHONY: all hzlib clean install-home install
@@ -10,9 +11,9 @@ all: autob5 autogb hzlib
 autob5: autogb
 	rm -f autob5;ln -s autogb autob5
 autogb:autogb.c  hzlib
-	$(CC) $(CFLAG) $(LIBS) autogb.c -lhz -o autogb
+	$(CC) $(CFLAGS) $(LIBS) $(LDFLAGS) autogb.c -lhz -o autogb
 autogb-static:autogb.c  hzlib
-	$(CC) $(CFLAG) autogb.c lib/libhz.a -o autogb
+	$(CC) $(CFLAGS) $(LDFLAGS) autogb.c lib/libhz.a -o autogb
 hzlib:
 	cd hzconvert;make
 xchat-plugins: 
diff --git a/hzconvert/Makefile b/hzconvert/Makefile
index d8d4d19..2809d0a 100644
--- a/hzconvert/Makefile
+++ b/hzconvert/Makefile
@@ -1,5 +1,5 @@
 CC=gcc -I../include
-CFLAGS=-O2 -Wall -g -D_REENTRANT
+CFLAGS+=-Wall -D_REENTRANT
 SHARED_LIB=libhz.so
 SHARED_LIB_MAJOR=$(SHARED_LIB).0
 SHARED_LIB_MINOR=$(SHARED_LIB_MAJOR).0
@@ -15,7 +15,7 @@ STATIC_OBJS=b2g.static.o b2u.static.o g2u.static.o hz2gb.static.o b2g_tables.sta
 all: share-lib static-lib
 
 share-lib: $(OBJS)
-	gcc -fPIC -shared -Wl,-soname,$(SHARED_LIB_MAJOR) -o ../lib/$(SHARED_LIB_MINOR)  $(OBJS) -ldl
+	gcc -fPIC -shared -Wl,-soname,$(SHARED_LIB_MAJOR) $(LDFLAGS) -o ../lib/$(SHARED_LIB_MINOR)  $(OBJS) -ldl
 	rm -f ../lib/$(SHARED_LIB_MAJOR)
 	rm -f ../lib/$(SHARED_LIB)
 	cd ../lib;ln -s $(SHARED_LIB_MINOR) $(SHARED_LIB_MAJOR)
