--- src/Makefile.am	2008-08-27 10:58:59.000000000 +0200
+++ src/Makefile.am.new	2009-09-25 17:58:19.000000000 +0200
@@ -1,4 +1,4 @@
-AM_CPPFLAGS= -Wall -g $(GUILE_CFLAGS) $(GLIB_CFLAGS) $(LOUDMOUTH_CFLAGS) -DDATADIR=\"$(datadir)\" -DLOCALEDIR=\"$(localedir)\"
+AM_CPPFLAGS= $(GUILE_CFLAGS) $(GLIB_CFLAGS) $(LOUDMOUTH_CFLAGS) -DDATADIR=\"$(datadir)\" -DLOCALEDIR=\"$(localedir)\"
 
 bin_PROGRAMS=freetalk
 
@@ -9,6 +9,6 @@
 	roster.h interpreter.h extensions.h primitives.h presence.h \
 	file_transfer.h register.h compat.h
 
-freetalk_LDFLAGS=$(LOUDMOUTH_LIBS) $(GLIB_LIBS) $(GUILE_LDFLAGS)
+LIBS=$(LOUDMOUTH_LIBS) $(GLIB_LIBS) $(GUILE_LIBS) -lguile -lreadline -lncurses
 
 CLEANFILES = *~
