--- xttmgr.c	2004-11-04 02:26:12.725304184 -0800
+++ xttmgr-0.9.6/xttmgr.c	2004-11-04 02:35:42.502684856 -0800
@@ -64,6 +64,7 @@
 /* XFree86 的設定檔 */
 static char *conf3_path    = "/etc/X11/XF86Config";
 static char *conf_path     = "/etc/X11/XF86Config-4";
+static char *confx_path    = "/etc/X11/xorg.conf";
 /* xfs 的字型設定檔 */ 
 static char *fsconfig      = "/etc/X11/fs/config";
 /* 所管理的字型存放位置 */
@@ -1522,8 +1523,11 @@
        int  i;
        int  mode = 0;
        int  have_xtt = 0;
-
-       if ( !File_Exist( conf_path ) ) {
+       /* Adding xorg.conf */
+       if ( File_Exist( confx_path ) ) {
+           strcpy( conf_file, confx_path );
+       }
+       else if ( !File_Exist( conf_path ) ) {
 	  fprintf( stderr, "Warning: Can't find `%s'.\n", conf_path );
 	  fprintf( stderr, "Using config file :`%s'.\n", conf3_path );
 	  strcpy( conf_file, conf3_path );
