--- /common/platform/others.go	2022-04-24 07:38:50.000000000 +0800
+++ /common/platform/others.go	2022-06-04 14:23:21.021117755 +0800
@@ -32,6 +32,7 @@
 		filepath.Join("/usr/local/share/xray/", file),
 		filepath.Join("/usr/share/xray/", file),
 		filepath.Join("/opt/share/xray/", file),
+		filepath.Join("/usr/share/v2ray", file),
 	} {
 		if _, err := os.Stat(p); os.IsNotExist(err) {
 			continue
