--- src/gui/text/qfontengine_ft.cpp	2008-07-25 21:00:30.000000000 +0800
+++ src/gui/text/qfontengine_ft.cpp	2008-07-25 21:57:01.000000000 +0800
@@ -520,6 +520,7 @@
     subpixelType = Subpixel_None;
     defaultGlyphFormat = Format_None;
     canUploadGlyphsToServer = false;
+    embeddedbitmap = true;
 }
 
 QFontEngineFT::~QFontEngineFT()
@@ -649,7 +650,7 @@
 
     }
 #ifndef Q_WS_QWS
-    if (format != Format_Mono)
+    if (format != Format_Mono && !embeddedbitmap)
         load_flags |= FT_LOAD_NO_BITMAP;
 #endif
 
--- src/gui/text/qfontengine_ft_p.h	2008-07-25 21:00:30.000000000 +0800
+++ src/gui/text/qfontengine_ft_p.h	2008-07-25 21:02:39.000000000 +0800
@@ -278,6 +278,8 @@
     SubpixelAntialiasingType subpixelType;
     bool canUploadGlyphsToServer;
 
+    bool embeddedbitmap;
+
 private:
     GlyphFormat defaultGlyphFormat;
     FT_Matrix matrix;

--- src/gui/text/qfontengine_x11.cpp	2008-07-25 21:00:30.000000000 +0800
+++ src/gui/text/qfontengine_x11.cpp	2008-07-25 21:59:26.000000000 +0800
@@ -950,6 +950,14 @@
     }
 #endif
 
+#ifdef FC_EMBEDDED_BITMAP
+    {
+        FcBool b;
+        if (FcPatternGetBool(pattern, FC_EMBEDDED_BITMAP, 0, &b) == FcResultMatch)
+            embeddedbitmap = b;
+    }
+#endif
+
     GlyphFormat defaultFormat = Format_None;
 
 #ifndef QT_NO_XRENDER
