diff -Nur qt-x11-opensource-src-4.4.1.orig/src/gui/text/qfontengine_ft.cpp qt-x11-opensource-src-4.4.1/src/gui/text/qfontengine_ft.cpp
--- qt-x11-opensource-src-4.4.1.orig/src/gui/text/qfontengine_ft.cpp	2008-08-14 13:34:46.152348586 +0800
+++ qt-x11-opensource-src-4.4.1/src/gui/text/qfontengine_ft.cpp	2008-08-15 09:30:13.148348327 +0800
@@ -62,6 +62,7 @@
 #include <ft2build.h>
 #include FT_FREETYPE_H
 #include FT_OUTLINE_H
+#include FT_SYNTHESIS_H
 #include FT_TRUETYPE_TABLES_H
 #include FT_TYPE1_TABLES_H
 #include FT_GLYPH_H
@@ -519,6 +520,7 @@
     cache_cost = 100;
     kerning_pairs_loaded = false;
     transform = false;
+    embolden = false;
     antialias = true;
     default_load_flags = 0;
     subpixelType = Subpixel_None;
@@ -566,10 +568,7 @@
 
     FT_Face face = lockFace();
 
-    //underline metrics
     if (FT_IS_SCALABLE(face)) {
-        line_thickness =  QFixed::fromFixed(FT_MulFix(face->underline_thickness, face->size->metrics.y_scale));
-        underline_position = QFixed::fromFixed(-FT_MulFix(face->underline_position, face->size->metrics.y_scale));
         bool fake_oblique = (fontDef.style != QFont::StyleNormal) && !(face->style_flags & FT_STYLE_FLAG_ITALIC);
         if (fake_oblique)
             matrix.xy = 0x10000*3/10;
@@ -577,6 +576,12 @@
         freetype->matrix = matrix;
         if (fake_oblique)
             transform = true;
+    // fake bold
+        if ((fontDef.weight == QFont::Bold) && !(face->style_flags & FT_STYLE_FLAG_BOLD))
+            embolden = true;
+    // underline metrics
+        line_thickness =  QFixed::fromFixed(FT_MulFix(face->underline_thickness, face->size->metrics.y_scale));
+        underline_position = QFixed::fromFixed(-FT_MulFix(face->underline_position, face->size->metrics.y_scale));
     } else {
         // copied from QFontEngineQPF
         // ad hoc algorithm
@@ -685,6 +690,8 @@
 
     FT_GlyphSlot slot = face->glyph;
 
+    if (embolden) FT_GlyphSlot_Embolden(slot);
+
     int left  = slot->metrics.horiBearingX;
     int right = slot->metrics.horiBearingX + slot->metrics.width;
     int top    = slot->metrics.horiBearingY;
@@ -999,6 +1006,8 @@
     int s = 0;
     if ((fontDef.style != QFont::StyleNormal) && !(freetype->face->style_flags & FT_STYLE_FLAG_ITALIC))
         s = SynthesizedItalic;
+    if ((fontDef.weight == QFont::Bold) && !(freetype->face->style_flags & FT_STYLE_FLAG_BOLD))
+        s |= SynthesizedBold;
     if (fontDef.stretch != 100 && FT_IS_SCALABLE(freetype->face))
         s |= SynthesizedStretch;
     return s;
diff -Nur qt-x11-opensource-src-4.4.1.orig/src/gui/text/qfontengine_ft_p.h qt-x11-opensource-src-4.4.1/src/gui/text/qfontengine_ft_p.h
--- qt-x11-opensource-src-4.4.1.orig/src/gui/text/qfontengine_ft_p.h	2008-08-14 13:34:46.152348586 +0800
+++ qt-x11-opensource-src-4.4.1/src/gui/text/qfontengine_ft_p.h	2008-08-15 09:30:47.838350235 +0800
@@ -277,6 +277,7 @@
     bool antialias;
     bool outline_drawing;
     bool transform;
+    bool embolden;
     SubpixelAntialiasingType subpixelType;
     bool canUploadGlyphsToServer;
 
