diff -Nu text.orig/qfontengine_ft.cpp text/qfontengine_ft.cpp
--- qt-x11-opensource-src-4.5.0.orig/src/gui/text/qfontengine_ft.cpp	2009-03-13 16:57:13.990907230 +0800
+++ qt-x11-opensource-src-4.5.0src/gui/text/qfontengine_ft.cpp	2009-03-13 17:02:58.515907474 +0800
@@ -60,6 +60,7 @@
 #include <ft2build.h>
 #include FT_FREETYPE_H
 #include FT_OUTLINE_H
+#include FT_SYNTHESIS_H
 #include FT_TRUETYPE_TABLES_H
 #include FT_TYPE1_TABLES_H
 #include FT_GLYPH_H
@@ -605,6 +606,7 @@
     cache_cost = 100;
     kerning_pairs_loaded = false;
     transform = false;
+    embolden = false;
     antialias = true;
     default_load_flags = 0;
     default_hint_style = HintNone;
@@ -660,10 +662,7 @@
 
     FT_Face face = lockFace();
 
-    //underline metrics
     if (FT_IS_SCALABLE(face)) {
-        line_thickness =  QFixed::fromFixed(FT_MulFix(face->underline_thickness, face->size->metrics.y_scale));
-        underline_position = QFixed::fromFixed(-FT_MulFix(face->underline_position, face->size->metrics.y_scale));
         bool fake_oblique = (fontDef.style != QFont::StyleNormal) && !(face->style_flags & FT_STYLE_FLAG_ITALIC);
         if (fake_oblique)
             matrix.xy = 0x10000*3/10;
@@ -671,6 +670,13 @@
         freetype->matrix = matrix;
         if (fake_oblique)
             transform = true;
+    // fake bold
+	if ((fontDef.weight == QFont::Bold) && !(face->style_flags & FT_STYLE_FLAG_BOLD))
+  	     embolden = true;
+    // underline metrics
+	line_thickness =  QFixed::fromFixed(FT_MulFix(face->underline_thickness, face->size->metrics.y_scale));
+        underline_position = QFixed::fromFixed(-FT_MulFix(face->underline_position, face->size->metrics.y_scale));
+
     } else {
         // copied from QFontEngineQPF
         // ad hoc algorithm
@@ -770,6 +776,7 @@
     }
 
     FT_GlyphSlot slot = face->glyph;
+    if (embolden) FT_GlyphSlot_Embolden(slot);
     int left  = slot->metrics.horiBearingX;
     int right = slot->metrics.horiBearingX + slot->metrics.width;
     int top    = slot->metrics.horiBearingY;
@@ -1176,6 +1183,8 @@
     int s = 0;
     if ((fontDef.style != QFont::StyleNormal) && !(freetype->face->style_flags & FT_STYLE_FLAG_ITALIC))
         s = SynthesizedItalic;
+    if ((fontDef.weight == QFont::Bold) && !(freetype->face->style_flags & FT_STYLE_FLAG_BOLD))
+	s |= SynthesizedBold;
     if (fontDef.stretch != 100 && FT_IS_SCALABLE(freetype->face))
         s |= SynthesizedStretch;
     return s;
diff -Nu text.orig/qfontengine_ft_p.h text/qfontengine_ft_p.h
--- qt-x11-opensource-src-4.5.0.orig/src/gui/text/qfontengine_ft_p.h	2009-03-13 16:57:13.990907230 +0800
+++ qt-x11-opensource-src-4.5.0/src/gui/text/qfontengine_ft_p.h	2009-03-13 17:04:15.018907506 +0800
@@ -288,6 +288,7 @@
 
     bool antialias;
     bool transform;
+    bool embolden;
     SubpixelAntialiasingType subpixelType;
     int lcdFilterType;
     bool canUploadGlyphsToServer;
