--- src/gui/text/qfontengine_x11.cpp
+++ src/gui/text/qfontengine_x11.cpp
@@ -809,6 +809,7 @@ Q_GUI_EXPORT void qt_x11ft_convert_patte
         *antialias = b;
 }
 
+#include <freetype/config/ftoption.h>
 
 QFontEngineX11FT::QFontEngineX11FT(FcPattern *pattern, const QFontDef &fd, int screen)
     : QFontEngineFT(fd)
@@ -824,6 +825,7 @@ QFontEngineX11FT::QFontEngineX11FT(FcPat
     face_id.index = face_index;
 
     subpixelType = Subpixel_None;
+#ifdef FT_CONFIG_OPTION_SUBPIXEL_RENDERING
     if (antialias) {
         int subpixel = 0;
         if (FcPatternGetInteger(pattern, FC_RGBA, 0, &subpixel) == FcResultNoMatch
@@ -841,6 +843,7 @@ QFontEngineX11FT::QFontEngineX11FT(FcPat
             default: break;
         }
     }
+#endif
 
 #ifdef FC_HINT_STYLE
     {
