diff -Nur qt-x11-opensource-src-4.4.1.orig/src/gui/text/qfontengine_x11.cpp qt-x11-opensource-src-4.4.1/src/gui/text/qfontengine_x11.cpp
--- qt-x11-opensource-src-4.4.1.orig/src/gui/text/qfontengine_x11.cpp	2008-08-14 13:34:46.160350812 +0800
+++ qt-x11-opensource-src-4.4.1/src/gui/text/qfontengine_x11.cpp	2008-08-15 10:20:21.922346239 +0800
@@ -890,6 +890,7 @@
         *antialias = b;
 }
 
+#include <freetype/config/ftoption.h>
 
 QFontEngineX11FT::QFontEngineX11FT(FcPattern *pattern, const QFontDef &fd, int screen)
     : QFontEngineFT(fd)
@@ -907,6 +908,7 @@
     canUploadGlyphsToServer = qApp->thread() == QThread::currentThread();
 
     subpixelType = Subpixel_None;
+#ifdef FT_CONFIG_OPTION_SUBPIXEL_RENDERING
     if (antialias) {
         int subpixel = 0;
         if (FcPatternGetInteger(pattern, FC_RGBA, 0, &subpixel) == FcResultNoMatch
@@ -924,6 +926,7 @@
             default: break;
         }
     }
+#endif
 
 #ifdef FC_HINT_STYLE
     {
