diff -Naur openfetion-2.1.0/src/fx_about.c openfetion-2.1.0.new/src/fx_about.c
--- openfetion-2.1.0/src/fx_about.c	2010-12-26 12:56:55.000000000 +0800
+++ openfetion-2.1.0.new/src/fx_about.c	2011-01-04 01:49:56.699000055 +0800
@@ -137,7 +137,7 @@
 	GtkTextTag    *tag;
 	GtkTextTag    *tag1;
 	GtkTextIter    iter;
-	const char title[] = N_("Openfetion "FETION_VERSION"\n\n");
+	const char title[] = N_("Openfetion 2.1.0\n\n");
 	const char intro[] = N_("OpenFetion is a Fetion client for linux based on GTK+2.0, "
 				"implementing Fetion Protocol Version 4.\n"
 				"It supports most features of Fetion. \n"
