diff -urN openfetion-all-2.1.0.old/config-ofetion.h.cmake openfetion-all-2.1.0/config-ofetion.h.cmake
--- openfetion-all-2.1.0.old/config-ofetion.h.cmake	2010-12-26 12:56:55.000000000 +0800
+++ openfetion-all-2.1.0/config-ofetion.h.cmake	2011-01-01 20:14:23.076114002 +0800
@@ -1,3 +1,4 @@
+#cmakedefine PACKAGE_VERSION "@PACKAGE_VERSION@"
 #cmakedefine USE_LIBNOTIFY 1
 #cmakedefine USE_GSTREAMER 1
 #cmakedefine USE_LIBXSS 1
diff -urN openfetion-all-2.1.0.old/lib/fetion_types.h openfetion-all-2.1.0/lib/fetion_types.h
--- openfetion-all-2.1.0.old/lib/fetion_types.h	2010-12-26 12:56:55.000000000 +0800
+++ openfetion-all-2.1.0/lib/fetion_types.h	2011-01-01 19:58:37.894114002 +0800
@@ -32,7 +32,6 @@
 #endif
 
 #define FETION_NAME "OpenFetion"
-#define FETION_VERSION "2.0.7"
 #define PROTO_VERSION "4.0.2510"
 #define NAVIGATION_URI "nav.fetion.com.cn"
 #define PGGROUP_SPACE_URI "http://group.feixin.10086.cn/space/Index/"
diff -urN openfetion-all-2.1.0.old/src/fx_about.c openfetion-all-2.1.0/src/fx_about.c
--- openfetion-all-2.1.0.old/src/fx_about.c	2010-12-26 12:56:56.000000000 +0800
+++ openfetion-all-2.1.0/src/fx_about.c	2011-01-01 20:14:40.050114000 +0800
@@ -137,7 +137,7 @@
 	GtkTextTag    *tag;
 	GtkTextTag    *tag1;
 	GtkTextIter    iter;
-	const char title[] = N_("Openfetion "FETION_VERSION"\n\n");
+	const char title[] = N_("Openfetion "PACKAGE_VERSION"\n\n");
 	const char intro[] = N_("OpenFetion is a Fetion client for linux based on GTK+2.0, "
 				"implementing Fetion Protocol Version 4.\n"
 				"It supports most features of Fetion. \n"
