--- src/fx_tree.c.orig	2010-07-28 18:46:43.354977604 +0800
+++ src/fx_tree.c	2010-07-28 18:51:41.461648395 +0800
@@ -202,7 +202,7 @@
 	foreach_contactlist(user->contactList , contact){
 		pb = gdk_pixbuf_new_from_file_at_size(SKIN_DIR"fetion.png"
 				, config->iconSize , config->iconSize , NULL);
-		fx_tree_get_group_iter_by_id(GTK_TREE_MODEL(store) , contact->groupid , &iter );
+		fx_tree_get_group_iter_by_id(GTK_TREE_MODEL(store) , contact->groupid , &iter1 );
 		if(contact->state > 0)
 			gtk_tree_store_prepend(store , &iter1 , &iter);
 		else
