--- include/openfetion/fetion_sip.h~	2010-07-20 17:31:19.000000000 +0800
+++ include/openfetion/fetion_sip.h	2010-07-28 19:56:22.111260643 +0800
@@ -94,6 +94,8 @@
 	INCOMING_ACTION_UNKNOWN
 } IncomingActionType;
 
+extern char *strptime(const char *s, const char *format, struct tm *tm);
+
 extern FetionSip* fetion_sip_new(FetionConnection* tcp , const char* sid);
 
 extern FetionSip* fetion_sip_clone(FetionSip* sip);
