diff --git a/driver/Makefile b/driver/Makefile
index 518cbb1..d5e7087 100644
--- a/driver/Makefile
+++ b/driver/Makefile
@@ -1,8 +1,8 @@
-CC          = gcc
-CFLAGS      = -g
+CC          ?= gcc
+CFLAGS      ?= -g
 
 all: ucc.c linux.c
-	$(CC) -o ucc $(CFLAGS) $^
+	$(CC) -o ucc $(CFLAGS) $(LDFLAGS) $^
 
 clean:
 	rm -f *.o ucc
diff --git a/ucl/Makefile b/ucl/Makefile
index 62bd670..66168a6 100644
--- a/ucl/Makefile
+++ b/ucl/Makefile
@@ -4,12 +4,12 @@ C_SRC       = alloc.c ast.c decl.c declchk.c dumpast.c emit.c \
               stmtchk.c str.c symbol.c tranexpr.c transtmt.c type.c \
               ucl.c uildasm.c vector.c x86.c x86linux.c
 OBJS        = $(C_SRC:.c=.o)
-CC          = gcc
-CFLAGS      = -g -D_UCC
+CC          ?= gcc
+CFLAGS      ?= -g -D_UCC
 UCC         = ../driver/ucc
 
 all: $(OBJS) assert.o
-	$(CC) -o ucl $(CFLAGS) $(OBJS)
+	$(CC) -o ucl $(CFLAGS) $(LDFLAGS) $(OBJS)
 
 clean:
 	rm -f *.o ucl
