From aa3c213319f6df61f727a604798e8c120b7c37f4 Mon Sep 17 00:00:00 2001
From: tastytea <tastytea@tastytea.de>
Date: Fri, 30 Apr 2021 00:03:42 +0200
Subject: [PATCH] Remove TTY check in test.

---
 tests/test_std.js | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tests/test_std.js b/tests/test_std.js
index 3ea6e34..a12cb81 100644
--- a/tests/test_std.js
+++ b/tests/test_std.js
@@ -143,8 +143,6 @@ function test_os()
 {
     var fd, fpath, fname, fdir, buf, buf2, i, files, err, fdate, st, link_path;
 
-    assert(os.isatty(0));
-
     fdir = "test_tmp_dir";
     fname = "tmp_file.txt";
     fpath = fdir + "/" + fname;
-- 
2.26.3

