diff -Nur -x '*.orig' -x '*~' fontconfig-2.5.0/conf.d/10-antialias.conf fontconfig-2.5.0.new/conf.d/10-antialias.conf
--- conf.d/10-antialias.conf	1970-01-01 08:00:00.000000000 +0800
+++ conf.d/10-antialias.conf	2008-02-28 04:22:09.000000000 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use the Antialiasing --> 
+  <match target="font">
+    <edit name="antialias" mode="assign"><bool>true</bool></edit>
+  </match>
+</fontconfig>
diff -Nur -x '*.orig' -x '*~' fontconfig-2.5.0/conf.d/10-hinting.conf fontconfig-2.5.0.new/conf.d/10-hinting.conf
--- conf.d/10-hinting.conf	1970-01-01 08:00:00.000000000 +0800
+++ conf.d/10-hinting.conf	2008-02-28 04:22:33.000000000 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Hinting --> 
+  <match target="font">
+    <edit name="hinting" mode="assign"><bool>true</bool></edit>
+  </match>
+</fontconfig>
diff -Nur -x '*.orig' -x '*~' fontconfig-2.5.0/conf.d/10-hinting-full.conf fontconfig-2.5.0.new/conf.d/10-hinting-full.conf
--- conf.d/10-hinting-full.conf	1970-01-01 08:00:00.000000000 +0800
+++ conf.d/10-hinting-full.conf	2008-02-28 04:24:01.000000000 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Full Hinting --> 
+  <match target="font">
+    <edit name="hintstyle" mode="assign"><const>hintfull</const></edit>
+  </match>
+</fontconfig>
diff -Nur -x '*.orig' -x '*~' fontconfig-2.5.0/conf.d/10-hinting-medium.conf fontconfig-2.5.0.new/conf.d/10-hinting-medium.conf
--- conf.d/10-hinting-medium.conf	1970-01-01 08:00:00.000000000 +0800
+++ conf.d/10-hinting-medium.conf	2008-02-28 04:23:23.000000000 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Medium Hinting --> 
+  <match target="font">
+    <edit name="hintstyle" mode="assign"><const>hintmedium</const></edit>
+  </match>
+</fontconfig>
diff -Nur -x '*.orig' -x '*~' fontconfig-2.5.0/conf.d/10-hinting-slight.conf fontconfig-2.5.0.new/conf.d/10-hinting-slight.conf
--- conf.d/10-hinting-slight.conf	1970-01-01 08:00:00.000000000 +0800
+++ conf.d/10-hinting-slight.conf	2008-02-28 04:24:26.000000000 +0800
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<fontconfig>
+<!--  Use Slight Hinting --> 
+  <match target="font">
+    <edit name="hintstyle" mode="assign"><const>hintslight</const></edit>
+  </match>
+</fontconfig>
