--- a/Documentation/laptops/thinkpad-acpi.txt
+++ b/Documentation/laptops/thinkpad-acpi.txt
@@ -1370,6 +1370,10 @@
 	A percentage value from 1-99%, controlling when charging should stop, or
 	0 for the default value.
 
+battery sysfs attribute: force_discharge
+
+	Force discharging (0 or 1)
+
 Multiple Commands, Module Parameters
 ------------------------------------
 
--- a/drivers/platform/x86/thinkpad_acpi.c
+++ b/drivers/platform/x86/thinkpad_acpi.c
@@ -8358,7 +8358,7 @@
 
 /* Modify battery_init() if you modify them */
 #define BATTERY_MAX_COUNT 3
-#define BATTERY_MAX_ATTRS 2
+#define BATTERY_MAX_ATTRS 3
 
 static struct battery {
 	char name[3 + 1 + 1];
@@ -8437,6 +8437,46 @@
 	return snprintf(buf, PAGE_SIZE, "%d\n", value & 0xFF);
 }
 
+static ssize_t battery_force_discharge_show(struct device *dev,
+						struct device_attribute *attr,
+						char *buf)
+{
+	int bat = battery_attribute_get_battery(attr);
+	int value;
+
+	if (!hkey_handle || !acpi_evalf(hkey_handle, &value, "BDSG",
+					"dd", bat))
+		return -EIO;
+
+	return snprintf(buf, PAGE_SIZE, "%d\n", value & 0x1);
+}
+
+static ssize_t battery_force_discharge_store(struct device *dev,
+						struct device_attribute *attr,
+						const char *buf, size_t count)
+{
+	int bat = battery_attribute_get_battery(attr);
+	int res = -EINVAL;
+	unsigned long value;
+
+	res = kstrtoul(buf, 0, &value);
+	if (res || value > 1)
+		return res ? res : -EINVAL;
+
+	if (!hkey_handle || !acpi_evalf(hkey_handle, &res, "BDSG", "dd", bat))
+		return -EIO;
+
+	/* Keep the break on AC disconnect state */
+	value |= (res >> 1) & 0x1;
+	/* Set the battery */
+	value |= (bat << 8);
+
+	if (!hkey_handle || !acpi_evalf(hkey_handle, &res, "BDSS",
+					"dd", (int) value) || res < 0)
+		return -EIO;
+	return count;
+}
+
 static int __init battery_init(struct ibm_init_struct *iibm)
 {
 	int res;
@@ -8480,6 +8520,13 @@
 				       S_IWUSR | S_IRUGO,
 				       battery_stop_charge_thresh_show,
 				       battery_stop_charge_thresh_store),
+			.var = (void *) (unsigned long) (i + 1)
+		};
+		batteries[i].attributes[j++] = (struct dev_ext_attribute) {
+			.attr = __ATTR(force_discharge,
+				       S_IWUSR | S_IRUGO,
+				       battery_force_discharge_show,
+				       battery_force_discharge_store),
 			.var = (void *) (unsigned long) (i + 1)
 		};
 