diff -Nur qt-everywhere-opensource-src-4.7.1.orig/src/gui/text/qfontengine_ft.cpp qt-everywhere-opensource-src-4.7.1/src/gui/text/qfontengine_ft.cpp
--- qt-everywhere-opensource-src-4.7.1.orig/src/gui/text/qfontengine_ft.cpp	2011-01-11 17:48:54.476768399 +0800
+++ qt-everywhere-opensource-src-4.7.1/src/gui/text/qfontengine_ft.cpp	2011-01-11 18:07:13.907768399 +0800
@@ -895,6 +895,7 @@
 
     if (set->outline_drawing)
         load_flags |= FT_LOAD_NO_BITMAP;
+    load_flags |= FT_LOAD_IGNORE_GLOBAL_ADVANCE_WIDTH;
 
     if (format == Format_Mono) {
         load_target = FT_LOAD_TARGET_MONO;
@@ -928,6 +929,7 @@
 
     if (transform)
         load_flags |= FT_LOAD_NO_BITMAP;
+    load_flags |= FT_LOAD_IGNORE_GLOBAL_ADVANCE_WIDTH;
 
     FT_Face face = freetype->face;
     FT_Error err = FT_Load_Glyph(face, glyph, load_flags);
