--- a/examples/basic/CMakeLists.txt
+++ b/examples/basic/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7.0)
+cmake_minimum_required(VERSION 3.10.0)
 
 project(basic LANGUAGES CXX)
 
--- a/examples/calculator/CMakeLists.txt
+++ b/examples/calculator/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7.0)
+cmake_minimum_required(VERSION 3.10.0)
 
 project(calculator LANGUAGES CXX)
 
--- a/examples/sending_arguments/CMakeLists.txt
+++ b/examples/sending_arguments/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7.0)
+cmake_minimum_required(VERSION 3.10.0)
 
 project(sending_arguments LANGUAGES CXX)
 
--- a/examples/separate_object/CMakeLists.txt
+++ b/examples/separate_object/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7.0)
+cmake_minimum_required(VERSION 3.10.0)
 
 project(separate_object LANGUAGES CXX)
 
--- a/examples/windows_raise_widget/CMakeLists.txt
+++ b/examples/windows_raise_widget/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.7.0)
+cmake_minimum_required(VERSION 3.10.0)
 
 project(windows_raise_widget LANGUAGES CXX)
 
